// tailgrid/dist/index-WxTJUbsU.js
import * as n from "react";
import { a as S, u as l } from "./index-CDSAOBSC.js";
function I(t, e, { checkForDefaultPrevented: o = !0 } = {}) {
  return function(c) {
    if (t == null || t(c), o === !1 || !c.defaultPrevented)
      return e == null ? void 0 : e(c);
  };
}
var C = n.useId || (() => {
}), R = 0;
function P(t) {
  const [e, o] = n.useState(C());
  return S(() => {
    o((s) => s ?? String(R++));
  }, [t]), t || (e ? `radix-${e}` : "");
}
function k({
  prop: t,
  defaultProp: e,
  onChange: o = () => {
  }
}) {
  const [s, c] = h({ defaultProp: e, onChange: o }), u = t !== void 0, d = u ? t : s, a = l(o), i = n.useCallback(
    (f) => {
      if (u) {
        const r = typeof f == "function" ? f(t) : f;
        r !== t && a(r);
      } else
        c(f);
    },
    [u, t, c, a]
  );
  return [d, i];
}
function h({
  defaultProp: t,
  onChange: e
}) {
  const o = n.useState(t), [s] = o, c = n.useRef(s), u = l(e);
  return n.useEffect(() => {
    c.current !== s && (u(s), c.current = s);
  }, [s, c, u]), o;
}
export {
  k as a,
  I as c,
  P as u
};
