// tailgrid/dist/proxy-g3OKwrzk.js
import { jsxs as pr, jsx as Os } from "react/jsx-runtime";
import { createContext as mt, useRef as Ut, useLayoutEffect as gr, useEffect as Fe, useContext as O, useId as yr, useCallback as Is, useMemo as Xt, useInsertionEffect as vr, forwardRef as Tr, Fragment as js, createElement as xr, Component as Pr } from "react";
const Ns = mt({});
function Sr(t) {
  const e = Ut(null);
  return e.current === null && (e.current = t()), e.current;
}
const ke = typeof window < "u", br = ke ? gr : Fe, Oe = /* @__PURE__ */ mt(null), Us = mt({
  transformPagePoint: (t) => t,
  isStatic: !1,
  reducedMotion: "never"
});
function Ar(t = !0) {
  const e = O(Oe);
  if (e === null)
    return [!0, null];
  const { isPresent: n, onExitComplete: s, register: i } = e, o = yr();
  Fe(() => {
    if (t)
      return i(o);
  }, [t]);
  const r = Is(() => t && s && s(o), [o, s, t]);
  return !n && s ? [!1, r] : [!0];
}
const I = /* @__NO_SIDE_EFFECTS__ */ (t) => t;
let pt = I, X = I;
process.env.NODE_ENV !== "production" && (pt = (t, e) => {
  !t && typeof console < "u" && console.warn(e);
}, X = (t, e) => {
  if (!t)
    throw new Error(e);
});
const wr = {
  useManualTiming: !1
}, It = [
  "read",
  // Read
  "resolveKeyframes",
  // Write/Read/Write/Read
  "update",
  // Compute
  "preRender",
  // Compute
  "render",
  // Write
  "postRender"
  // Compute
], gn = {
  value: null
};
function Vr(t, e) {
  let n = /* @__PURE__ */ new Set(), s = /* @__PURE__ */ new Set(), i = !1, o = !1;
  const r = /* @__PURE__ */ new WeakSet();
  let a = {
    delta: 0,
    timestamp: 0,
    isProcessing: !1
  }, l = 0;
  function u(h) {
    r.has(h) && (c.schedule(h), t()), l++, h(a);
  }
  const c = {
    /**
     * Schedule a process to run on the next frame.
     */
    schedule: (h, f = !1, d = !1) => {
      const p = d && i ? n : s;
      return f && r.add(h), p.has(h) || p.add(h), h;
    },
    /**
     * Cancel the provided callback from running on the next frame.
     */
    cancel: (h) => {
      s.delete(h), r.delete(h);
    },
    /**
     * Execute all schedule callbacks.
     */
    process: (h) => {
      if (a = h, i) {
        o = !0;
        return;
      }
      i = !0, [n, s] = [s, n], n.forEach(u), e && gn.value && gn.value.frameloop[e].push(l), l = 0, n.clear(), i = !1, o && (o = !1, c.process(h));
    }
  };
  return c;
}
const Dr = 40;
function _s(t, e) {
  let n = !1, s = !0;
  const i = {
    delta: 0,
    timestamp: 0,
    isProcessing: !1
  }, o = () => n = !0, r = It.reduce((g, v) => (g[v] = Vr(o, e ? v : void 0), g), {}), { read: a, resolveKeyframes: l, update: u, preRender: c, render: h, postRender: f } = r, d = () => {
    const g = performance.now();
    n = !1, i.delta = s ? 1e3 / 60 : Math.max(Math.min(g - i.timestamp, Dr), 1), i.timestamp = g, i.isProcessing = !0, a.process(i), l.process(i), u.process(i), c.process(i), h.process(i), f.process(i), i.isProcessing = !1, n && e && (s = !1, t(d));
  }, m = () => {
    n = !0, s = !0, i.isProcessing || t(d);
  };
  return { schedule: It.reduce((g, v) => {
    const T = r[v];
    return g[v] = (b, P = !1, A = !1) => (n || m(), T.schedule(b, P, A)), g;
  }, {}), cancel: (g) => {
    for (let v = 0; v < It.length; v++)
      r[It[v]].cancel(g);
  }, state: i, steps: r };
}
const { schedule: w, cancel: q, state: R, steps: se } = /* @__PURE__ */ _s(typeof requestAnimationFrame < "u" ? requestAnimationFrame : I, !0), Ks = mt({ strict: !1 }), yn = {
  animation: [
    "animate",
    "variants",
    "whileHover",
    "whileTap",
    "exit",
    "whileInView",
    "whileFocus",
    "whileDrag"
  ],
  exit: ["exit"],
  drag: ["drag", "dragControls"],
  focus: ["whileFocus"],
  hover: ["whileHover", "onHoverStart", "onHoverEnd"],
  tap: ["whileTap", "onTap", "onTapStart", "onTapCancel"],
  pan: ["onPan", "onPanStart", "onPanSessionStart", "onPanEnd"],
  inView: ["whileInView", "onViewportEnter", "onViewportLeave"],
  layout: ["layout", "layoutId"]
}, ht = {};
for (const t in yn)
  ht[t] = {
    isEnabled: (e) => yn[t].some((n) => !!e[n])
  };
function Mr(t) {
  for (const e in t)
    ht[e] = {
      ...ht[e],
      ...t[e]
    };
}
const Cr = /* @__PURE__ */ new Set([
  "animate",
  "exit",
  "variants",
  "initial",
  "style",
  "values",
  "variants",
  "transition",
  "transformTemplate",
  "custom",
  "inherit",
  "onBeforeLayoutMeasure",
  "onAnimationStart",
  "onAnimationComplete",
  "onUpdate",
  "onDragStart",
  "onDrag",
  "onDragEnd",
  "onMeasureDragConstraints",
  "onDirectionLock",
  "onDragTransitionEnd",
  "_dragX",
  "_dragY",
  "onHoverStart",
  "onHoverEnd",
  "onViewportEnter",
  "onViewportLeave",
  "globalTapTarget",
  "ignoreStrict",
  "viewport"
]);
function Wt(t) {
  return t.startsWith("while") || t.startsWith("drag") && t !== "draggable" || t.startsWith("layout") || t.startsWith("onTap") || t.startsWith("onPan") || t.startsWith("onLayout") || Cr.has(t);
}
let $s = (t) => !Wt(t);
function Rr(t) {
  t && ($s = (e) => e.startsWith("on") ? !Wt(e) : t(e));
}
try {
  Rr(require("@emotion/is-prop-valid").default);
} catch {
}
function Er(t, e, n) {
  const s = {};
  for (const i in t)
    i === "values" && typeof t.values == "object" || ($s(i) || n === !0 && Wt(i) || !e && !Wt(i) || // If trying to use native HTML drag events, forward drag listeners
    t.draggable && i.startsWith("onDrag")) && (s[i] = t[i]);
  return s;
}
const vn = /* @__PURE__ */ new Set();
function qt(t, e, n) {
  t || vn.has(e) || (console.warn(e), vn.add(e));
}
function Lr(t) {
  if (typeof Proxy > "u")
    return t;
  const e = /* @__PURE__ */ new Map(), n = (...s) => (process.env.NODE_ENV !== "production" && qt(!1, "motion() is deprecated. Use motion.create() instead."), t(...s));
  return new Proxy(n, {
    /**
     * Called when `motion` is referenced with a prop: `motion.div`, `motion.input` etc.
     * The prop name is passed through as `key` and we can use that to generate a `motion`
     * DOM component with that name.
     */
    get: (s, i) => i === "create" ? t : (e.has(i) || e.set(i, t(i)), e.get(i))
  });
}
const Zt = /* @__PURE__ */ mt({});
function Jt(t) {
  return t !== null && typeof t == "object" && typeof t.start == "function";
}
function Vt(t) {
  return typeof t == "string" || Array.isArray(t);
}
const Ie = [
  "animate",
  "whileInView",
  "whileFocus",
  "whileHover",
  "whileTap",
  "whileDrag",
  "exit"
], je = ["initial", ...Ie];
function Qt(t) {
  return Jt(t.animate) || je.some((e) => Vt(t[e]));
}
function Ws(t) {
  return !!(Qt(t) || t.variants);
}
function Br(t, e) {
  if (Qt(t)) {
    const { initial: n, animate: s } = t;
    return {
      initial: n === !1 || Vt(n) ? n : void 0,
      animate: Vt(s) ? s : void 0
    };
  }
  return t.inherit !== !1 ? e : {};
}
function Fr(t) {
  const { initial: e, animate: n } = Br(t, O(Zt));
  return Xt(() => ({ initial: e, animate: n }), [Tn(e), Tn(n)]);
}
function Tn(t) {
  return Array.isArray(t) ? t.join(" ") : t;
}
const kr = Symbol.for("motionComponentSymbol");
function ot(t) {
  return t && typeof t == "object" && Object.prototype.hasOwnProperty.call(t, "current");
}
function Or(t, e, n) {
  return Is(
    (s) => {
      s && t.onMount && t.onMount(s), e && (s ? e.mount(s) : e.unmount()), n && (typeof n == "function" ? n(s) : ot(n) && (n.current = s));
    },
    /**
     * Only pass a new ref callback to React if we've received a visual element
     * factory. Otherwise we'll be mounting/remounting every time externalRef
     * or other dependencies change.
     */
    [e]
  );
}
const { schedule: Ne } = /* @__PURE__ */ _s(queueMicrotask, !1), Ue = (t) => t.replace(/([a-z])([A-Z])/gu, "$1-$2").toLowerCase(), Ir = "framerAppearId", Gs = "data-" + Ue(Ir), zs = mt({});
function jr(t, e, n, s, i) {
  var o, r;
  const { visualElement: a } = O(Zt), l = O(Ks), u = O(Oe), c = O(Us).reducedMotion, h = Ut(null);
  s = s || l.renderer, !h.current && s && (h.current = s(t, {
    visualState: e,
    parent: a,
    props: n,
    presenceContext: u,
    blockInitialAnimation: u ? u.initial === !1 : !1,
    reducedMotionConfig: c
  }));
  const f = h.current, d = O(zs);
  f && !f.projection && i && (f.type === "html" || f.type === "svg") && Nr(h.current, n, i, d);
  const m = Ut(!1);
  vr(() => {
    f && m.current && f.update(n, u);
  });
  const p = n[Gs], y = Ut(!!p && !(!((o = window.MotionHandoffIsComplete) === null || o === void 0) && o.call(window, p)) && ((r = window.MotionHasOptimisedAnimation) === null || r === void 0 ? void 0 : r.call(window, p)));
  return br(() => {
    f && (m.current = !0, window.MotionIsMounted = !0, f.updateFeatures(), Ne.render(f.render), y.current && f.animationState && f.animationState.animateChanges());
  }), Fe(() => {
    f && (!y.current && f.animationState && f.animationState.animateChanges(), y.current && (queueMicrotask(() => {
      var g;
      (g = window.MotionHandoffMarkAsComplete) === null || g === void 0 || g.call(window, p);
    }), y.current = !1));
  }), f;
}
function Nr(t, e, n, s) {
  const { layoutId: i, layout: o, drag: r, dragConstraints: a, layoutScroll: l, layoutRoot: u } = e;
  t.projection = new n(t.latestValues, e["data-framer-portal-id"] ? void 0 : Hs(t.parent)), t.projection.setOptions({
    layoutId: i,
    layout: o,
    alwaysMeasureLayout: !!r || a && ot(a),
    visualElement: t,
    /**
     * TODO: Update options in an effect. This could be tricky as it'll be too late
     * to update by the time layout animations run.
     * We also need to fix this safeToRemove by linking it up to the one returned by usePresence,
     * ensuring it gets called if there's no potential layout animations.
     *
     */
    animationType: typeof o == "string" ? o : "both",
    initialPromotionConfig: s,
    layoutScroll: l,
    layoutRoot: u
  });
}
function Hs(t) {
  if (t)
    return t.options.allowProjection !== !1 ? t.projection : Hs(t.parent);
}
function Ur({ preloadedFeatures: t, createVisualElement: e, useRender: n, useVisualState: s, Component: i }) {
  var o, r;
  t && Mr(t);
  function a(u, c) {
    let h;
    const f = {
      ...O(Us),
      ...u,
      layoutId: _r(u)
    }, { isStatic: d } = f, m = Fr(u), p = s(u, d);
    if (!d && ke) {
      Kr(f, t);
      const y = $r(f);
      h = y.MeasureLayout, m.visualElement = jr(i, p, f, e, y.ProjectionNode);
    }
    return pr(Zt.Provider, { value: m, children: [h && m.visualElement ? Os(h, { visualElement: m.visualElement, ...f }) : null, n(i, u, Or(p, m.visualElement, c), p, d, m.visualElement)] });
  }
  a.displayName = `motion.${typeof i == "string" ? i : `create(${(r = (o = i.displayName) !== null && o !== void 0 ? o : i.name) !== null && r !== void 0 ? r : ""})`}`;
  const l = Tr(a);
  return l[kr] = i, l;
}
function _r({ layoutId: t }) {
  const e = O(Ns).id;
  return e && t !== void 0 ? e + "-" + t : t;
}
function Kr(t, e) {
  const n = O(Ks).strict;
  if (process.env.NODE_ENV !== "production" && e && n) {
    const s = "You have rendered a `motion` component within a `LazyMotion` component. This will break tree shaking. Import and render a `m` component instead.";
    t.ignoreStrict ? pt(!1, s) : X(!1, s);
  }
}
function $r(t) {
  const { drag: e, layout: n } = ht;
  if (!e && !n)
    return {};
  const s = { ...e, ...n };
  return {
    MeasureLayout: e != null && e.isEnabled(t) || n != null && n.isEnabled(t) ? s.MeasureLayout : void 0,
    ProjectionNode: s.ProjectionNode
  };
}
const Ys = (t) => (e) => typeof e == "string" && e.startsWith(t), _e = /* @__PURE__ */ Ys("--"), Wr = /* @__PURE__ */ Ys("var(--"), Ke = (t) => Wr(t) ? Gr.test(t.split("/*")[0].trim()) : !1, Gr = /var\(--(?:[\w-]+\s*|[\w-]+\s*,(?:\s*[^)(\s]|\s*\((?:[^)(]|\([^)(]*\))*\))+\s*)\)$/iu, Dt = {};
function zr(t) {
  for (const e in t)
    Dt[e] = t[e], _e(e) && (Dt[e].isCSSVariable = !0);
}
const gt = [
  "transformPerspective",
  "x",
  "y",
  "z",
  "translateX",
  "translateY",
  "translateZ",
  "scale",
  "scaleX",
  "scaleY",
  "rotate",
  "rotateX",
  "rotateY",
  "rotateZ",
  "skew",
  "skewX",
  "skewY"
], it = new Set(gt);
function Xs(t, { layout: e, layoutId: n }) {
  return it.has(t) || t.startsWith("origin") || (e || n !== void 0) && (!!Dt[t] || t === "opacity");
}
const L = (t) => !!(t && t.getVelocity), qs = (t, e) => e && typeof t == "number" ? e.transform(t) : t, H = (t, e, n) => n > e ? e : n < t ? t : n, yt = {
  test: (t) => typeof t == "number",
  parse: parseFloat,
  transform: (t) => t
}, Mt = {
  ...yt,
  transform: (t) => H(0, 1, t)
}, jt = {
  ...yt,
  default: 1
}, Bt = (t) => ({
  test: (e) => typeof e == "string" && e.endsWith(t) && e.split(" ").length === 1,
  parse: parseFloat,
  transform: (e) => `${e}${t}`
}), Y = /* @__PURE__ */ Bt("deg"), K = /* @__PURE__ */ Bt("%"), x = /* @__PURE__ */ Bt("px"), Hr = /* @__PURE__ */ Bt("vh"), Yr = /* @__PURE__ */ Bt("vw"), xn = {
  ...K,
  parse: (t) => K.parse(t) / 100,
  transform: (t) => K.transform(t * 100)
}, Xr = {
  // Border props
  borderWidth: x,
  borderTopWidth: x,
  borderRightWidth: x,
  borderBottomWidth: x,
  borderLeftWidth: x,
  borderRadius: x,
  radius: x,
  borderTopLeftRadius: x,
  borderTopRightRadius: x,
  borderBottomRightRadius: x,
  borderBottomLeftRadius: x,
  // Positioning props
  width: x,
  maxWidth: x,
  height: x,
  maxHeight: x,
  top: x,
  right: x,
  bottom: x,
  left: x,
  // Spacing props
  padding: x,
  paddingTop: x,
  paddingRight: x,
  paddingBottom: x,
  paddingLeft: x,
  margin: x,
  marginTop: x,
  marginRight: x,
  marginBottom: x,
  marginLeft: x,
  // Misc
  backgroundPositionX: x,
  backgroundPositionY: x
}, qr = {
  rotate: Y,
  rotateX: Y,
  rotateY: Y,
  rotateZ: Y,
  scale: jt,
  scaleX: jt,
  scaleY: jt,
  scaleZ: jt,
  skew: Y,
  skewX: Y,
  skewY: Y,
  distance: x,
  translateX: x,
  translateY: x,
  translateZ: x,
  x,
  y: x,
  z: x,
  perspective: x,
  transformPerspective: x,
  opacity: Mt,
  originX: xn,
  originY: xn,
  originZ: x
}, Pn = {
  ...yt,
  transform: Math.round
}, $e = {
  ...Xr,
  ...qr,
  zIndex: Pn,
  size: x,
  // SVG
  fillOpacity: Mt,
  strokeOpacity: Mt,
  numOctaves: Pn
}, Zr = {
  x: "translateX",
  y: "translateY",
  z: "translateZ",
  transformPerspective: "perspective"
}, Jr = gt.length;
function Qr(t, e, n) {
  let s = "", i = !0;
  for (let o = 0; o < Jr; o++) {
    const r = gt[o], a = t[r];
    if (a === void 0)
      continue;
    let l = !0;
    if (typeof a == "number" ? l = a === (r.startsWith("scale") ? 1 : 0) : l = parseFloat(a) === 0, !l || n) {
      const u = qs(a, $e[r]);
      if (!l) {
        i = !1;
        const c = Zr[r] || r;
        s += `${c}(${u}) `;
      }
      n && (e[r] = u);
    }
  }
  return s = s.trim(), n ? s = n(e, i ? "" : s) : i && (s = "none"), s;
}
function We(t, e, n) {
  const { style: s, vars: i, transformOrigin: o } = t;
  let r = !1, a = !1;
  for (const l in e) {
    const u = e[l];
    if (it.has(l)) {
      r = !0;
      continue;
    } else if (_e(l)) {
      i[l] = u;
      continue;
    } else {
      const c = qs(u, $e[l]);
      l.startsWith("origin") ? (a = !0, o[l] = c) : s[l] = c;
    }
  }
  if (e.transform || (r || n ? s.transform = Qr(e, t.transform, n) : s.transform && (s.transform = "none")), a) {
    const { originX: l = "50%", originY: u = "50%", originZ: c = 0 } = o;
    s.transformOrigin = `${l} ${u} ${c}`;
  }
}
const Ge = () => ({
  style: {},
  transform: {},
  transformOrigin: {},
  vars: {}
});
function Zs(t, e, n) {
  for (const s in e)
    !L(e[s]) && !Xs(s, n) && (t[s] = e[s]);
}
function to({ transformTemplate: t }, e) {
  return Xt(() => {
    const n = Ge();
    return We(n, e, t), Object.assign({}, n.vars, n.style);
  }, [e]);
}
function eo(t, e) {
  const n = t.style || {}, s = {};
  return Zs(s, n, t), Object.assign(s, to(t, e)), s;
}
function no(t, e) {
  const n = {}, s = eo(t, e);
  return t.drag && t.dragListener !== !1 && (n.draggable = !1, s.userSelect = s.WebkitUserSelect = s.WebkitTouchCallout = "none", s.touchAction = t.drag === !0 ? "none" : `pan-${t.drag === "x" ? "y" : "x"}`), t.tabIndex === void 0 && (t.onTap || t.onTapStart || t.whileTap) && (n.tabIndex = 0), n.style = s, n;
}
const so = [
  "animate",
  "circle",
  "defs",
  "desc",
  "ellipse",
  "g",
  "image",
  "line",
  "filter",
  "marker",
  "mask",
  "metadata",
  "path",
  "pattern",
  "polygon",
  "polyline",
  "rect",
  "stop",
  "switch",
  "symbol",
  "svg",
  "text",
  "tspan",
  "use",
  "view"
];
function ze(t) {
  return (
    /**
     * If it's not a string, it's a custom React component. Currently we only support
     * HTML custom React components.
     */
    typeof t != "string" || /**
     * If it contains a dash, the element is a custom HTML webcomponent.
     */
    t.includes("-") ? !1 : (
      /**
       * If it's in our list of lowercase SVG tags, it's an SVG component
       */
      !!(so.indexOf(t) > -1 || /**
       * If it contains a capital letter, it's an SVG component
       */
      /[A-Z]/u.test(t))
    )
  );
}
const io = {
  offset: "stroke-dashoffset",
  array: "stroke-dasharray"
}, ro = {
  offset: "strokeDashoffset",
  array: "strokeDasharray"
};
function oo(t, e, n = 1, s = 0, i = !0) {
  t.pathLength = 1;
  const o = i ? io : ro;
  t[o.offset] = x.transform(-s);
  const r = x.transform(e), a = x.transform(n);
  t[o.array] = `${r} ${a}`;
}
function Sn(t, e, n) {
  return typeof t == "string" ? t : x.transform(e + n * t);
}
function ao(t, e, n) {
  const s = Sn(e, t.x, t.width), i = Sn(n, t.y, t.height);
  return `${s} ${i}`;
}
function He(t, {
  attrX: e,
  attrY: n,
  attrScale: s,
  originX: i,
  originY: o,
  pathLength: r,
  pathSpacing: a = 1,
  pathOffset: l = 0,
  // This is object creation, which we try to avoid per-frame.
  ...u
}, c, h) {
  if (We(t, u, h), c) {
    t.style.viewBox && (t.attrs.viewBox = t.style.viewBox);
    return;
  }
  t.attrs = t.style, t.style = {};
  const { attrs: f, style: d, dimensions: m } = t;
  f.transform && (m && (d.transform = f.transform), delete f.transform), m && (i !== void 0 || o !== void 0 || d.transform) && (d.transformOrigin = ao(m, i !== void 0 ? i : 0.5, o !== void 0 ? o : 0.5)), e !== void 0 && (f.x = e), n !== void 0 && (f.y = n), s !== void 0 && (f.scale = s), r !== void 0 && oo(f, r, a, l, !1);
}
const Js = () => ({
  ...Ge(),
  attrs: {}
}), Ye = (t) => typeof t == "string" && t.toLowerCase() === "svg";
function lo(t, e, n, s) {
  const i = Xt(() => {
    const o = Js();
    return He(o, e, Ye(s), t.transformTemplate), {
      ...o.attrs,
      style: { ...o.style }
    };
  }, [e]);
  if (t.style) {
    const o = {};
    Zs(o, t.style, t), i.style = { ...o, ...i.style };
  }
  return i;
}
function uo(t = !1) {
  return (n, s, i, { latestValues: o }, r) => {
    const l = (ze(n) ? lo : no)(s, o, r, n), u = Er(s, typeof n == "string", t), c = n !== js ? { ...u, ...l, ref: i } : {}, { children: h } = s, f = Xt(() => L(h) ? h.get() : h, [h]);
    return xr(n, {
      ...c,
      children: f
    });
  };
}
function bn(t) {
  const e = [{}, {}];
  return t == null || t.values.forEach((n, s) => {
    e[0][s] = n.get(), e[1][s] = n.getVelocity();
  }), e;
}
function Xe(t, e, n, s) {
  if (typeof e == "function") {
    const [i, o] = bn(s);
    e = e(n !== void 0 ? n : t.custom, i, o);
  }
  if (typeof e == "string" && (e = t.variants && t.variants[e]), typeof e == "function") {
    const [i, o] = bn(s);
    e = e(n !== void 0 ? n : t.custom, i, o);
  }
  return e;
}
const pe = (t) => Array.isArray(t), co = (t) => !!(t && typeof t == "object" && t.mix && t.toValue), ho = (t) => pe(t) ? t[t.length - 1] || 0 : t;
function _t(t) {
  const e = L(t) ? t.get() : t;
  return co(e) ? e.toValue() : e;
}
function fo({ scrapeMotionValuesFromProps: t, createRenderState: e, onUpdate: n }, s, i, o) {
  const r = {
    latestValues: mo(s, i, o, t),
    renderState: e()
  };
  return n && (r.onMount = (a) => n({ props: s, current: a, ...r }), r.onUpdate = (a) => n(a)), r;
}
const Qs = (t) => (e, n) => {
  const s = O(Zt), i = O(Oe), o = () => fo(t, e, s, i);
  return n ? o() : Sr(o);
};
function mo(t, e, n, s) {
  const i = {}, o = s(t, {});
  for (const f in o)
    i[f] = _t(o[f]);
  let { initial: r, animate: a } = t;
  const l = Qt(t), u = Ws(t);
  e && u && !l && t.inherit !== !1 && (r === void 0 && (r = e.initial), a === void 0 && (a = e.animate));
  let c = n ? n.initial === !1 : !1;
  c = c || r === !1;
  const h = c ? a : r;
  if (h && typeof h != "boolean" && !Jt(h)) {
    const f = Array.isArray(h) ? h : [h];
    for (let d = 0; d < f.length; d++) {
      const m = Xe(t, f[d]);
      if (m) {
        const { transitionEnd: p, transition: y, ...g } = m;
        for (const v in g) {
          let T = g[v];
          if (Array.isArray(T)) {
            const b = c ? T.length - 1 : 0;
            T = T[b];
          }
          T !== null && (i[v] = T);
        }
        for (const v in p)
          i[v] = p[v];
      }
    }
  }
  return i;
}
function qe(t, e, n) {
  var s;
  const { style: i } = t, o = {};
  for (const r in i)
    (L(i[r]) || e.style && L(e.style[r]) || Xs(r, t) || ((s = n == null ? void 0 : n.getValue(r)) === null || s === void 0 ? void 0 : s.liveStyle) !== void 0) && (o[r] = i[r]);
  return o;
}
const po = {
  useVisualState: Qs({
    scrapeMotionValuesFromProps: qe,
    createRenderState: Ge
  })
};
function ti(t, e) {
  try {
    e.dimensions = typeof t.getBBox == "function" ? t.getBBox() : t.getBoundingClientRect();
  } catch {
    e.dimensions = {
      x: 0,
      y: 0,
      width: 0,
      height: 0
    };
  }
}
function ei(t, { style: e, vars: n }, s, i) {
  Object.assign(t.style, e, i && i.getProjectionStyles(s));
  for (const o in n)
    t.style.setProperty(o, n[o]);
}
const ni = /* @__PURE__ */ new Set([
  "baseFrequency",
  "diffuseConstant",
  "kernelMatrix",
  "kernelUnitLength",
  "keySplines",
  "keyTimes",
  "limitingConeAngle",
  "markerHeight",
  "markerWidth",
  "numOctaves",
  "targetX",
  "targetY",
  "surfaceScale",
  "specularConstant",
  "specularExponent",
  "stdDeviation",
  "tableValues",
  "viewBox",
  "gradientTransform",
  "pathLength",
  "startOffset",
  "textLength",
  "lengthAdjust"
]);
function si(t, e, n, s) {
  ei(t, e, void 0, s);
  for (const i in e.attrs)
    t.setAttribute(ni.has(i) ? i : Ue(i), e.attrs[i]);
}
function ii(t, e, n) {
  const s = qe(t, e, n);
  for (const i in t)
    if (L(t[i]) || L(e[i])) {
      const o = gt.indexOf(i) !== -1 ? "attr" + i.charAt(0).toUpperCase() + i.substring(1) : i;
      s[o] = t[i];
    }
  return s;
}
const An = ["x", "y", "width", "height", "cx", "cy", "r"], go = {
  useVisualState: Qs({
    scrapeMotionValuesFromProps: ii,
    createRenderState: Js,
    onUpdate: ({ props: t, prevProps: e, current: n, renderState: s, latestValues: i }) => {
      if (!n)
        return;
      let o = !!t.drag;
      if (!o) {
        for (const a in i)
          if (it.has(a)) {
            o = !0;
            break;
          }
      }
      if (!o)
        return;
      let r = !e;
      if (e)
        for (let a = 0; a < An.length; a++) {
          const l = An[a];
          t[l] !== e[l] && (r = !0);
        }
      r && w.read(() => {
        ti(n, s), w.render(() => {
          He(s, i, Ye(n.tagName), t.transformTemplate), si(n, s);
        });
      });
    }
  })
};
function yo(t, e) {
  return function(s, { forwardMotionProps: i } = { forwardMotionProps: !1 }) {
    const r = {
      ...ze(s) ? go : po,
      preloadedFeatures: t,
      useRender: uo(i),
      createVisualElement: e,
      Component: s
    };
    return Ur(r);
  };
}
function Ct(t, e, n) {
  const s = t.getProps();
  return Xe(s, e, n !== void 0 ? n : s.custom, t);
}
function Ze(t, e) {
  return t ? t[e] || t.default || t : void 0;
}
const ri = /* @__PURE__ */ new Set([
  "width",
  "height",
  "top",
  "left",
  "right",
  "bottom",
  ...gt
]);
function Je(t, e) {
  t.indexOf(e) === -1 && t.push(e);
}
function Qe(t, e) {
  const n = t.indexOf(e);
  n > -1 && t.splice(n, 1);
}
class tn {
  constructor() {
    this.subscriptions = [];
  }
  add(e) {
    return Je(this.subscriptions, e), () => Qe(this.subscriptions, e);
  }
  notify(e, n, s) {
    const i = this.subscriptions.length;
    if (i)
      if (i === 1)
        this.subscriptions[0](e, n, s);
      else
        for (let o = 0; o < i; o++) {
          const r = this.subscriptions[o];
          r && r(e, n, s);
        }
  }
  getSize() {
    return this.subscriptions.length;
  }
  clear() {
    this.subscriptions.length = 0;
  }
}
function oi(t, e) {
  return e ? t * (1e3 / e) : 0;
}
let Kt;
function vo() {
  Kt = void 0;
}
const $ = {
  now: () => (Kt === void 0 && $.set(R.isProcessing || wr.useManualTiming ? R.timestamp : performance.now()), Kt),
  set: (t) => {
    Kt = t, queueMicrotask(vo);
  }
}, wn = 30, To = (t) => !isNaN(parseFloat(t));
class xo {
  /**
   * @param init - The initiating value
   * @param config - Optional configuration options
   *
   * -  `transformer`: A function to transform incoming values with.
   */
  constructor(e, n = {}) {
    this.version = "12.5.0", this.canTrackVelocity = null, this.events = {}, this.updateAndNotify = (s, i = !0) => {
      const o = $.now();
      this.updatedAt !== o && this.setPrevFrameValue(), this.prev = this.current, this.setCurrent(s), this.current !== this.prev && this.events.change && this.events.change.notify(this.current), i && this.events.renderRequest && this.events.renderRequest.notify(this.current);
    }, this.hasAnimated = !1, this.setCurrent(e), this.owner = n.owner;
  }
  setCurrent(e) {
    this.current = e, this.updatedAt = $.now(), this.canTrackVelocity === null && e !== void 0 && (this.canTrackVelocity = To(this.current));
  }
  setPrevFrameValue(e = this.current) {
    this.prevFrameValue = e, this.prevUpdatedAt = this.updatedAt;
  }
  /**
   * Adds a function that will be notified when the `MotionValue` is updated.
   *
   * It returns a function that, when called, will cancel the subscription.
   *
   * When calling `onChange` inside a React component, it should be wrapped with the
   * `useEffect` hook. As it returns an unsubscribe function, this should be returned
   * from the `useEffect` function to ensure you don't add duplicate subscribers..
   *
   * ```jsx
   * export const MyComponent = () => {
   *   const x = useMotionValue(0)
   *   const y = useMotionValue(0)
   *   const opacity = useMotionValue(1)
   *
   *   useEffect(() => {
   *     function updateOpacity() {
   *       const maxXY = Math.max(x.get(), y.get())
   *       const newOpacity = transform(maxXY, [0, 100], [1, 0])
   *       opacity.set(newOpacity)
   *     }
   *
   *     const unsubscribeX = x.on("change", updateOpacity)
   *     const unsubscribeY = y.on("change", updateOpacity)
   *
   *     return () => {
   *       unsubscribeX()
   *       unsubscribeY()
   *     }
   *   }, [])
   *
   *   return <motion.div style={{ x }} />
   * }
   * ```
   *
   * @param subscriber - A function that receives the latest value.
   * @returns A function that, when called, will cancel this subscription.
   *
   * @deprecated
   */
  onChange(e) {
    return process.env.NODE_ENV !== "production" && qt(!1, 'value.onChange(callback) is deprecated. Switch to value.on("change", callback).'), this.on("change", e);
  }
  on(e, n) {
    this.events[e] || (this.events[e] = new tn());
    const s = this.events[e].add(n);
    return e === "change" ? () => {
      s(), w.read(() => {
        this.events.change.getSize() || this.stop();
      });
    } : s;
  }
  clearListeners() {
    for (const e in this.events)
      this.events[e].clear();
  }
  /**
   * Attaches a passive effect to the `MotionValue`.
   */
  attach(e, n) {
    this.passiveEffect = e, this.stopPassiveEffect = n;
  }
  /**
   * Sets the state of the `MotionValue`.
   *
   * @remarks
   *
   * ```jsx
   * const x = useMotionValue(0)
   * x.set(10)
   * ```
   *
   * @param latest - Latest value to set.
   * @param render - Whether to notify render subscribers. Defaults to `true`
   *
   * @public
   */
  set(e, n = !0) {
    !n || !this.passiveEffect ? this.updateAndNotify(e, n) : this.passiveEffect(e, this.updateAndNotify);
  }
  setWithVelocity(e, n, s) {
    this.set(n), this.prev = void 0, this.prevFrameValue = e, this.prevUpdatedAt = this.updatedAt - s;
  }
  /**
   * Set the state of the `MotionValue`, stopping any active animations,
   * effects, and resets velocity to `0`.
   */
  jump(e, n = !0) {
    this.updateAndNotify(e), this.prev = e, this.prevUpdatedAt = this.prevFrameValue = void 0, n && this.stop(), this.stopPassiveEffect && this.stopPassiveEffect();
  }
  /**
   * Returns the latest state of `MotionValue`
   *
   * @returns - The latest state of `MotionValue`
   *
   * @public
   */
  get() {
    return this.current;
  }
  /**
   * @public
   */
  getPrevious() {
    return this.prev;
  }
  /**
   * Returns the latest velocity of `MotionValue`
   *
   * @returns - The latest velocity of `MotionValue`. Returns `0` if the state is non-numerical.
   *
   * @public
   */
  getVelocity() {
    const e = $.now();
    if (!this.canTrackVelocity || this.prevFrameValue === void 0 || e - this.updatedAt > wn)
      return 0;
    const n = Math.min(this.updatedAt - this.prevUpdatedAt, wn);
    return oi(parseFloat(this.current) - parseFloat(this.prevFrameValue), n);
  }
  /**
   * Registers a new animation to control this `MotionValue`. Only one
   * animation can drive a `MotionValue` at one time.
   *
   * ```jsx
   * value.start()
   * ```
   *
   * @param animation - A function that starts the provided animation
   */
  start(e) {
    return this.stop(), new Promise((n) => {
      this.hasAnimated = !0, this.animation = e(n), this.events.animationStart && this.events.animationStart.notify();
    }).then(() => {
      this.events.animationComplete && this.events.animationComplete.notify(), this.clearAnimation();
    });
  }
  /**
   * Stop the currently active animation.
   *
   * @public
   */
  stop() {
    this.animation && (this.animation.stop(), this.events.animationCancel && this.events.animationCancel.notify()), this.clearAnimation();
  }
  /**
   * Returns `true` if this value is currently animating.
   *
   * @public
   */
  isAnimating() {
    return !!this.animation;
  }
  clearAnimation() {
    delete this.animation;
  }
  /**
   * Destroy and clean up subscribers to this `MotionValue`.
   *
   * The `MotionValue` hooks like `useMotionValue` and `useTransform` automatically
   * handle the lifecycle of the returned `MotionValue`, so this method is only necessary if you've manually
   * created a `MotionValue` via the `motionValue` function.
   *
   * @public
   */
  destroy() {
    this.clearListeners(), this.stop(), this.stopPassiveEffect && this.stopPassiveEffect();
  }
}
function Rt(t, e) {
  return new xo(t, e);
}
function Po(t, e, n) {
  t.hasValue(e) ? t.getValue(e).set(n) : t.addValue(e, Rt(n));
}
function So(t, e) {
  const n = Ct(t, e);
  let { transitionEnd: s = {}, transition: i = {}, ...o } = n || {};
  o = { ...o, ...s };
  for (const r in o) {
    const a = ho(o[r]);
    Po(t, r, a);
  }
}
function bo(t) {
  return !!(L(t) && t.add);
}
function ge(t, e) {
  const n = t.getValue("willChange");
  if (bo(n))
    return n.add(e);
}
function ai(t) {
  return t.props[Gs];
}
// @__NO_SIDE_EFFECTS__
function en(t) {
  let e;
  return () => (e === void 0 && (e = t()), e);
}
const Ao = /* @__PURE__ */ en(() => window.ScrollTimeline !== void 0);
class wo {
  constructor(e) {
    this.stop = () => this.runAll("stop"), this.animations = e.filter(Boolean);
  }
  get finished() {
    return Promise.all(this.animations.map((e) => "finished" in e ? e.finished : e));
  }
  /**
   * TODO: Filter out cancelled or stopped animations before returning
   */
  getAll(e) {
    return this.animations[0][e];
  }
  setAll(e, n) {
    for (let s = 0; s < this.animations.length; s++)
      this.animations[s][e] = n;
  }
  attachTimeline(e, n) {
    const s = this.animations.map((i) => {
      if (Ao() && i.attachTimeline)
        return i.attachTimeline(e);
      if (typeof n == "function")
        return n(i);
    });
    return () => {
      s.forEach((i, o) => {
        i && i(), this.animations[o].stop();
      });
    };
  }
  get time() {
    return this.getAll("time");
  }
  set time(e) {
    this.setAll("time", e);
  }
  get speed() {
    return this.getAll("speed");
  }
  set speed(e) {
    this.setAll("speed", e);
  }
  get startTime() {
    return this.getAll("startTime");
  }
  get duration() {
    let e = 0;
    for (let n = 0; n < this.animations.length; n++)
      e = Math.max(e, this.animations[n].duration);
    return e;
  }
  runAll(e) {
    this.animations.forEach((n) => n[e]());
  }
  flatten() {
    this.runAll("flatten");
  }
  play() {
    this.runAll("play");
  }
  pause() {
    this.runAll("pause");
  }
  cancel() {
    this.runAll("cancel");
  }
  complete() {
    this.runAll("complete");
  }
}
class Vo extends wo {
  then(e, n) {
    return Promise.all(this.animations).then(e).catch(n);
  }
}
const W = /* @__NO_SIDE_EFFECTS__ */ (t) => t * 1e3, z = /* @__NO_SIDE_EFFECTS__ */ (t) => t / 1e3;
function nn(t) {
  return typeof t == "function";
}
function Vn(t, e) {
  t.timeline = e, t.onfinish = null;
}
const sn = (t) => Array.isArray(t) && typeof t[0] == "number", Do = {
  linearEasing: void 0
};
function Mo(t, e) {
  const n = /* @__PURE__ */ en(t);
  return () => {
    var s;
    return (s = Do[e]) !== null && s !== void 0 ? s : n();
  };
}
const Gt = /* @__PURE__ */ Mo(() => {
  try {
    document.createElement("div").animate({ opacity: 0 }, { easing: "linear(0, 1)" });
  } catch {
    return !1;
  }
  return !0;
}, "linearEasing"), ft = /* @__NO_SIDE_EFFECTS__ */ (t, e, n) => {
  const s = e - t;
  return s === 0 ? 1 : (n - t) / s;
}, li = (t, e, n = 10) => {
  let s = "";
  const i = Math.max(Math.round(e / n), 2);
  for (let o = 0; o < i; o++)
    s += t(/* @__PURE__ */ ft(0, i - 1, o)) + ", ";
  return `linear(${s.substring(0, s.length - 2)})`;
};
function ui(t) {
  return !!(typeof t == "function" && Gt() || !t || typeof t == "string" && (t in ye || Gt()) || sn(t) || Array.isArray(t) && t.every(ui));
}
const Tt = ([t, e, n, s]) => `cubic-bezier(${t}, ${e}, ${n}, ${s})`, ye = {
  linear: "linear",
  ease: "ease",
  easeIn: "ease-in",
  easeOut: "ease-out",
  easeInOut: "ease-in-out",
  circIn: /* @__PURE__ */ Tt([0, 0.65, 0.55, 1]),
  circOut: /* @__PURE__ */ Tt([0.55, 0, 1, 0.45]),
  backIn: /* @__PURE__ */ Tt([0.31, 0.01, 0.66, -0.59]),
  backOut: /* @__PURE__ */ Tt([0.33, 1.53, 0.69, 0.99])
};
function ci(t, e) {
  if (t)
    return typeof t == "function" && Gt() ? li(t, e) : sn(t) ? Tt(t) : Array.isArray(t) ? t.map((n) => ci(n, e) || ye.easeOut) : ye[t];
}
const hi = (t, e, n) => (((1 - 3 * n + 3 * e) * t + (3 * n - 6 * e)) * t + 3 * e) * t, Co = 1e-7, Ro = 12;
function Eo(t, e, n, s, i) {
  let o, r, a = 0;
  do
    r = e + (n - e) / 2, o = hi(r, s, i) - t, o > 0 ? n = r : e = r;
  while (Math.abs(o) > Co && ++a < Ro);
  return r;
}
function Ft(t, e, n, s) {
  if (t === e && n === s)
    return I;
  const i = (o) => Eo(o, 0, 1, t, n);
  return (o) => o === 0 || o === 1 ? o : hi(i(o), e, s);
}
const fi = (t) => (e) => e <= 0.5 ? t(2 * e) / 2 : (2 - t(2 * (1 - e))) / 2, di = (t) => (e) => 1 - t(1 - e), mi = /* @__PURE__ */ Ft(0.33, 1.53, 0.69, 0.99), rn = /* @__PURE__ */ di(mi), pi = /* @__PURE__ */ fi(rn), gi = (t) => (t *= 2) < 1 ? 0.5 * rn(t) : 0.5 * (2 - Math.pow(2, -10 * (t - 1))), on = (t) => 1 - Math.sin(Math.acos(t)), yi = di(on), vi = fi(on), Ti = (t) => /^0[^.\s]+$/u.test(t);
function Lo(t) {
  return typeof t == "number" ? t === 0 : t !== null ? t === "none" || t === "0" || Ti(t) : !0;
}
const Pt = (t) => Math.round(t * 1e5) / 1e5, an = /-?(?:\d+(?:\.\d+)?|\.\d+)/gu;
function Bo(t) {
  return t == null;
}
const Fo = /^(?:#[\da-f]{3,8}|(?:rgb|hsl)a?\((?:-?[\d.]+%?[,\s]+){2}-?[\d.]+%?\s*(?:[,/]\s*)?(?:\b\d+(?:\.\d+)?|\.\d+)?%?\))$/iu, ln = (t, e) => (n) => !!(typeof n == "string" && Fo.test(n) && n.startsWith(t) || e && !Bo(n) && Object.prototype.hasOwnProperty.call(n, e)), xi = (t, e, n) => (s) => {
  if (typeof s != "string")
    return s;
  const [i, o, r, a] = s.match(an);
  return {
    [t]: parseFloat(i),
    [e]: parseFloat(o),
    [n]: parseFloat(r),
    alpha: a !== void 0 ? parseFloat(a) : 1
  };
}, ko = (t) => H(0, 255, t), ie = {
  ...yt,
  transform: (t) => Math.round(ko(t))
}, et = {
  test: /* @__PURE__ */ ln("rgb", "red"),
  parse: /* @__PURE__ */ xi("red", "green", "blue"),
  transform: ({ red: t, green: e, blue: n, alpha: s = 1 }) => "rgba(" + ie.transform(t) + ", " + ie.transform(e) + ", " + ie.transform(n) + ", " + Pt(Mt.transform(s)) + ")"
};
function Oo(t) {
  let e = "", n = "", s = "", i = "";
  return t.length > 5 ? (e = t.substring(1, 3), n = t.substring(3, 5), s = t.substring(5, 7), i = t.substring(7, 9)) : (e = t.substring(1, 2), n = t.substring(2, 3), s = t.substring(3, 4), i = t.substring(4, 5), e += e, n += n, s += s, i += i), {
    red: parseInt(e, 16),
    green: parseInt(n, 16),
    blue: parseInt(s, 16),
    alpha: i ? parseInt(i, 16) / 255 : 1
  };
}
const ve = {
  test: /* @__PURE__ */ ln("#"),
  parse: Oo,
  transform: et.transform
}, at = {
  test: /* @__PURE__ */ ln("hsl", "hue"),
  parse: /* @__PURE__ */ xi("hue", "saturation", "lightness"),
  transform: ({ hue: t, saturation: e, lightness: n, alpha: s = 1 }) => "hsla(" + Math.round(t) + ", " + K.transform(Pt(e)) + ", " + K.transform(Pt(n)) + ", " + Pt(Mt.transform(s)) + ")"
}, E = {
  test: (t) => et.test(t) || ve.test(t) || at.test(t),
  parse: (t) => et.test(t) ? et.parse(t) : at.test(t) ? at.parse(t) : ve.parse(t),
  transform: (t) => typeof t == "string" ? t : t.hasOwnProperty("red") ? et.transform(t) : at.transform(t)
}, Io = /(?:#[\da-f]{3,8}|(?:rgb|hsl)a?\((?:-?[\d.]+%?[,\s]+){2}-?[\d.]+%?\s*(?:[,/]\s*)?(?:\b\d+(?:\.\d+)?|\.\d+)?%?\))/giu;
function jo(t) {
  var e, n;
  return isNaN(t) && typeof t == "string" && (((e = t.match(an)) === null || e === void 0 ? void 0 : e.length) || 0) + (((n = t.match(Io)) === null || n === void 0 ? void 0 : n.length) || 0) > 0;
}
const Pi = "number", Si = "color", No = "var", Uo = "var(", Dn = "${}", _o = /var\s*\(\s*--(?:[\w-]+\s*|[\w-]+\s*,(?:\s*[^)(\s]|\s*\((?:[^)(]|\([^)(]*\))*\))+\s*)\)|#[\da-f]{3,8}|(?:rgb|hsl)a?\((?:-?[\d.]+%?[,\s]+){2}-?[\d.]+%?\s*(?:[,/]\s*)?(?:\b\d+(?:\.\d+)?|\.\d+)?%?\)|-?(?:\d+(?:\.\d+)?|\.\d+)/giu;
function Et(t) {
  const e = t.toString(), n = [], s = {
    color: [],
    number: [],
    var: []
  }, i = [];
  let o = 0;
  const a = e.replace(_o, (l) => (E.test(l) ? (s.color.push(o), i.push(Si), n.push(E.parse(l))) : l.startsWith(Uo) ? (s.var.push(o), i.push(No), n.push(l)) : (s.number.push(o), i.push(Pi), n.push(parseFloat(l))), ++o, Dn)).split(Dn);
  return { values: n, split: a, indexes: s, types: i };
}
function bi(t) {
  return Et(t).values;
}
function Ai(t) {
  const { split: e, types: n } = Et(t), s = e.length;
  return (i) => {
    let o = "";
    for (let r = 0; r < s; r++)
      if (o += e[r], i[r] !== void 0) {
        const a = n[r];
        a === Pi ? o += Pt(i[r]) : a === Si ? o += E.transform(i[r]) : o += i[r];
      }
    return o;
  };
}
const Ko = (t) => typeof t == "number" ? 0 : t;
function $o(t) {
  const e = bi(t);
  return Ai(t)(e.map(Ko));
}
const Z = {
  test: jo,
  parse: bi,
  createTransformer: Ai,
  getAnimatableNone: $o
}, Wo = /* @__PURE__ */ new Set(["brightness", "contrast", "saturate", "opacity"]);
function Go(t) {
  const [e, n] = t.slice(0, -1).split("(");
  if (e === "drop-shadow")
    return t;
  const [s] = n.match(an) || [];
  if (!s)
    return t;
  const i = n.replace(s, "");
  let o = Wo.has(e) ? 1 : 0;
  return s !== n && (o *= 100), e + "(" + o + i + ")";
}
const zo = /\b([a-z-]*)\(.*?\)/gu, Te = {
  ...Z,
  getAnimatableNone: (t) => {
    const e = t.match(zo);
    return e ? e.map(Go).join(" ") : t;
  }
}, Ho = {
  ...$e,
  // Color props
  color: E,
  backgroundColor: E,
  outlineColor: E,
  fill: E,
  stroke: E,
  // Border props
  borderColor: E,
  borderTopColor: E,
  borderRightColor: E,
  borderBottomColor: E,
  borderLeftColor: E,
  filter: Te,
  WebkitFilter: Te
}, wi = (t) => Ho[t];
function Vi(t, e) {
  let n = wi(t);
  return n !== Te && (n = Z), n.getAnimatableNone ? n.getAnimatableNone(e) : void 0;
}
const Yo = /* @__PURE__ */ new Set(["auto", "none", "0"]);
function Xo(t, e, n) {
  let s = 0, i;
  for (; s < t.length && !i; ) {
    const o = t[s];
    typeof o == "string" && !Yo.has(o) && Et(o).values.length && (i = t[s]), s++;
  }
  if (i && n)
    for (const o of e)
      t[o] = Vi(n, i);
}
const nt = (t) => t * 180 / Math.PI, xe = (t) => {
  const e = nt(Math.atan2(t[1], t[0]));
  return Pe(e);
}, qo = {
  x: 4,
  y: 5,
  translateX: 4,
  translateY: 5,
  scaleX: 0,
  scaleY: 3,
  scale: (t) => (Math.abs(t[0]) + Math.abs(t[3])) / 2,
  rotate: xe,
  rotateZ: xe,
  skewX: (t) => nt(Math.atan(t[1])),
  skewY: (t) => nt(Math.atan(t[2])),
  skew: (t) => (Math.abs(t[1]) + Math.abs(t[2])) / 2
}, Pe = (t) => (t = t % 360, t < 0 && (t += 360), t), Mn = xe, Cn = (t) => Math.sqrt(t[0] * t[0] + t[1] * t[1]), Rn = (t) => Math.sqrt(t[4] * t[4] + t[5] * t[5]), Zo = {
  x: 12,
  y: 13,
  z: 14,
  translateX: 12,
  translateY: 13,
  translateZ: 14,
  scaleX: Cn,
  scaleY: Rn,
  scale: (t) => (Cn(t) + Rn(t)) / 2,
  rotateX: (t) => Pe(nt(Math.atan2(t[6], t[5]))),
  rotateY: (t) => Pe(nt(Math.atan2(-t[2], t[0]))),
  rotateZ: Mn,
  rotate: Mn,
  skewX: (t) => nt(Math.atan(t[4])),
  skewY: (t) => nt(Math.atan(t[1])),
  skew: (t) => (Math.abs(t[1]) + Math.abs(t[4])) / 2
};
function En(t) {
  return t.includes("scale") ? 1 : 0;
}
function Se(t, e) {
  if (!t || t === "none")
    return En(e);
  const n = t.match(/^matrix3d\(([-\d.e\s,]+)\)$/u);
  let s, i;
  if (n)
    s = Zo, i = n;
  else {
    const a = t.match(/^matrix\(([-\d.e\s,]+)\)$/u);
    s = qo, i = a;
  }
  if (!i)
    return En(e);
  const o = s[e], r = i[1].split(",").map(Qo);
  return typeof o == "function" ? o(r) : r[o];
}
const Jo = (t, e) => {
  const { transform: n = "none" } = getComputedStyle(t);
  return Se(n, e);
};
function Qo(t) {
  return parseFloat(t.trim());
}
const Ln = (t) => t === yt || t === x, ta = /* @__PURE__ */ new Set(["x", "y", "z"]), ea = gt.filter((t) => !ta.has(t));
function na(t) {
  const e = [];
  return ea.forEach((n) => {
    const s = t.getValue(n);
    s !== void 0 && (e.push([n, s.get()]), s.set(n.startsWith("scale") ? 1 : 0));
  }), e;
}
const dt = {
  // Dimensions
  width: ({ x: t }, { paddingLeft: e = "0", paddingRight: n = "0" }) => t.max - t.min - parseFloat(e) - parseFloat(n),
  height: ({ y: t }, { paddingTop: e = "0", paddingBottom: n = "0" }) => t.max - t.min - parseFloat(e) - parseFloat(n),
  top: (t, { top: e }) => parseFloat(e),
  left: (t, { left: e }) => parseFloat(e),
  bottom: ({ y: t }, { top: e }) => parseFloat(e) + (t.max - t.min),
  right: ({ x: t }, { left: e }) => parseFloat(e) + (t.max - t.min),
  // Transform
  x: (t, { transform: e }) => Se(e, "x"),
  y: (t, { transform: e }) => Se(e, "y")
};
dt.translateX = dt.x;
dt.translateY = dt.y;
const st = /* @__PURE__ */ new Set();
let be = !1, Ae = !1;
function Di() {
  if (Ae) {
    const t = Array.from(st).filter((s) => s.needsMeasurement), e = new Set(t.map((s) => s.element)), n = /* @__PURE__ */ new Map();
    e.forEach((s) => {
      const i = na(s);
      i.length && (n.set(s, i), s.render());
    }), t.forEach((s) => s.measureInitialState()), e.forEach((s) => {
      s.render();
      const i = n.get(s);
      i && i.forEach(([o, r]) => {
        var a;
        (a = s.getValue(o)) === null || a === void 0 || a.set(r);
      });
    }), t.forEach((s) => s.measureEndState()), t.forEach((s) => {
      s.suspendedScrollY !== void 0 && window.scrollTo(0, s.suspendedScrollY);
    });
  }
  Ae = !1, be = !1, st.forEach((t) => t.complete()), st.clear();
}
function Mi() {
  st.forEach((t) => {
    t.readKeyframes(), t.needsMeasurement && (Ae = !0);
  });
}
function sa() {
  Mi(), Di();
}
class un {
  constructor(e, n, s, i, o, r = !1) {
    this.isComplete = !1, this.isAsync = !1, this.needsMeasurement = !1, this.isScheduled = !1, this.unresolvedKeyframes = [...e], this.onComplete = n, this.name = s, this.motionValue = i, this.element = o, this.isAsync = r;
  }
  scheduleResolve() {
    this.isScheduled = !0, this.isAsync ? (st.add(this), be || (be = !0, w.read(Mi), w.resolveKeyframes(Di))) : (this.readKeyframes(), this.complete());
  }
  readKeyframes() {
    const { unresolvedKeyframes: e, name: n, element: s, motionValue: i } = this;
    for (let o = 0; o < e.length; o++)
      if (e[o] === null)
        if (o === 0) {
          const r = i == null ? void 0 : i.get(), a = e[e.length - 1];
          if (r !== void 0)
            e[0] = r;
          else if (s && n) {
            const l = s.readValue(n, a);
            l != null && (e[0] = l);
          }
          e[0] === void 0 && (e[0] = a), i && r === void 0 && i.set(e[0]);
        } else
          e[o] = e[o - 1];
  }
  setFinalKeyframe() {
  }
  measureInitialState() {
  }
  renderEndStyles() {
  }
  measureEndState() {
  }
  complete() {
    this.isComplete = !0, this.onComplete(this.unresolvedKeyframes, this.finalKeyframe), st.delete(this);
  }
  cancel() {
    this.isComplete || (this.isScheduled = !1, st.delete(this));
  }
  resume() {
    this.isComplete || this.scheduleResolve();
  }
}
const Ci = (t) => /^-?(?:\d+(?:\.\d+)?|\.\d+)$/u.test(t), ia = (
  // eslint-disable-next-line redos-detector/no-unsafe-regex -- false positive, as it can match a lot of words
  /^var\(--(?:([\w-]+)|([\w-]+), ?([a-zA-Z\d ()%#.,-]+))\)/u
);
function ra(t) {
  const e = ia.exec(t);
  if (!e)
    return [,];
  const [, n, s, i] = e;
  return [`--${n ?? s}`, i];
}
const oa = 4;
function Ri(t, e, n = 1) {
  X(n <= oa, `Max CSS variable fallback depth detected in property "${t}". This may indicate a circular fallback dependency.`);
  const [s, i] = ra(t);
  if (!s)
    return;
  const o = window.getComputedStyle(e).getPropertyValue(s);
  if (o) {
    const r = o.trim();
    return Ci(r) ? parseFloat(r) : r;
  }
  return Ke(i) ? Ri(i, e, n + 1) : i;
}
const Ei = (t) => (e) => e.test(t), aa = {
  test: (t) => t === "auto",
  parse: (t) => t
}, Li = [yt, x, K, Y, Yr, Hr, aa], Bn = (t) => Li.find(Ei(t));
class Bi extends un {
  constructor(e, n, s, i, o) {
    super(e, n, s, i, o, !0);
  }
  readKeyframes() {
    const { unresolvedKeyframes: e, element: n, name: s } = this;
    if (!n || !n.current)
      return;
    super.readKeyframes();
    for (let l = 0; l < e.length; l++) {
      let u = e[l];
      if (typeof u == "string" && (u = u.trim(), Ke(u))) {
        const c = Ri(u, n.current);
        c !== void 0 && (e[l] = c), l === e.length - 1 && (this.finalKeyframe = u);
      }
    }
    if (this.resolveNoneKeyframes(), !ri.has(s) || e.length !== 2)
      return;
    const [i, o] = e, r = Bn(i), a = Bn(o);
    if (r !== a)
      if (Ln(r) && Ln(a))
        for (let l = 0; l < e.length; l++) {
          const u = e[l];
          typeof u == "string" && (e[l] = parseFloat(u));
        }
      else
        this.needsMeasurement = !0;
  }
  resolveNoneKeyframes() {
    const { unresolvedKeyframes: e, name: n } = this, s = [];
    for (let i = 0; i < e.length; i++)
      Lo(e[i]) && s.push(i);
    s.length && Xo(e, s, n);
  }
  measureInitialState() {
    const { element: e, unresolvedKeyframes: n, name: s } = this;
    if (!e || !e.current)
      return;
    s === "height" && (this.suspendedScrollY = window.pageYOffset), this.measuredOrigin = dt[s](e.measureViewportBox(), window.getComputedStyle(e.current)), n[0] = this.measuredOrigin;
    const i = n[n.length - 1];
    i !== void 0 && e.getValue(s, i).jump(i, !1);
  }
  measureEndState() {
    var e;
    const { element: n, name: s, unresolvedKeyframes: i } = this;
    if (!n || !n.current)
      return;
    const o = n.getValue(s);
    o && o.jump(this.measuredOrigin, !1);
    const r = i.length - 1, a = i[r];
    i[r] = dt[s](n.measureViewportBox(), window.getComputedStyle(n.current)), a !== null && this.finalKeyframe === void 0 && (this.finalKeyframe = a), !((e = this.removedTransforms) === null || e === void 0) && e.length && this.removedTransforms.forEach(([l, u]) => {
      n.getValue(l).set(u);
    }), this.resolveNoneKeyframes();
  }
}
const Fn = (t, e) => e === "zIndex" ? !1 : !!(typeof t == "number" || Array.isArray(t) || typeof t == "string" && // It's animatable if we have a string
(Z.test(t) || t === "0") && // And it contains numbers and/or colors
!t.startsWith("url("));
function la(t) {
  const e = t[0];
  if (t.length === 1)
    return !0;
  for (let n = 0; n < t.length; n++)
    if (t[n] !== e)
      return !0;
}
function ua(t, e, n, s) {
  const i = t[0];
  if (i === null)
    return !1;
  if (e === "display" || e === "visibility")
    return !0;
  const o = t[t.length - 1], r = Fn(i, e), a = Fn(o, e);
  return pt(r === a, `You are trying to animate ${e} from "${i}" to "${o}". ${i} is not an animatable value - to enable this animation set ${i} to a value animatable to ${o} via the \`style\` property.`), !r || !a ? !1 : la(t) || (n === "spring" || nn(n)) && s;
}
const ca = (t) => t !== null;
function te(t, { repeat: e, repeatType: n = "loop" }, s) {
  const i = t.filter(ca), o = e && n !== "loop" && e % 2 === 1 ? 0 : i.length - 1;
  return !o || s === void 0 ? i[o] : s;
}
const ha = 40;
class Fi {
  constructor({ autoplay: e = !0, delay: n = 0, type: s = "keyframes", repeat: i = 0, repeatDelay: o = 0, repeatType: r = "loop", ...a }) {
    this.isStopped = !1, this.hasAttemptedResolve = !1, this.createdAt = $.now(), this.options = {
      autoplay: e,
      delay: n,
      type: s,
      repeat: i,
      repeatDelay: o,
      repeatType: r,
      ...a
    }, this.updateFinishedPromise();
  }
  /**
   * This method uses the createdAt and resolvedAt to calculate the
   * animation startTime. *Ideally*, we would use the createdAt time as t=0
   * as the following frame would then be the first frame of the animation in
   * progress, which would feel snappier.
   *
   * However, if there's a delay (main thread work) between the creation of
   * the animation and the first commited frame, we prefer to use resolvedAt
   * to avoid a sudden jump into the animation.
   */
  calcStartTime() {
    return this.resolvedAt ? this.resolvedAt - this.createdAt > ha ? this.resolvedAt : this.createdAt : this.createdAt;
  }
  /**
   * A getter for resolved data. If keyframes are not yet resolved, accessing
   * this.resolved will synchronously flush all pending keyframe resolvers.
   * This is a deoptimisation, but at its worst still batches read/writes.
   */
  get resolved() {
    return !this._resolved && !this.hasAttemptedResolve && sa(), this._resolved;
  }
  /**
   * A method to be called when the keyframes resolver completes. This method
   * will check if its possible to run the animation and, if not, skip it.
   * Otherwise, it will call initPlayback on the implementing class.
   */
  onKeyframesResolved(e, n) {
    this.resolvedAt = $.now(), this.hasAttemptedResolve = !0;
    const { name: s, type: i, velocity: o, delay: r, onComplete: a, onUpdate: l, isGenerator: u } = this.options;
    if (!u && !ua(e, s, i, o))
      if (r)
        this.options.duration = 0;
      else {
        l && l(te(e, this.options, n)), a && a(), this.resolveFinishedPromise();
        return;
      }
    const c = this.initPlayback(e, n);
    c !== !1 && (this._resolved = {
      keyframes: e,
      finalKeyframe: n,
      ...c
    }, this.onPostResolved());
  }
  onPostResolved() {
  }
  /**
   * Allows the returned animation to be awaited or promise-chained. Currently
   * resolves when the animation finishes at all but in a future update could/should
   * reject if its cancels.
   */
  then(e, n) {
    return this.currentFinishedPromise.then(e, n);
  }
  flatten() {
    this.options.allowFlatten && (this.options.type = "keyframes", this.options.ease = "linear");
  }
  updateFinishedPromise() {
    this.currentFinishedPromise = new Promise((e) => {
      this.resolveFinishedPromise = e;
    });
  }
}
const we = 2e4;
function ki(t) {
  let e = 0;
  const n = 50;
  let s = t.next(e);
  for (; !s.done && e < we; )
    e += n, s = t.next(e);
  return e >= we ? 1 / 0 : e;
}
const D = (t, e, n) => t + (e - t) * n;
function re(t, e, n) {
  return n < 0 && (n += 1), n > 1 && (n -= 1), n < 1 / 6 ? t + (e - t) * 6 * n : n < 1 / 2 ? e : n < 2 / 3 ? t + (e - t) * (2 / 3 - n) * 6 : t;
}
function fa({ hue: t, saturation: e, lightness: n, alpha: s }) {
  t /= 360, e /= 100, n /= 100;
  let i = 0, o = 0, r = 0;
  if (!e)
    i = o = r = n;
  else {
    const a = n < 0.5 ? n * (1 + e) : n + e - n * e, l = 2 * n - a;
    i = re(l, a, t + 1 / 3), o = re(l, a, t), r = re(l, a, t - 1 / 3);
  }
  return {
    red: Math.round(i * 255),
    green: Math.round(o * 255),
    blue: Math.round(r * 255),
    alpha: s
  };
}
function zt(t, e) {
  return (n) => n > 0 ? e : t;
}
const oe = (t, e, n) => {
  const s = t * t, i = n * (e * e - s) + s;
  return i < 0 ? 0 : Math.sqrt(i);
}, da = [ve, et, at], ma = (t) => da.find((e) => e.test(t));
function kn(t) {
  const e = ma(t);
  if (pt(!!e, `'${t}' is not an animatable color. Use the equivalent color code instead.`), !e)
    return !1;
  let n = e.parse(t);
  return e === at && (n = fa(n)), n;
}
const On = (t, e) => {
  const n = kn(t), s = kn(e);
  if (!n || !s)
    return zt(t, e);
  const i = { ...n };
  return (o) => (i.red = oe(n.red, s.red, o), i.green = oe(n.green, s.green, o), i.blue = oe(n.blue, s.blue, o), i.alpha = D(n.alpha, s.alpha, o), et.transform(i));
}, pa = (t, e) => (n) => e(t(n)), kt = (...t) => t.reduce(pa), Ve = /* @__PURE__ */ new Set(["none", "hidden"]);
function ga(t, e) {
  return Ve.has(t) ? (n) => n <= 0 ? t : e : (n) => n >= 1 ? e : t;
}
function ya(t, e) {
  return (n) => D(t, e, n);
}
function cn(t) {
  return typeof t == "number" ? ya : typeof t == "string" ? Ke(t) ? zt : E.test(t) ? On : xa : Array.isArray(t) ? Oi : typeof t == "object" ? E.test(t) ? On : va : zt;
}
function Oi(t, e) {
  const n = [...t], s = n.length, i = t.map((o, r) => cn(o)(o, e[r]));
  return (o) => {
    for (let r = 0; r < s; r++)
      n[r] = i[r](o);
    return n;
  };
}
function va(t, e) {
  const n = { ...t, ...e }, s = {};
  for (const i in n)
    t[i] !== void 0 && e[i] !== void 0 && (s[i] = cn(t[i])(t[i], e[i]));
  return (i) => {
    for (const o in s)
      n[o] = s[o](i);
    return n;
  };
}
function Ta(t, e) {
  var n;
  const s = [], i = { color: 0, var: 0, number: 0 };
  for (let o = 0; o < e.values.length; o++) {
    const r = e.types[o], a = t.indexes[r][i[r]], l = (n = t.values[a]) !== null && n !== void 0 ? n : 0;
    s[o] = l, i[r]++;
  }
  return s;
}
const xa = (t, e) => {
  const n = Z.createTransformer(e), s = Et(t), i = Et(e);
  return s.indexes.var.length === i.indexes.var.length && s.indexes.color.length === i.indexes.color.length && s.indexes.number.length >= i.indexes.number.length ? Ve.has(t) && !i.values.length || Ve.has(e) && !s.values.length ? ga(t, e) : kt(Oi(Ta(s, i), i.values), n) : (pt(!0, `Complex values '${t}' and '${e}' too different to mix. Ensure all colors are of the same type, and that each contains the same quantity of number and color values. Falling back to instant transition.`), zt(t, e));
};
function Ii(t, e, n) {
  return typeof t == "number" && typeof e == "number" && typeof n == "number" ? D(t, e, n) : cn(t)(t, e);
}
const Pa = 5;
function ji(t, e, n) {
  const s = Math.max(e - Pa, 0);
  return oi(n - t(s), e - s);
}
const V = {
  // Default spring physics
  stiffness: 100,
  damping: 10,
  mass: 1,
  velocity: 0,
  // Default duration/bounce-based options
  duration: 800,
  // in ms
  bounce: 0.3,
  visualDuration: 0.3,
  // in seconds
  // Rest thresholds
  restSpeed: {
    granular: 0.01,
    default: 2
  },
  restDelta: {
    granular: 5e-3,
    default: 0.5
  },
  // Limits
  minDuration: 0.01,
  // in seconds
  maxDuration: 10,
  // in seconds
  minDamping: 0.05,
  maxDamping: 1
}, In = 1e-3;
function Sa({ duration: t = V.duration, bounce: e = V.bounce, velocity: n = V.velocity, mass: s = V.mass }) {
  let i, o;
  pt(t <= /* @__PURE__ */ W(V.maxDuration), "Spring duration must be 10 seconds or less");
  let r = 1 - e;
  r = H(V.minDamping, V.maxDamping, r), t = H(V.minDuration, V.maxDuration, /* @__PURE__ */ z(t)), r < 1 ? (i = (u) => {
    const c = u * r, h = c * t, f = c - n, d = De(u, r), m = Math.exp(-h);
    return In - f / d * m;
  }, o = (u) => {
    const h = u * r * t, f = h * n + n, d = Math.pow(r, 2) * Math.pow(u, 2) * t, m = Math.exp(-h), p = De(Math.pow(u, 2), r);
    return (-i(u) + In > 0 ? -1 : 1) * ((f - d) * m) / p;
  }) : (i = (u) => {
    const c = Math.exp(-u * t), h = (u - n) * t + 1;
    return -1e-3 + c * h;
  }, o = (u) => {
    const c = Math.exp(-u * t), h = (n - u) * (t * t);
    return c * h;
  });
  const a = 5 / t, l = Aa(i, o, a);
  if (t = /* @__PURE__ */ W(t), isNaN(l))
    return {
      stiffness: V.stiffness,
      damping: V.damping,
      duration: t
    };
  {
    const u = Math.pow(l, 2) * s;
    return {
      stiffness: u,
      damping: r * 2 * Math.sqrt(s * u),
      duration: t
    };
  }
}
const ba = 12;
function Aa(t, e, n) {
  let s = n;
  for (let i = 1; i < ba; i++)
    s = s - t(s) / e(s);
  return s;
}
function De(t, e) {
  return t * Math.sqrt(1 - e * e);
}
const wa = ["duration", "bounce"], Va = ["stiffness", "damping", "mass"];
function jn(t, e) {
  return e.some((n) => t[n] !== void 0);
}
function Da(t) {
  let e = {
    velocity: V.velocity,
    stiffness: V.stiffness,
    damping: V.damping,
    mass: V.mass,
    isResolvedFromDuration: !1,
    ...t
  };
  if (!jn(t, Va) && jn(t, wa))
    if (t.visualDuration) {
      const n = t.visualDuration, s = 2 * Math.PI / (n * 1.2), i = s * s, o = 2 * H(0.05, 1, 1 - (t.bounce || 0)) * Math.sqrt(i);
      e = {
        ...e,
        mass: V.mass,
        stiffness: i,
        damping: o
      };
    } else {
      const n = Sa(t);
      e = {
        ...e,
        ...n,
        mass: V.mass
      }, e.isResolvedFromDuration = !0;
    }
  return e;
}
function Ni(t = V.visualDuration, e = V.bounce) {
  const n = typeof t != "object" ? {
    visualDuration: t,
    keyframes: [0, 1],
    bounce: e
  } : t;
  let { restSpeed: s, restDelta: i } = n;
  const o = n.keyframes[0], r = n.keyframes[n.keyframes.length - 1], a = { done: !1, value: o }, { stiffness: l, damping: u, mass: c, duration: h, velocity: f, isResolvedFromDuration: d } = Da({
    ...n,
    velocity: -/* @__PURE__ */ z(n.velocity || 0)
  }), m = f || 0, p = u / (2 * Math.sqrt(l * c)), y = r - o, g = /* @__PURE__ */ z(Math.sqrt(l / c)), v = Math.abs(y) < 5;
  s || (s = v ? V.restSpeed.granular : V.restSpeed.default), i || (i = v ? V.restDelta.granular : V.restDelta.default);
  let T;
  if (p < 1) {
    const P = De(g, p);
    T = (A) => {
      const C = Math.exp(-p * g * A);
      return r - C * ((m + p * g * y) / P * Math.sin(P * A) + y * Math.cos(P * A));
    };
  } else if (p === 1)
    T = (P) => r - Math.exp(-g * P) * (y + (m + g * y) * P);
  else {
    const P = g * Math.sqrt(p * p - 1);
    T = (A) => {
      const C = Math.exp(-p * g * A), S = Math.min(P * A, 300);
      return r - C * ((m + p * g * y) * Math.sinh(S) + P * y * Math.cosh(S)) / P;
    };
  }
  const b = {
    calculatedDuration: d && h || null,
    next: (P) => {
      const A = T(P);
      if (d)
        a.done = P >= h;
      else {
        let C = 0;
        p < 1 && (C = P === 0 ? /* @__PURE__ */ W(m) : ji(T, P, A));
        const S = Math.abs(C) <= s, k = Math.abs(r - A) <= i;
        a.done = S && k;
      }
      return a.value = a.done ? r : A, a;
    },
    toString: () => {
      const P = Math.min(ki(b), we), A = li((C) => b.next(P * C).value, P, 30);
      return P + "ms " + A;
    }
  };
  return b;
}
function Nn({ keyframes: t, velocity: e = 0, power: n = 0.8, timeConstant: s = 325, bounceDamping: i = 10, bounceStiffness: o = 500, modifyTarget: r, min: a, max: l, restDelta: u = 0.5, restSpeed: c }) {
  const h = t[0], f = {
    done: !1,
    value: h
  }, d = (S) => a !== void 0 && S < a || l !== void 0 && S > l, m = (S) => a === void 0 ? l : l === void 0 || Math.abs(a - S) < Math.abs(l - S) ? a : l;
  let p = n * e;
  const y = h + p, g = r === void 0 ? y : r(y);
  g !== y && (p = g - h);
  const v = (S) => -p * Math.exp(-S / s), T = (S) => g + v(S), b = (S) => {
    const k = v(S), j = T(S);
    f.done = Math.abs(k) <= u, f.value = f.done ? g : j;
  };
  let P, A;
  const C = (S) => {
    d(f.value) && (P = S, A = Ni({
      keyframes: [f.value, m(f.value)],
      velocity: ji(T, S, f.value),
      // TODO: This should be passing * 1000
      damping: i,
      stiffness: o,
      restDelta: u,
      restSpeed: c
    }));
  };
  return C(0), {
    calculatedDuration: null,
    next: (S) => {
      let k = !1;
      return !A && P === void 0 && (k = !0, b(S), C(S)), P !== void 0 && S >= P ? A.next(S - P) : (!k && b(S), f);
    }
  };
}
const Ma = /* @__PURE__ */ Ft(0.42, 0, 1, 1), Ca = /* @__PURE__ */ Ft(0, 0, 0.58, 1), Ui = /* @__PURE__ */ Ft(0.42, 0, 0.58, 1), Ra = (t) => Array.isArray(t) && typeof t[0] != "number", Un = {
  linear: I,
  easeIn: Ma,
  easeInOut: Ui,
  easeOut: Ca,
  circIn: on,
  circInOut: vi,
  circOut: yi,
  backIn: rn,
  backInOut: pi,
  backOut: mi,
  anticipate: gi
}, _n = (t) => {
  if (sn(t)) {
    X(t.length === 4, "Cubic bezier arrays must contain four numerical values.");
    const [e, n, s, i] = t;
    return Ft(e, n, s, i);
  } else if (typeof t == "string")
    return X(Un[t] !== void 0, `Invalid easing type '${t}'`), Un[t];
  return t;
};
function Ea(t, e, n) {
  const s = [], i = n || Ii, o = t.length - 1;
  for (let r = 0; r < o; r++) {
    let a = i(t[r], t[r + 1]);
    if (e) {
      const l = Array.isArray(e) ? e[r] || I : e;
      a = kt(l, a);
    }
    s.push(a);
  }
  return s;
}
function La(t, e, { clamp: n = !0, ease: s, mixer: i } = {}) {
  const o = t.length;
  if (X(o === e.length, "Both input and output ranges must be the same length"), o === 1)
    return () => e[0];
  if (o === 2 && e[0] === e[1])
    return () => e[1];
  const r = t[0] === t[1];
  t[0] > t[o - 1] && (t = [...t].reverse(), e = [...e].reverse());
  const a = Ea(e, s, i), l = a.length, u = (c) => {
    if (r && c < t[0])
      return e[0];
    let h = 0;
    if (l > 1)
      for (; h < t.length - 2 && !(c < t[h + 1]); h++)
        ;
    const f = /* @__PURE__ */ ft(t[h], t[h + 1], c);
    return a[h](f);
  };
  return n ? (c) => u(H(t[0], t[o - 1], c)) : u;
}
function Ba(t, e) {
  const n = t[t.length - 1];
  for (let s = 1; s <= e; s++) {
    const i = /* @__PURE__ */ ft(0, e, s);
    t.push(D(n, 1, i));
  }
}
function Fa(t) {
  const e = [0];
  return Ba(e, t.length - 1), e;
}
function ka(t, e) {
  return t.map((n) => n * e);
}
function Oa(t, e) {
  return t.map(() => e || Ui).splice(0, t.length - 1);
}
function St({ duration: t = 300, keyframes: e, times: n, ease: s = "easeInOut" }) {
  const i = Ra(s) ? s.map(_n) : _n(s), o = {
    done: !1,
    value: e[0]
  }, r = ka(
    // Only use the provided offsets if they're the correct length
    // TODO Maybe we should warn here if there's a length mismatch
    n && n.length === e.length ? n : Fa(e),
    t
  ), a = La(r, e, {
    ease: Array.isArray(i) ? i : Oa(e, i)
  });
  return {
    calculatedDuration: t,
    next: (l) => (o.value = a(l), o.done = l >= t, o)
  };
}
const Ia = (t) => {
  const e = ({ timestamp: n }) => t(n);
  return {
    start: () => w.update(e, !0),
    stop: () => q(e),
    /**
     * If we're processing this frame we can use the
     * framelocked timestamp to keep things in sync.
     */
    now: () => R.isProcessing ? R.timestamp : $.now()
  };
}, ja = {
  decay: Nn,
  inertia: Nn,
  tween: St,
  keyframes: St,
  spring: Ni
}, Na = (t) => t / 100;
class hn extends Fi {
  constructor(e) {
    super(e), this.holdTime = null, this.cancelTime = null, this.currentTime = 0, this.playbackSpeed = 1, this.pendingPlayState = "running", this.startTime = null, this.state = "idle", this.stop = () => {
      if (this.resolver.cancel(), this.isStopped = !0, this.state === "idle")
        return;
      this.teardown();
      const { onStop: l } = this.options;
      l && l();
    };
    const { name: n, motionValue: s, element: i, keyframes: o } = this.options, r = (i == null ? void 0 : i.KeyframeResolver) || un, a = (l, u) => this.onKeyframesResolved(l, u);
    this.resolver = new r(o, a, n, s, i), this.resolver.scheduleResolve();
  }
  flatten() {
    super.flatten(), this._resolved && Object.assign(this._resolved, this.initPlayback(this._resolved.keyframes));
  }
  initPlayback(e) {
    const { type: n = "keyframes", repeat: s = 0, repeatDelay: i = 0, repeatType: o, velocity: r = 0 } = this.options, a = nn(n) ? n : ja[n] || St;
    let l, u;
    process.env.NODE_ENV !== "production" && a !== St && X(e.length <= 2, `Only two keyframes currently supported with spring and inertia animations. Trying to animate ${e}`), a !== St && typeof e[0] != "number" && (l = kt(Na, Ii(e[0], e[1])), e = [0, 100]);
    const c = a({ ...this.options, keyframes: e });
    o === "mirror" && (u = a({
      ...this.options,
      keyframes: [...e].reverse(),
      velocity: -r
    })), c.calculatedDuration === null && (c.calculatedDuration = ki(c));
    const { calculatedDuration: h } = c, f = h + i, d = f * (s + 1) - i;
    return {
      generator: c,
      mirroredGenerator: u,
      mapPercentToKeyframes: l,
      calculatedDuration: h,
      resolvedDuration: f,
      totalDuration: d
    };
  }
  onPostResolved() {
    const { autoplay: e = !0 } = this.options;
    this.play(), this.pendingPlayState === "paused" || !e ? this.pause() : this.state = this.pendingPlayState;
  }
  tick(e, n = !1) {
    const { resolved: s } = this;
    if (!s) {
      const { keyframes: S } = this.options;
      return { done: !0, value: S[S.length - 1] };
    }
    const { finalKeyframe: i, generator: o, mirroredGenerator: r, mapPercentToKeyframes: a, keyframes: l, calculatedDuration: u, totalDuration: c, resolvedDuration: h } = s;
    if (this.startTime === null)
      return o.next(0);
    const { delay: f, repeat: d, repeatType: m, repeatDelay: p, onUpdate: y } = this.options;
    this.speed > 0 ? this.startTime = Math.min(this.startTime, e) : this.speed < 0 && (this.startTime = Math.min(e - c / this.speed, this.startTime)), n ? this.currentTime = e : this.holdTime !== null ? this.currentTime = this.holdTime : this.currentTime = Math.round(e - this.startTime) * this.speed;
    const g = this.currentTime - f * (this.speed >= 0 ? 1 : -1), v = this.speed >= 0 ? g < 0 : g > c;
    this.currentTime = Math.max(g, 0), this.state === "finished" && this.holdTime === null && (this.currentTime = c);
    let T = this.currentTime, b = o;
    if (d) {
      const S = Math.min(this.currentTime, c) / h;
      let k = Math.floor(S), j = S % 1;
      !j && S >= 1 && (j = 1), j === 1 && k--, k = Math.min(k, d + 1), !!(k % 2) && (m === "reverse" ? (j = 1 - j, p && (j -= p / h)) : m === "mirror" && (b = r)), T = H(0, 1, j) * h;
    }
    const P = v ? { done: !1, value: l[0] } : b.next(T);
    a && (P.value = a(P.value));
    let { done: A } = P;
    !v && u !== null && (A = this.speed >= 0 ? this.currentTime >= c : this.currentTime <= 0);
    const C = this.holdTime === null && (this.state === "finished" || this.state === "running" && A);
    return C && i !== void 0 && (P.value = te(l, this.options, i)), y && y(P.value), C && this.finish(), P;
  }
  get duration() {
    const { resolved: e } = this;
    return e ? /* @__PURE__ */ z(e.calculatedDuration) : 0;
  }
  get time() {
    return /* @__PURE__ */ z(this.currentTime);
  }
  set time(e) {
    e = /* @__PURE__ */ W(e), this.currentTime = e, this.holdTime !== null || this.speed === 0 ? this.holdTime = e : this.driver && (this.startTime = this.driver.now() - e / this.speed);
  }
  get speed() {
    return this.playbackSpeed;
  }
  set speed(e) {
    const n = this.playbackSpeed !== e;
    this.playbackSpeed = e, n && (this.time = /* @__PURE__ */ z(this.currentTime));
  }
  play() {
    if (this.resolver.isScheduled || this.resolver.resume(), !this._resolved) {
      this.pendingPlayState = "running";
      return;
    }
    if (this.isStopped)
      return;
    const { driver: e = Ia, onPlay: n, startTime: s } = this.options;
    this.driver || (this.driver = e((o) => this.tick(o))), n && n();
    const i = this.driver.now();
    this.holdTime !== null ? this.startTime = i - this.holdTime : this.startTime ? this.state === "finished" && (this.startTime = i) : this.startTime = s ?? this.calcStartTime(), this.state === "finished" && this.updateFinishedPromise(), this.cancelTime = this.startTime, this.holdTime = null, this.state = "running", this.driver.start();
  }
  pause() {
    var e;
    if (!this._resolved) {
      this.pendingPlayState = "paused";
      return;
    }
    this.state = "paused", this.holdTime = (e = this.currentTime) !== null && e !== void 0 ? e : 0;
  }
  complete() {
    this.state !== "running" && this.play(), this.pendingPlayState = this.state = "finished", this.holdTime = null;
  }
  finish() {
    this.teardown(), this.state = "finished";
    const { onComplete: e } = this.options;
    e && e();
  }
  cancel() {
    this.cancelTime !== null && this.tick(this.cancelTime), this.teardown(), this.updateFinishedPromise();
  }
  teardown() {
    this.state = "idle", this.stopDriver(), this.resolveFinishedPromise(), this.updateFinishedPromise(), this.startTime = this.cancelTime = null, this.resolver.cancel();
  }
  stopDriver() {
    this.driver && (this.driver.stop(), this.driver = void 0);
  }
  sample(e) {
    return this.startTime = 0, this.tick(e, !0);
  }
}
const Ua = /* @__PURE__ */ new Set([
  "opacity",
  "clipPath",
  "filter",
  "transform"
  // TODO: Can be accelerated but currently disabled until https://issues.chromium.org/issues/41491098 is resolved
  // or until we implement support for linear() easing.
  // "background-color"
]);
function _a(t, e, n, { delay: s = 0, duration: i = 300, repeat: o = 0, repeatType: r = "loop", ease: a = "easeInOut", times: l } = {}) {
  const u = { [e]: n };
  l && (u.offset = l);
  const c = ci(a, i);
  return Array.isArray(c) && (u.easing = c), t.animate(u, {
    delay: s,
    duration: i,
    easing: Array.isArray(c) ? "linear" : c,
    fill: "both",
    iterations: o + 1,
    direction: r === "reverse" ? "alternate" : "normal"
  });
}
const Ka = /* @__PURE__ */ en(() => Object.hasOwnProperty.call(Element.prototype, "animate")), Ht = 10, $a = 2e4;
function Wa(t) {
  return nn(t.type) || t.type === "spring" || !ui(t.ease);
}
function Ga(t, e) {
  const n = new hn({
    ...e,
    keyframes: t,
    repeat: 0,
    delay: 0,
    isGenerator: !0
  });
  let s = { done: !1, value: t[0] };
  const i = [];
  let o = 0;
  for (; !s.done && o < $a; )
    s = n.sample(o), i.push(s.value), o += Ht;
  return {
    times: void 0,
    keyframes: i,
    duration: o - Ht,
    ease: "linear"
  };
}
const _i = {
  anticipate: gi,
  backInOut: pi,
  circInOut: vi
};
function za(t) {
  return t in _i;
}
class Kn extends Fi {
  constructor(e) {
    super(e);
    const { name: n, motionValue: s, element: i, keyframes: o } = this.options;
    this.resolver = new Bi(o, (r, a) => this.onKeyframesResolved(r, a), n, s, i), this.resolver.scheduleResolve();
  }
  initPlayback(e, n) {
    let { duration: s = 300, times: i, ease: o, type: r, motionValue: a, name: l, startTime: u } = this.options;
    if (!a.owner || !a.owner.current)
      return !1;
    if (typeof o == "string" && Gt() && za(o) && (o = _i[o]), Wa(this.options)) {
      const { onComplete: h, onUpdate: f, motionValue: d, element: m, ...p } = this.options, y = Ga(e, p);
      e = y.keyframes, e.length === 1 && (e[1] = e[0]), s = y.duration, i = y.times, o = y.ease, r = "keyframes";
    }
    const c = _a(a.owner.current, l, e, { ...this.options, duration: s, times: i, ease: o });
    return c.startTime = u ?? this.calcStartTime(), this.pendingTimeline ? (Vn(c, this.pendingTimeline), this.pendingTimeline = void 0) : c.onfinish = () => {
      const { onComplete: h } = this.options;
      a.set(te(e, this.options, n)), h && h(), this.cancel(), this.resolveFinishedPromise();
    }, {
      animation: c,
      duration: s,
      times: i,
      type: r,
      ease: o,
      keyframes: e
    };
  }
  get duration() {
    const { resolved: e } = this;
    if (!e)
      return 0;
    const { duration: n } = e;
    return /* @__PURE__ */ z(n);
  }
  get time() {
    const { resolved: e } = this;
    if (!e)
      return 0;
    const { animation: n } = e;
    return /* @__PURE__ */ z(n.currentTime || 0);
  }
  set time(e) {
    const { resolved: n } = this;
    if (!n)
      return;
    const { animation: s } = n;
    s.currentTime = /* @__PURE__ */ W(e);
  }
  get speed() {
    const { resolved: e } = this;
    if (!e)
      return 1;
    const { animation: n } = e;
    return n.playbackRate;
  }
  set speed(e) {
    const { resolved: n } = this;
    if (!n)
      return;
    const { animation: s } = n;
    s.playbackRate = e;
  }
  get state() {
    const { resolved: e } = this;
    if (!e)
      return "idle";
    const { animation: n } = e;
    return n.playState;
  }
  get startTime() {
    const { resolved: e } = this;
    if (!e)
      return null;
    const { animation: n } = e;
    return n.startTime;
  }
  /**
   * Replace the default DocumentTimeline with another AnimationTimeline.
   * Currently used for scroll animations.
   */
  attachTimeline(e) {
    if (!this._resolved)
      this.pendingTimeline = e;
    else {
      const { resolved: n } = this;
      if (!n)
        return I;
      const { animation: s } = n;
      Vn(s, e);
    }
    return I;
  }
  play() {
    if (this.isStopped)
      return;
    const { resolved: e } = this;
    if (!e)
      return;
    const { animation: n } = e;
    n.playState === "finished" && this.updateFinishedPromise(), n.play();
  }
  pause() {
    const { resolved: e } = this;
    if (!e)
      return;
    const { animation: n } = e;
    n.pause();
  }
  stop() {
    if (this.resolver.cancel(), this.isStopped = !0, this.state === "idle")
      return;
    this.resolveFinishedPromise(), this.updateFinishedPromise();
    const { resolved: e } = this;
    if (!e)
      return;
    const { animation: n, keyframes: s, duration: i, type: o, ease: r, times: a } = e;
    if (n.playState === "idle" || n.playState === "finished")
      return;
    if (this.time) {
      const { motionValue: u, onUpdate: c, onComplete: h, element: f, ...d } = this.options, m = new hn({
        ...d,
        keyframes: s,
        duration: i,
        type: o,
        ease: r,
        times: a,
        isGenerator: !0
      }), p = /* @__PURE__ */ W(this.time);
      u.setWithVelocity(m.sample(p - Ht).value, m.sample(p).value, Ht);
    }
    const { onStop: l } = this.options;
    l && l(), this.cancel();
  }
  complete() {
    const { resolved: e } = this;
    e && e.animation.finish();
  }
  cancel() {
    const { resolved: e } = this;
    e && e.animation.cancel();
  }
  static supports(e) {
    const { motionValue: n, name: s, repeatDelay: i, repeatType: o, damping: r, type: a } = e;
    if (!n || !n.owner || !(n.owner.current instanceof HTMLElement))
      return !1;
    const { onUpdate: l, transformTemplate: u } = n.owner.getProps();
    return Ka() && s && Ua.has(s) && /**
     * If we're outputting values to onUpdate then we can't use WAAPI as there's
     * no way to read the value from WAAPI every frame.
     */
    !l && !u && !i && o !== "mirror" && r !== 0 && a !== "inertia";
  }
}
const Ha = {
  type: "spring",
  stiffness: 500,
  damping: 25,
  restSpeed: 10
}, Ya = (t) => ({
  type: "spring",
  stiffness: 550,
  damping: t === 0 ? 2 * Math.sqrt(550) : 30,
  restSpeed: 10
}), Xa = {
  type: "keyframes",
  duration: 0.8
}, qa = {
  type: "keyframes",
  ease: [0.25, 0.1, 0.35, 1],
  duration: 0.3
}, Za = (t, { keyframes: e }) => e.length > 2 ? Xa : it.has(t) ? t.startsWith("scale") ? Ya(e[1]) : Ha : qa;
function Ja({ when: t, delay: e, delayChildren: n, staggerChildren: s, staggerDirection: i, repeat: o, repeatType: r, repeatDelay: a, from: l, elapsed: u, ...c }) {
  return !!Object.keys(c).length;
}
const fn = (t, e, n, s = {}, i, o) => (r) => {
  const a = Ze(s, t) || {}, l = a.delay || s.delay || 0;
  let { elapsed: u = 0 } = s;
  u = u - /* @__PURE__ */ W(l);
  let c = {
    keyframes: Array.isArray(n) ? n : [null, n],
    ease: "easeOut",
    velocity: e.getVelocity(),
    ...a,
    delay: -u,
    onUpdate: (f) => {
      e.set(f), a.onUpdate && a.onUpdate(f);
    },
    onComplete: () => {
      r(), a.onComplete && a.onComplete();
    },
    name: t,
    motionValue: e,
    element: o ? void 0 : i
  };
  Ja(a) || (c = {
    ...c,
    ...Za(t, c)
  }), c.duration && (c.duration = /* @__PURE__ */ W(c.duration)), c.repeatDelay && (c.repeatDelay = /* @__PURE__ */ W(c.repeatDelay)), c.from !== void 0 && (c.keyframes[0] = c.from);
  let h = !1;
  if ((c.type === !1 || c.duration === 0 && !c.repeatDelay) && (c.duration = 0, c.delay === 0 && (h = !0)), c.allowFlatten = !a.type && !a.ease, h && !o && e.get() !== void 0) {
    const f = te(c.keyframes, a);
    if (f !== void 0)
      return w.update(() => {
        c.onUpdate(f), c.onComplete();
      }), new Vo([]);
  }
  return !o && Kn.supports(c) ? new Kn(c) : new hn(c);
};
function Qa({ protectedKeys: t, needsAnimating: e }, n) {
  const s = t.hasOwnProperty(n) && e[n] !== !0;
  return e[n] = !1, s;
}
function Ki(t, e, { delay: n = 0, transitionOverride: s, type: i } = {}) {
  var o;
  let { transition: r = t.getDefaultTransition(), transitionEnd: a, ...l } = e;
  s && (r = s);
  const u = [], c = i && t.animationState && t.animationState.getState()[i];
  for (const h in l) {
    const f = t.getValue(h, (o = t.latestValues[h]) !== null && o !== void 0 ? o : null), d = l[h];
    if (d === void 0 || c && Qa(c, h))
      continue;
    const m = {
      delay: n,
      ...Ze(r || {}, h)
    };
    let p = !1;
    if (window.MotionHandoffAnimation) {
      const g = ai(t);
      if (g) {
        const v = window.MotionHandoffAnimation(g, h, w);
        v !== null && (m.startTime = v, p = !0);
      }
    }
    ge(t, h), f.start(fn(h, f, d, t.shouldReduceMotion && ri.has(h) ? { type: !1 } : m, t, p));
    const y = f.animation;
    y && u.push(y);
  }
  return a && Promise.all(u).then(() => {
    w.update(() => {
      a && So(t, a);
    });
  }), u;
}
function Me(t, e, n = {}) {
  var s;
  const i = Ct(t, e, n.type === "exit" ? (s = t.presenceContext) === null || s === void 0 ? void 0 : s.custom : void 0);
  let { transition: o = t.getDefaultTransition() || {} } = i || {};
  n.transitionOverride && (o = n.transitionOverride);
  const r = i ? () => Promise.all(Ki(t, i, n)) : () => Promise.resolve(), a = t.variantChildren && t.variantChildren.size ? (u = 0) => {
    const { delayChildren: c = 0, staggerChildren: h, staggerDirection: f } = o;
    return tl(t, e, c + u, h, f, n);
  } : () => Promise.resolve(), { when: l } = o;
  if (l) {
    const [u, c] = l === "beforeChildren" ? [r, a] : [a, r];
    return u().then(() => c());
  } else
    return Promise.all([r(), a(n.delay)]);
}
function tl(t, e, n = 0, s = 0, i = 1, o) {
  const r = [], a = (t.variantChildren.size - 1) * s, l = i === 1 ? (u = 0) => u * s : (u = 0) => a - u * s;
  return Array.from(t.variantChildren).sort(el).forEach((u, c) => {
    u.notify("AnimationStart", e), r.push(Me(u, e, {
      ...o,
      delay: n + l(c)
    }).then(() => u.notify("AnimationComplete", e)));
  }), Promise.all(r);
}
function el(t, e) {
  return t.sortNodePosition(e);
}
function nl(t, e, n = {}) {
  t.notify("AnimationStart", e);
  let s;
  if (Array.isArray(e)) {
    const i = e.map((o) => Me(t, o, n));
    s = Promise.all(i);
  } else if (typeof e == "string")
    s = Me(t, e, n);
  else {
    const i = typeof e == "function" ? Ct(t, e, n.custom) : e;
    s = Promise.all(Ki(t, i, n));
  }
  return s.then(() => {
    t.notify("AnimationComplete", e);
  });
}
function $i(t, e) {
  if (!Array.isArray(e))
    return !1;
  const n = e.length;
  if (n !== t.length)
    return !1;
  for (let s = 0; s < n; s++)
    if (e[s] !== t[s])
      return !1;
  return !0;
}
const sl = je.length;
function Wi(t) {
  if (!t)
    return;
  if (!t.isControllingVariants) {
    const n = t.parent ? Wi(t.parent) || {} : {};
    return t.props.initial !== void 0 && (n.initial = t.props.initial), n;
  }
  const e = {};
  for (let n = 0; n < sl; n++) {
    const s = je[n], i = t.props[s];
    (Vt(i) || i === !1) && (e[s] = i);
  }
  return e;
}
const il = [...Ie].reverse(), rl = Ie.length;
function ol(t) {
  return (e) => Promise.all(e.map(({ animation: n, options: s }) => nl(t, n, s)));
}
function al(t) {
  let e = ol(t), n = $n(), s = !0;
  const i = (l) => (u, c) => {
    var h;
    const f = Ct(t, c, l === "exit" ? (h = t.presenceContext) === null || h === void 0 ? void 0 : h.custom : void 0);
    if (f) {
      const { transition: d, transitionEnd: m, ...p } = f;
      u = { ...u, ...p, ...m };
    }
    return u;
  };
  function o(l) {
    e = l(t);
  }
  function r(l) {
    const { props: u } = t, c = Wi(t.parent) || {}, h = [], f = /* @__PURE__ */ new Set();
    let d = {}, m = 1 / 0;
    for (let y = 0; y < rl; y++) {
      const g = il[y], v = n[g], T = u[g] !== void 0 ? u[g] : c[g], b = Vt(T), P = g === l ? v.isActive : null;
      P === !1 && (m = y);
      let A = T === c[g] && T !== u[g] && b;
      if (A && s && t.manuallyAnimateOnMount && (A = !1), v.protectedKeys = { ...d }, // If it isn't active and hasn't *just* been set as inactive
      !v.isActive && P === null || // If we didn't and don't have any defined prop for this animation type
      !T && !v.prevProp || // Or if the prop doesn't define an animation
      Jt(T) || typeof T == "boolean")
        continue;
      const C = ll(v.prevProp, T);
      let S = C || // If we're making this variant active, we want to always make it active
      g === l && v.isActive && !A && b || // If we removed a higher-priority variant (i is in reverse order)
      y > m && b, k = !1;
      const j = Array.isArray(T) ? T : [T];
      let rt = j.reduce(i(g), {});
      P === !1 && (rt = {});
      const { prevResolvedValues: mn = {} } = v, mr = {
        ...mn,
        ...rt
      }, pn = (B) => {
        S = !0, f.has(B) && (k = !0, f.delete(B)), v.needsAnimating[B] = !0;
        const G = t.getValue(B);
        G && (G.liveStyle = !1);
      };
      for (const B in mr) {
        const G = rt[B], ee = mn[B];
        if (d.hasOwnProperty(B))
          continue;
        let ne = !1;
        pe(G) && pe(ee) ? ne = !$i(G, ee) : ne = G !== ee, ne ? G != null ? pn(B) : f.add(B) : G !== void 0 && f.has(B) ? pn(B) : v.protectedKeys[B] = !0;
      }
      v.prevProp = T, v.prevResolvedValues = rt, v.isActive && (d = { ...d, ...rt }), s && t.blockInitialAnimation && (S = !1), S && (!(A && C) || k) && h.push(...j.map((B) => ({
        animation: B,
        options: { type: g }
      })));
    }
    if (f.size) {
      const y = {};
      if (typeof u.initial != "boolean") {
        const g = Ct(t, Array.isArray(u.initial) ? u.initial[0] : u.initial);
        g && g.transition && (y.transition = g.transition);
      }
      f.forEach((g) => {
        const v = t.getBaseTarget(g), T = t.getValue(g);
        T && (T.liveStyle = !0), y[g] = v ?? null;
      }), h.push({ animation: y });
    }
    let p = !!h.length;
    return s && (u.initial === !1 || u.initial === u.animate) && !t.manuallyAnimateOnMount && (p = !1), s = !1, p ? e(h) : Promise.resolve();
  }
  function a(l, u) {
    var c;
    if (n[l].isActive === u)
      return Promise.resolve();
    (c = t.variantChildren) === null || c === void 0 || c.forEach((f) => {
      var d;
      return (d = f.animationState) === null || d === void 0 ? void 0 : d.setActive(l, u);
    }), n[l].isActive = u;
    const h = r(l);
    for (const f in n)
      n[f].protectedKeys = {};
    return h;
  }
  return {
    animateChanges: r,
    setActive: a,
    setAnimateFunction: o,
    getState: () => n,
    reset: () => {
      n = $n(), s = !0;
    }
  };
}
function ll(t, e) {
  return typeof e == "string" ? e !== t : Array.isArray(e) ? !$i(e, t) : !1;
}
function Q(t = !1) {
  return {
    isActive: t,
    protectedKeys: {},
    needsAnimating: {},
    prevResolvedValues: {}
  };
}
function $n() {
  return {
    animate: Q(!0),
    whileInView: Q(),
    whileHover: Q(),
    whileTap: Q(),
    whileDrag: Q(),
    whileFocus: Q(),
    exit: Q()
  };
}
class J {
  constructor(e) {
    this.isMounted = !1, this.node = e;
  }
  update() {
  }
}
class ul extends J {
  /**
   * We dynamically generate the AnimationState manager as it contains a reference
   * to the underlying animation library. We only want to load that if we load this,
   * so people can optionally code split it out using the `m` component.
   */
  constructor(e) {
    super(e), e.animationState || (e.animationState = al(e));
  }
  updateAnimationControlsSubscription() {
    const { animate: e } = this.node.getProps();
    Jt(e) && (this.unmountControls = e.subscribe(this.node));
  }
  /**
   * Subscribe any provided AnimationControls to the component's VisualElement
   */
  mount() {
    this.updateAnimationControlsSubscription();
  }
  update() {
    const { animate: e } = this.node.getProps(), { animate: n } = this.node.prevProps || {};
    e !== n && this.updateAnimationControlsSubscription();
  }
  unmount() {
    var e;
    this.node.animationState.reset(), (e = this.unmountControls) === null || e === void 0 || e.call(this);
  }
}
let cl = 0;
class hl extends J {
  constructor() {
    super(...arguments), this.id = cl++;
  }
  update() {
    if (!this.node.presenceContext)
      return;
    const { isPresent: e, onExitComplete: n } = this.node.presenceContext, { isPresent: s } = this.node.prevPresenceContext || {};
    if (!this.node.animationState || e === s)
      return;
    const i = this.node.animationState.setActive("exit", !e);
    n && !e && i.then(() => {
      n(this.id);
    });
  }
  mount() {
    const { register: e, onExitComplete: n } = this.node.presenceContext || {};
    n && n(this.id), e && (this.unmount = e(this.id));
  }
  unmount() {
  }
}
const fl = {
  animation: {
    Feature: ul
  },
  exit: {
    Feature: hl
  }
}, _ = {
  x: !1,
  y: !1
};
function Gi() {
  return _.x || _.y;
}
function dl(t) {
  return t === "x" || t === "y" ? _[t] ? null : (_[t] = !0, () => {
    _[t] = !1;
  }) : _.x || _.y ? null : (_.x = _.y = !0, () => {
    _.x = _.y = !1;
  });
}
function Lt(t, e, n, s = { passive: !0 }) {
  return t.addEventListener(e, n, s), () => t.removeEventListener(e, n);
}
const dn = (t) => t.pointerType === "mouse" ? typeof t.button != "number" || t.button <= 0 : t.isPrimary !== !1;
function Ot(t) {
  return {
    point: {
      x: t.pageX,
      y: t.pageY
    }
  };
}
const ml = (t) => (e) => dn(e) && t(e, Ot(e));
function bt(t, e, n, s) {
  return Lt(t, e, ml(n), s);
}
function zi({ top: t, left: e, right: n, bottom: s }) {
  return {
    x: { min: e, max: n },
    y: { min: t, max: s }
  };
}
function pl({ x: t, y: e }) {
  return { top: e.min, right: t.max, bottom: e.max, left: t.min };
}
function gl(t, e) {
  if (!e)
    return t;
  const n = e({ x: t.left, y: t.top }), s = e({ x: t.right, y: t.bottom });
  return {
    top: n.y,
    left: n.x,
    bottom: s.y,
    right: s.x
  };
}
const Hi = 1e-4, yl = 1 - Hi, vl = 1 + Hi, Yi = 0.01, Tl = 0 - Yi, xl = 0 + Yi;
function F(t) {
  return t.max - t.min;
}
function Pl(t, e, n) {
  return Math.abs(t - e) <= n;
}
function Wn(t, e, n, s = 0.5) {
  t.origin = s, t.originPoint = D(e.min, e.max, t.origin), t.scale = F(n) / F(e), t.translate = D(n.min, n.max, t.origin) - t.originPoint, (t.scale >= yl && t.scale <= vl || isNaN(t.scale)) && (t.scale = 1), (t.translate >= Tl && t.translate <= xl || isNaN(t.translate)) && (t.translate = 0);
}
function At(t, e, n, s) {
  Wn(t.x, e.x, n.x, s ? s.originX : void 0), Wn(t.y, e.y, n.y, s ? s.originY : void 0);
}
function Gn(t, e, n) {
  t.min = n.min + e.min, t.max = t.min + F(e);
}
function Sl(t, e, n) {
  Gn(t.x, e.x, n.x), Gn(t.y, e.y, n.y);
}
function zn(t, e, n) {
  t.min = e.min - n.min, t.max = t.min + F(e);
}
function wt(t, e, n) {
  zn(t.x, e.x, n.x), zn(t.y, e.y, n.y);
}
const Hn = () => ({
  translate: 0,
  scale: 1,
  origin: 0,
  originPoint: 0
}), lt = () => ({
  x: Hn(),
  y: Hn()
}), Yn = () => ({ min: 0, max: 0 }), M = () => ({
  x: Yn(),
  y: Yn()
});
function U(t) {
  return [t("x"), t("y")];
}
function ae(t) {
  return t === void 0 || t === 1;
}
function Ce({ scale: t, scaleX: e, scaleY: n }) {
  return !ae(t) || !ae(e) || !ae(n);
}
function tt(t) {
  return Ce(t) || Xi(t) || t.z || t.rotate || t.rotateX || t.rotateY || t.skewX || t.skewY;
}
function Xi(t) {
  return Xn(t.x) || Xn(t.y);
}
function Xn(t) {
  return t && t !== "0%";
}
function Yt(t, e, n) {
  const s = t - n, i = e * s;
  return n + i;
}
function qn(t, e, n, s, i) {
  return i !== void 0 && (t = Yt(t, i, s)), Yt(t, n, s) + e;
}
function Re(t, e = 0, n = 1, s, i) {
  t.min = qn(t.min, e, n, s, i), t.max = qn(t.max, e, n, s, i);
}
function qi(t, { x: e, y: n }) {
  Re(t.x, e.translate, e.scale, e.originPoint), Re(t.y, n.translate, n.scale, n.originPoint);
}
const Zn = 0.999999999999, Jn = 1.0000000000001;
function bl(t, e, n, s = !1) {
  const i = n.length;
  if (!i)
    return;
  e.x = e.y = 1;
  let o, r;
  for (let a = 0; a < i; a++) {
    o = n[a], r = o.projectionDelta;
    const { visualElement: l } = o.options;
    l && l.props.style && l.props.style.display === "contents" || (s && o.options.layoutScroll && o.scroll && o !== o.root && ct(t, {
      x: -o.scroll.offset.x,
      y: -o.scroll.offset.y
    }), r && (e.x *= r.x.scale, e.y *= r.y.scale, qi(t, r)), s && tt(o.latestValues) && ct(t, o.latestValues));
  }
  e.x < Jn && e.x > Zn && (e.x = 1), e.y < Jn && e.y > Zn && (e.y = 1);
}
function ut(t, e) {
  t.min = t.min + e, t.max = t.max + e;
}
function Qn(t, e, n, s, i = 0.5) {
  const o = D(t.min, t.max, i);
  Re(t, e, n, o, s);
}
function ct(t, e) {
  Qn(t.x, e.x, e.scaleX, e.scale, e.originX), Qn(t.y, e.y, e.scaleY, e.scale, e.originY);
}
function Zi(t, e) {
  return zi(gl(t.getBoundingClientRect(), e));
}
function Al(t, e, n) {
  const s = Zi(t, n), { scroll: i } = e;
  return i && (ut(s.x, i.offset.x), ut(s.y, i.offset.y)), s;
}
const Ji = ({ current: t }) => t ? t.ownerDocument.defaultView : null, ts = (t, e) => Math.abs(t - e);
function wl(t, e) {
  const n = ts(t.x, e.x), s = ts(t.y, e.y);
  return Math.sqrt(n ** 2 + s ** 2);
}
class Qi {
  constructor(e, n, { transformPagePoint: s, contextWindow: i, dragSnapToOrigin: o = !1 } = {}) {
    if (this.startEvent = null, this.lastMoveEvent = null, this.lastMoveEventInfo = null, this.handlers = {}, this.contextWindow = window, this.updatePoint = () => {
      if (!(this.lastMoveEvent && this.lastMoveEventInfo))
        return;
      const h = ue(this.lastMoveEventInfo, this.history), f = this.startEvent !== null, d = wl(h.offset, { x: 0, y: 0 }) >= 3;
      if (!f && !d)
        return;
      const { point: m } = h, { timestamp: p } = R;
      this.history.push({ ...m, timestamp: p });
      const { onStart: y, onMove: g } = this.handlers;
      f || (y && y(this.lastMoveEvent, h), this.startEvent = this.lastMoveEvent), g && g(this.lastMoveEvent, h);
    }, this.handlePointerMove = (h, f) => {
      this.lastMoveEvent = h, this.lastMoveEventInfo = le(f, this.transformPagePoint), w.update(this.updatePoint, !0);
    }, this.handlePointerUp = (h, f) => {
      this.end();
      const { onEnd: d, onSessionEnd: m, resumeAnimation: p } = this.handlers;
      if (this.dragSnapToOrigin && p && p(), !(this.lastMoveEvent && this.lastMoveEventInfo))
        return;
      const y = ue(h.type === "pointercancel" ? this.lastMoveEventInfo : le(f, this.transformPagePoint), this.history);
      this.startEvent && d && d(h, y), m && m(h, y);
    }, !dn(e))
      return;
    this.dragSnapToOrigin = o, this.handlers = n, this.transformPagePoint = s, this.contextWindow = i || window;
    const r = Ot(e), a = le(r, this.transformPagePoint), { point: l } = a, { timestamp: u } = R;
    this.history = [{ ...l, timestamp: u }];
    const { onSessionStart: c } = n;
    c && c(e, ue(a, this.history)), this.removeListeners = kt(bt(this.contextWindow, "pointermove", this.handlePointerMove), bt(this.contextWindow, "pointerup", this.handlePointerUp), bt(this.contextWindow, "pointercancel", this.handlePointerUp));
  }
  updateHandlers(e) {
    this.handlers = e;
  }
  end() {
    this.removeListeners && this.removeListeners(), q(this.updatePoint);
  }
}
function le(t, e) {
  return e ? { point: e(t.point) } : t;
}
function es(t, e) {
  return { x: t.x - e.x, y: t.y - e.y };
}
function ue({ point: t }, e) {
  return {
    point: t,
    delta: es(t, tr(e)),
    offset: es(t, Vl(e)),
    velocity: Dl(e, 0.1)
  };
}
function Vl(t) {
  return t[0];
}
function tr(t) {
  return t[t.length - 1];
}
function Dl(t, e) {
  if (t.length < 2)
    return { x: 0, y: 0 };
  let n = t.length - 1, s = null;
  const i = tr(t);
  for (; n >= 0 && (s = t[n], !(i.timestamp - s.timestamp > /* @__PURE__ */ W(e))); )
    n--;
  if (!s)
    return { x: 0, y: 0 };
  const o = /* @__PURE__ */ z(i.timestamp - s.timestamp);
  if (o === 0)
    return { x: 0, y: 0 };
  const r = {
    x: (i.x - s.x) / o,
    y: (i.y - s.y) / o
  };
  return r.x === 1 / 0 && (r.x = 0), r.y === 1 / 0 && (r.y = 0), r;
}
function Ml(t, { min: e, max: n }, s) {
  return e !== void 0 && t < e ? t = s ? D(e, t, s.min) : Math.max(t, e) : n !== void 0 && t > n && (t = s ? D(n, t, s.max) : Math.min(t, n)), t;
}
function ns(t, e, n) {
  return {
    min: e !== void 0 ? t.min + e : void 0,
    max: n !== void 0 ? t.max + n - (t.max - t.min) : void 0
  };
}
function Cl(t, { top: e, left: n, bottom: s, right: i }) {
  return {
    x: ns(t.x, n, i),
    y: ns(t.y, e, s)
  };
}
function ss(t, e) {
  let n = e.min - t.min, s = e.max - t.max;
  return e.max - e.min < t.max - t.min && ([n, s] = [s, n]), { min: n, max: s };
}
function Rl(t, e) {
  return {
    x: ss(t.x, e.x),
    y: ss(t.y, e.y)
  };
}
function El(t, e) {
  let n = 0.5;
  const s = F(t), i = F(e);
  return i > s ? n = /* @__PURE__ */ ft(e.min, e.max - s, t.min) : s > i && (n = /* @__PURE__ */ ft(t.min, t.max - i, e.min)), H(0, 1, n);
}
function Ll(t, e) {
  const n = {};
  return e.min !== void 0 && (n.min = e.min - t.min), e.max !== void 0 && (n.max = e.max - t.min), n;
}
const Ee = 0.35;
function Bl(t = Ee) {
  return t === !1 ? t = 0 : t === !0 && (t = Ee), {
    x: is(t, "left", "right"),
    y: is(t, "top", "bottom")
  };
}
function is(t, e, n) {
  return {
    min: rs(t, e),
    max: rs(t, n)
  };
}
function rs(t, e) {
  return typeof t == "number" ? t : t[e] || 0;
}
const Fl = /* @__PURE__ */ new WeakMap();
class kl {
  constructor(e) {
    this.openDragLock = null, this.isDragging = !1, this.currentDirection = null, this.originPoint = { x: 0, y: 0 }, this.constraints = !1, this.hasMutatedConstraints = !1, this.elastic = M(), this.visualElement = e;
  }
  start(e, { snapToCursor: n = !1 } = {}) {
    const { presenceContext: s } = this.visualElement;
    if (s && s.isPresent === !1)
      return;
    const i = (c) => {
      const { dragSnapToOrigin: h } = this.getProps();
      h ? this.pauseAnimation() : this.stopAnimation(), n && this.snapToCursor(Ot(c).point);
    }, o = (c, h) => {
      const { drag: f, dragPropagation: d, onDragStart: m } = this.getProps();
      if (f && !d && (this.openDragLock && this.openDragLock(), this.openDragLock = dl(f), !this.openDragLock))
        return;
      this.isDragging = !0, this.currentDirection = null, this.resolveConstraints(), this.visualElement.projection && (this.visualElement.projection.isAnimationBlocked = !0, this.visualElement.projection.target = void 0), U((y) => {
        let g = this.getAxisMotionValue(y).get() || 0;
        if (K.test(g)) {
          const { projection: v } = this.visualElement;
          if (v && v.layout) {
            const T = v.layout.layoutBox[y];
            T && (g = F(T) * (parseFloat(g) / 100));
          }
        }
        this.originPoint[y] = g;
      }), m && w.postRender(() => m(c, h)), ge(this.visualElement, "transform");
      const { animationState: p } = this.visualElement;
      p && p.setActive("whileDrag", !0);
    }, r = (c, h) => {
      const { dragPropagation: f, dragDirectionLock: d, onDirectionLock: m, onDrag: p } = this.getProps();
      if (!f && !this.openDragLock)
        return;
      const { offset: y } = h;
      if (d && this.currentDirection === null) {
        this.currentDirection = Ol(y), this.currentDirection !== null && m && m(this.currentDirection);
        return;
      }
      this.updateAxis("x", h.point, y), this.updateAxis("y", h.point, y), this.visualElement.render(), p && p(c, h);
    }, a = (c, h) => this.stop(c, h), l = () => U((c) => {
      var h;
      return this.getAnimationState(c) === "paused" && ((h = this.getAxisMotionValue(c).animation) === null || h === void 0 ? void 0 : h.play());
    }), { dragSnapToOrigin: u } = this.getProps();
    this.panSession = new Qi(e, {
      onSessionStart: i,
      onStart: o,
      onMove: r,
      onSessionEnd: a,
      resumeAnimation: l
    }, {
      transformPagePoint: this.visualElement.getTransformPagePoint(),
      dragSnapToOrigin: u,
      contextWindow: Ji(this.visualElement)
    });
  }
  stop(e, n) {
    const s = this.isDragging;
    if (this.cancel(), !s)
      return;
    const { velocity: i } = n;
    this.startAnimation(i);
    const { onDragEnd: o } = this.getProps();
    o && w.postRender(() => o(e, n));
  }
  cancel() {
    this.isDragging = !1;
    const { projection: e, animationState: n } = this.visualElement;
    e && (e.isAnimationBlocked = !1), this.panSession && this.panSession.end(), this.panSession = void 0;
    const { dragPropagation: s } = this.getProps();
    !s && this.openDragLock && (this.openDragLock(), this.openDragLock = null), n && n.setActive("whileDrag", !1);
  }
  updateAxis(e, n, s) {
    const { drag: i } = this.getProps();
    if (!s || !Nt(e, i, this.currentDirection))
      return;
    const o = this.getAxisMotionValue(e);
    let r = this.originPoint[e] + s[e];
    this.constraints && this.constraints[e] && (r = Ml(r, this.constraints[e], this.elastic[e])), o.set(r);
  }
  resolveConstraints() {
    var e;
    const { dragConstraints: n, dragElastic: s } = this.getProps(), i = this.visualElement.projection && !this.visualElement.projection.layout ? this.visualElement.projection.measure(!1) : (e = this.visualElement.projection) === null || e === void 0 ? void 0 : e.layout, o = this.constraints;
    n && ot(n) ? this.constraints || (this.constraints = this.resolveRefConstraints()) : n && i ? this.constraints = Cl(i.layoutBox, n) : this.constraints = !1, this.elastic = Bl(s), o !== this.constraints && i && this.constraints && !this.hasMutatedConstraints && U((r) => {
      this.constraints !== !1 && this.getAxisMotionValue(r) && (this.constraints[r] = Ll(i.layoutBox[r], this.constraints[r]));
    });
  }
  resolveRefConstraints() {
    const { dragConstraints: e, onMeasureDragConstraints: n } = this.getProps();
    if (!e || !ot(e))
      return !1;
    const s = e.current;
    X(s !== null, "If `dragConstraints` is set as a React ref, that ref must be passed to another component's `ref` prop.");
    const { projection: i } = this.visualElement;
    if (!i || !i.layout)
      return !1;
    const o = Al(s, i.root, this.visualElement.getTransformPagePoint());
    let r = Rl(i.layout.layoutBox, o);
    if (n) {
      const a = n(pl(r));
      this.hasMutatedConstraints = !!a, a && (r = zi(a));
    }
    return r;
  }
  startAnimation(e) {
    const { drag: n, dragMomentum: s, dragElastic: i, dragTransition: o, dragSnapToOrigin: r, onDragTransitionEnd: a } = this.getProps(), l = this.constraints || {}, u = U((c) => {
      if (!Nt(c, n, this.currentDirection))
        return;
      let h = l && l[c] || {};
      r && (h = { min: 0, max: 0 });
      const f = i ? 200 : 1e6, d = i ? 40 : 1e7, m = {
        type: "inertia",
        velocity: s ? e[c] : 0,
        bounceStiffness: f,
        bounceDamping: d,
        timeConstant: 750,
        restDelta: 1,
        restSpeed: 10,
        ...o,
        ...h
      };
      return this.startAxisValueAnimation(c, m);
    });
    return Promise.all(u).then(a);
  }
  startAxisValueAnimation(e, n) {
    const s = this.getAxisMotionValue(e);
    return ge(this.visualElement, e), s.start(fn(e, s, 0, n, this.visualElement, !1));
  }
  stopAnimation() {
    U((e) => this.getAxisMotionValue(e).stop());
  }
  pauseAnimation() {
    U((e) => {
      var n;
      return (n = this.getAxisMotionValue(e).animation) === null || n === void 0 ? void 0 : n.pause();
    });
  }
  getAnimationState(e) {
    var n;
    return (n = this.getAxisMotionValue(e).animation) === null || n === void 0 ? void 0 : n.state;
  }
  /**
   * Drag works differently depending on which props are provided.
   *
   * - If _dragX and _dragY are provided, we output the gesture delta directly to those motion values.
   * - Otherwise, we apply the delta to the x/y motion values.
   */
  getAxisMotionValue(e) {
    const n = `_drag${e.toUpperCase()}`, s = this.visualElement.getProps(), i = s[n];
    return i || this.visualElement.getValue(e, (s.initial ? s.initial[e] : void 0) || 0);
  }
  snapToCursor(e) {
    U((n) => {
      const { drag: s } = this.getProps();
      if (!Nt(n, s, this.currentDirection))
        return;
      const { projection: i } = this.visualElement, o = this.getAxisMotionValue(n);
      if (i && i.layout) {
        const { min: r, max: a } = i.layout.layoutBox[n];
        o.set(e[n] - D(r, a, 0.5));
      }
    });
  }
  /**
   * When the viewport resizes we want to check if the measured constraints
   * have changed and, if so, reposition the element within those new constraints
   * relative to where it was before the resize.
   */
  scalePositionWithinConstraints() {
    if (!this.visualElement.current)
      return;
    const { drag: e, dragConstraints: n } = this.getProps(), { projection: s } = this.visualElement;
    if (!ot(n) || !s || !this.constraints)
      return;
    this.stopAnimation();
    const i = { x: 0, y: 0 };
    U((r) => {
      const a = this.getAxisMotionValue(r);
      if (a && this.constraints !== !1) {
        const l = a.get();
        i[r] = El({ min: l, max: l }, this.constraints[r]);
      }
    });
    const { transformTemplate: o } = this.visualElement.getProps();
    this.visualElement.current.style.transform = o ? o({}, "") : "none", s.root && s.root.updateScroll(), s.updateLayout(), this.resolveConstraints(), U((r) => {
      if (!Nt(r, e, null))
        return;
      const a = this.getAxisMotionValue(r), { min: l, max: u } = this.constraints[r];
      a.set(D(l, u, i[r]));
    });
  }
  addListeners() {
    if (!this.visualElement.current)
      return;
    Fl.set(this.visualElement, this);
    const e = this.visualElement.current, n = bt(e, "pointerdown", (l) => {
      const { drag: u, dragListener: c = !0 } = this.getProps();
      u && c && this.start(l);
    }), s = () => {
      const { dragConstraints: l } = this.getProps();
      ot(l) && l.current && (this.constraints = this.resolveRefConstraints());
    }, { projection: i } = this.visualElement, o = i.addEventListener("measure", s);
    i && !i.layout && (i.root && i.root.updateScroll(), i.updateLayout()), w.read(s);
    const r = Lt(window, "resize", () => this.scalePositionWithinConstraints()), a = i.addEventListener("didUpdate", ({ delta: l, hasLayoutChanged: u }) => {
      this.isDragging && u && (U((c) => {
        const h = this.getAxisMotionValue(c);
        h && (this.originPoint[c] += l[c].translate, h.set(h.get() + l[c].translate));
      }), this.visualElement.render());
    });
    return () => {
      r(), n(), o(), a && a();
    };
  }
  getProps() {
    const e = this.visualElement.getProps(), { drag: n = !1, dragDirectionLock: s = !1, dragPropagation: i = !1, dragConstraints: o = !1, dragElastic: r = Ee, dragMomentum: a = !0 } = e;
    return {
      ...e,
      drag: n,
      dragDirectionLock: s,
      dragPropagation: i,
      dragConstraints: o,
      dragElastic: r,
      dragMomentum: a
    };
  }
}
function Nt(t, e, n) {
  return (e === !0 || e === t) && (n === null || n === t);
}
function Ol(t, e = 10) {
  let n = null;
  return Math.abs(t.y) > e ? n = "y" : Math.abs(t.x) > e && (n = "x"), n;
}
class Il extends J {
  constructor(e) {
    super(e), this.removeGroupControls = I, this.removeListeners = I, this.controls = new kl(e);
  }
  mount() {
    const { dragControls: e } = this.node.getProps();
    e && (this.removeGroupControls = e.subscribe(this.controls)), this.removeListeners = this.controls.addListeners() || I;
  }
  unmount() {
    this.removeGroupControls(), this.removeListeners();
  }
}
const os = (t) => (e, n) => {
  t && w.postRender(() => t(e, n));
};
class jl extends J {
  constructor() {
    super(...arguments), this.removePointerDownListener = I;
  }
  onPointerDown(e) {
    this.session = new Qi(e, this.createPanHandlers(), {
      transformPagePoint: this.node.getTransformPagePoint(),
      contextWindow: Ji(this.node)
    });
  }
  createPanHandlers() {
    const { onPanSessionStart: e, onPanStart: n, onPan: s, onPanEnd: i } = this.node.getProps();
    return {
      onSessionStart: os(e),
      onStart: os(n),
      onMove: s,
      onEnd: (o, r) => {
        delete this.session, i && w.postRender(() => i(o, r));
      }
    };
  }
  mount() {
    this.removePointerDownListener = bt(this.node.current, "pointerdown", (e) => this.onPointerDown(e));
  }
  update() {
    this.session && this.session.updateHandlers(this.createPanHandlers());
  }
  unmount() {
    this.removePointerDownListener(), this.session && this.session.end();
  }
}
const $t = {
  /**
   * Global flag as to whether the tree has animated since the last time
   * we resized the window
   */
  hasAnimatedSinceResize: !0,
  /**
   * We set this to true once, on the first update. Any nodes added to the tree beyond that
   * update will be given a `data-projection-id` attribute.
   */
  hasEverUpdated: !1
};
function as(t, e) {
  return e.max === e.min ? 0 : t / (e.max - e.min) * 100;
}
const vt = {
  correct: (t, e) => {
    if (!e.target)
      return t;
    if (typeof t == "string")
      if (x.test(t))
        t = parseFloat(t);
      else
        return t;
    const n = as(t, e.target.x), s = as(t, e.target.y);
    return `${n}% ${s}%`;
  }
}, Nl = {
  correct: (t, { treeScale: e, projectionDelta: n }) => {
    const s = t, i = Z.parse(t);
    if (i.length > 5)
      return s;
    const o = Z.createTransformer(t), r = typeof i[0] != "number" ? 1 : 0, a = n.x.scale * e.x, l = n.y.scale * e.y;
    i[0 + r] /= a, i[1 + r] /= l;
    const u = D(a, l, 0.5);
    return typeof i[2 + r] == "number" && (i[2 + r] /= u), typeof i[3 + r] == "number" && (i[3 + r] /= u), o(i);
  }
};
class Ul extends Pr {
  /**
   * This only mounts projection nodes for components that
   * need measuring, we might want to do it for all components
   * in order to incorporate transforms
   */
  componentDidMount() {
    const { visualElement: e, layoutGroup: n, switchLayoutGroup: s, layoutId: i } = this.props, { projection: o } = e;
    zr(_l), o && (n.group && n.group.add(o), s && s.register && i && s.register(o), o.root.didUpdate(), o.addEventListener("animationComplete", () => {
      this.safeToRemove();
    }), o.setOptions({
      ...o.options,
      onExitComplete: () => this.safeToRemove()
    })), $t.hasEverUpdated = !0;
  }
  getSnapshotBeforeUpdate(e) {
    const { layoutDependency: n, visualElement: s, drag: i, isPresent: o } = this.props, r = s.projection;
    return r && (r.isPresent = o, i || e.layoutDependency !== n || n === void 0 || e.isPresent !== o ? r.willUpdate() : this.safeToRemove(), e.isPresent !== o && (o ? r.promote() : r.relegate() || w.postRender(() => {
      const a = r.getStack();
      (!a || !a.members.length) && this.safeToRemove();
    }))), null;
  }
  componentDidUpdate() {
    const { projection: e } = this.props.visualElement;
    e && (e.root.didUpdate(), Ne.postRender(() => {
      !e.currentAnimation && e.isLead() && this.safeToRemove();
    }));
  }
  componentWillUnmount() {
    const { visualElement: e, layoutGroup: n, switchLayoutGroup: s } = this.props, { projection: i } = e;
    i && (i.scheduleCheckAfterUnmount(), n && n.group && n.group.remove(i), s && s.deregister && s.deregister(i));
  }
  safeToRemove() {
    const { safeToRemove: e } = this.props;
    e && e();
  }
  render() {
    return null;
  }
}
function er(t) {
  const [e, n] = Ar(), s = O(Ns);
  return Os(Ul, { ...t, layoutGroup: s, switchLayoutGroup: O(zs), isPresent: e, safeToRemove: n });
}
const _l = {
  borderRadius: {
    ...vt,
    applyTo: [
      "borderTopLeftRadius",
      "borderTopRightRadius",
      "borderBottomLeftRadius",
      "borderBottomRightRadius"
    ]
  },
  borderTopLeftRadius: vt,
  borderTopRightRadius: vt,
  borderBottomLeftRadius: vt,
  borderBottomRightRadius: vt,
  boxShadow: Nl
};
function Kl(t, e, n) {
  const s = L(t) ? t : Rt(t);
  return s.start(fn("", s, e, n)), s.animation;
}
function $l(t) {
  return t instanceof SVGElement && t.tagName !== "svg";
}
const Wl = (t, e) => t.depth - e.depth;
class Gl {
  constructor() {
    this.children = [], this.isDirty = !1;
  }
  add(e) {
    Je(this.children, e), this.isDirty = !0;
  }
  remove(e) {
    Qe(this.children, e), this.isDirty = !0;
  }
  forEach(e) {
    this.isDirty && this.children.sort(Wl), this.isDirty = !1, this.children.forEach(e);
  }
}
function zl(t, e) {
  const n = $.now(), s = ({ timestamp: i }) => {
    const o = i - n;
    o >= e && (q(s), t(o - e));
  };
  return w.read(s, !0), () => q(s);
}
const nr = ["TopLeft", "TopRight", "BottomLeft", "BottomRight"], Hl = nr.length, ls = (t) => typeof t == "string" ? parseFloat(t) : t, us = (t) => typeof t == "number" || x.test(t);
function Yl(t, e, n, s, i, o) {
  i ? (t.opacity = D(
    0,
    // TODO Reinstate this if only child
    n.opacity !== void 0 ? n.opacity : 1,
    Xl(s)
  ), t.opacityExit = D(e.opacity !== void 0 ? e.opacity : 1, 0, ql(s))) : o && (t.opacity = D(e.opacity !== void 0 ? e.opacity : 1, n.opacity !== void 0 ? n.opacity : 1, s));
  for (let r = 0; r < Hl; r++) {
    const a = `border${nr[r]}Radius`;
    let l = cs(e, a), u = cs(n, a);
    if (l === void 0 && u === void 0)
      continue;
    l || (l = 0), u || (u = 0), l === 0 || u === 0 || us(l) === us(u) ? (t[a] = Math.max(D(ls(l), ls(u), s), 0), (K.test(u) || K.test(l)) && (t[a] += "%")) : t[a] = u;
  }
  (e.rotate || n.rotate) && (t.rotate = D(e.rotate || 0, n.rotate || 0, s));
}
function cs(t, e) {
  return t[e] !== void 0 ? t[e] : t.borderRadius;
}
const Xl = /* @__PURE__ */ sr(0, 0.5, yi), ql = /* @__PURE__ */ sr(0.5, 0.95, I);
function sr(t, e, n) {
  return (s) => s < t ? 0 : s > e ? 1 : n(/* @__PURE__ */ ft(t, e, s));
}
function hs(t, e) {
  t.min = e.min, t.max = e.max;
}
function N(t, e) {
  hs(t.x, e.x), hs(t.y, e.y);
}
function fs(t, e) {
  t.translate = e.translate, t.scale = e.scale, t.originPoint = e.originPoint, t.origin = e.origin;
}
function ds(t, e, n, s, i) {
  return t -= e, t = Yt(t, 1 / n, s), i !== void 0 && (t = Yt(t, 1 / i, s)), t;
}
function Zl(t, e = 0, n = 1, s = 0.5, i, o = t, r = t) {
  if (K.test(e) && (e = parseFloat(e), e = D(r.min, r.max, e / 100) - r.min), typeof e != "number")
    return;
  let a = D(o.min, o.max, s);
  t === o && (a -= e), t.min = ds(t.min, e, n, a, i), t.max = ds(t.max, e, n, a, i);
}
function ms(t, e, [n, s, i], o, r) {
  Zl(t, e[n], e[s], e[i], e.scale, o, r);
}
const Jl = ["x", "scaleX", "originX"], Ql = ["y", "scaleY", "originY"];
function ps(t, e, n, s) {
  ms(t.x, e, Jl, n ? n.x : void 0, s ? s.x : void 0), ms(t.y, e, Ql, n ? n.y : void 0, s ? s.y : void 0);
}
function gs(t) {
  return t.translate === 0 && t.scale === 1;
}
function ir(t) {
  return gs(t.x) && gs(t.y);
}
function ys(t, e) {
  return t.min === e.min && t.max === e.max;
}
function tu(t, e) {
  return ys(t.x, e.x) && ys(t.y, e.y);
}
function vs(t, e) {
  return Math.round(t.min) === Math.round(e.min) && Math.round(t.max) === Math.round(e.max);
}
function rr(t, e) {
  return vs(t.x, e.x) && vs(t.y, e.y);
}
function Ts(t) {
  return F(t.x) / F(t.y);
}
function xs(t, e) {
  return t.translate === e.translate && t.scale === e.scale && t.originPoint === e.originPoint;
}
class eu {
  constructor() {
    this.members = [];
  }
  add(e) {
    Je(this.members, e), e.scheduleRender();
  }
  remove(e) {
    if (Qe(this.members, e), e === this.prevLead && (this.prevLead = void 0), e === this.lead) {
      const n = this.members[this.members.length - 1];
      n && this.promote(n);
    }
  }
  relegate(e) {
    const n = this.members.findIndex((i) => e === i);
    if (n === 0)
      return !1;
    let s;
    for (let i = n; i >= 0; i--) {
      const o = this.members[i];
      if (o.isPresent !== !1) {
        s = o;
        break;
      }
    }
    return s ? (this.promote(s), !0) : !1;
  }
  promote(e, n) {
    const s = this.lead;
    if (e !== s && (this.prevLead = s, this.lead = e, e.show(), s)) {
      s.instance && s.scheduleRender(), e.scheduleRender(), e.resumeFrom = s, n && (e.resumeFrom.preserveOpacity = !0), s.snapshot && (e.snapshot = s.snapshot, e.snapshot.latestValues = s.animationValues || s.latestValues), e.root && e.root.isUpdating && (e.isLayoutDirty = !0);
      const { crossfade: i } = e.options;
      i === !1 && s.hide();
    }
  }
  exitAnimationComplete() {
    this.members.forEach((e) => {
      const { options: n, resumingFrom: s } = e;
      n.onExitComplete && n.onExitComplete(), s && s.options.onExitComplete && s.options.onExitComplete();
    });
  }
  scheduleRender() {
    this.members.forEach((e) => {
      e.instance && e.scheduleRender(!1);
    });
  }
  /**
   * Clear any leads that have been removed this render to prevent them from being
   * used in future animations and to prevent memory leaks
   */
  removeLeadSnapshot() {
    this.lead && this.lead.snapshot && (this.lead.snapshot = void 0);
  }
}
function nu(t, e, n) {
  let s = "";
  const i = t.x.translate / e.x, o = t.y.translate / e.y, r = (n == null ? void 0 : n.z) || 0;
  if ((i || o || r) && (s = `translate3d(${i}px, ${o}px, ${r}px) `), (e.x !== 1 || e.y !== 1) && (s += `scale(${1 / e.x}, ${1 / e.y}) `), n) {
    const { transformPerspective: u, rotate: c, rotateX: h, rotateY: f, skewX: d, skewY: m } = n;
    u && (s = `perspective(${u}px) ${s}`), c && (s += `rotate(${c}deg) `), h && (s += `rotateX(${h}deg) `), f && (s += `rotateY(${f}deg) `), d && (s += `skewX(${d}deg) `), m && (s += `skewY(${m}deg) `);
  }
  const a = t.x.scale * e.x, l = t.y.scale * e.y;
  return (a !== 1 || l !== 1) && (s += `scale(${a}, ${l})`), s || "none";
}
const ce = ["", "X", "Y", "Z"], su = { visibility: "hidden" }, Ps = 1e3;
let iu = 0;
function he(t, e, n, s) {
  const { latestValues: i } = e;
  i[t] && (n[t] = i[t], e.setStaticValue(t, 0), s && (s[t] = 0));
}
function or(t) {
  if (t.hasCheckedOptimisedAppear = !0, t.root === t)
    return;
  const { visualElement: e } = t.options;
  if (!e)
    return;
  const n = ai(e);
  if (window.MotionHasOptimisedAnimation(n, "transform")) {
    const { layout: i, layoutId: o } = t.options;
    window.MotionCancelOptimisedAnimation(n, "transform", w, !(i || o));
  }
  const { parent: s } = t;
  s && !s.hasCheckedOptimisedAppear && or(s);
}
function ar({ attachResizeListener: t, defaultParent: e, measureScroll: n, checkIsScrollRoot: s, resetTransform: i }) {
  return class {
    constructor(r = {}, a = e == null ? void 0 : e()) {
      this.id = iu++, this.animationId = 0, this.children = /* @__PURE__ */ new Set(), this.options = {}, this.isTreeAnimating = !1, this.isAnimationBlocked = !1, this.isLayoutDirty = !1, this.isProjectionDirty = !1, this.isSharedProjectionDirty = !1, this.isTransformDirty = !1, this.updateManuallyBlocked = !1, this.updateBlockedByResize = !1, this.isUpdating = !1, this.isSVG = !1, this.needsReset = !1, this.shouldResetTransform = !1, this.hasCheckedOptimisedAppear = !1, this.treeScale = { x: 1, y: 1 }, this.eventHandlers = /* @__PURE__ */ new Map(), this.hasTreeAnimated = !1, this.updateScheduled = !1, this.scheduleUpdate = () => this.update(), this.projectionUpdateScheduled = !1, this.checkUpdateFailed = () => {
        this.isUpdating && (this.isUpdating = !1, this.clearAllSnapshots());
      }, this.updateProjection = () => {
        this.projectionUpdateScheduled = !1, this.nodes.forEach(au), this.nodes.forEach(fu), this.nodes.forEach(du), this.nodes.forEach(lu);
      }, this.resolvedRelativeTargetAt = 0, this.hasProjected = !1, this.isVisible = !0, this.animationProgress = 0, this.sharedNodes = /* @__PURE__ */ new Map(), this.latestValues = r, this.root = a ? a.root || a : this, this.path = a ? [...a.path, a] : [], this.parent = a, this.depth = a ? a.depth + 1 : 0;
      for (let l = 0; l < this.path.length; l++)
        this.path[l].shouldResetTransform = !0;
      this.root === this && (this.nodes = new Gl());
    }
    addEventListener(r, a) {
      return this.eventHandlers.has(r) || this.eventHandlers.set(r, new tn()), this.eventHandlers.get(r).add(a);
    }
    notifyListeners(r, ...a) {
      const l = this.eventHandlers.get(r);
      l && l.notify(...a);
    }
    hasListeners(r) {
      return this.eventHandlers.has(r);
    }
    /**
     * Lifecycles
     */
    mount(r, a = this.root.hasTreeAnimated) {
      if (this.instance)
        return;
      this.isSVG = $l(r), this.instance = r;
      const { layoutId: l, layout: u, visualElement: c } = this.options;
      if (c && !c.current && c.mount(r), this.root.nodes.add(this), this.parent && this.parent.children.add(this), a && (u || l) && (this.isLayoutDirty = !0), t) {
        let h;
        const f = () => this.root.updateBlockedByResize = !1;
        t(r, () => {
          this.root.updateBlockedByResize = !0, h && h(), h = zl(f, 250), $t.hasAnimatedSinceResize && ($t.hasAnimatedSinceResize = !1, this.nodes.forEach(bs));
        });
      }
      l && this.root.registerSharedNode(l, this), this.options.animate !== !1 && c && (l || u) && this.addEventListener("didUpdate", ({ delta: h, hasLayoutChanged: f, hasRelativeLayoutChanged: d, layout: m }) => {
        if (this.isTreeAnimationBlocked()) {
          this.target = void 0, this.relativeTarget = void 0;
          return;
        }
        const p = this.options.transition || c.getDefaultTransition() || vu, { onLayoutAnimationStart: y, onLayoutAnimationComplete: g } = c.getProps(), v = !this.targetLayout || !rr(this.targetLayout, m), T = !f && d;
        if (this.options.layoutRoot || this.resumeFrom || T || f && (v || !this.currentAnimation)) {
          this.resumeFrom && (this.resumingFrom = this.resumeFrom, this.resumingFrom.resumingFrom = void 0), this.setAnimationOrigin(h, T);
          const b = {
            ...Ze(p, "layout"),
            onPlay: y,
            onComplete: g
          };
          (c.shouldReduceMotion || this.options.layoutRoot) && (b.delay = 0, b.type = !1), this.startAnimation(b);
        } else
          f || bs(this), this.isLead() && this.options.onExitComplete && this.options.onExitComplete();
        this.targetLayout = m;
      });
    }
    unmount() {
      this.options.layoutId && this.willUpdate(), this.root.nodes.remove(this);
      const r = this.getStack();
      r && r.remove(this), this.parent && this.parent.children.delete(this), this.instance = void 0, q(this.updateProjection);
    }
    // only on the root
    blockUpdate() {
      this.updateManuallyBlocked = !0;
    }
    unblockUpdate() {
      this.updateManuallyBlocked = !1;
    }
    isUpdateBlocked() {
      return this.updateManuallyBlocked || this.updateBlockedByResize;
    }
    isTreeAnimationBlocked() {
      return this.isAnimationBlocked || this.parent && this.parent.isTreeAnimationBlocked() || !1;
    }
    // Note: currently only running on root node
    startUpdate() {
      this.isUpdateBlocked() || (this.isUpdating = !0, this.nodes && this.nodes.forEach(mu), this.animationId++);
    }
    getTransformTemplate() {
      const { visualElement: r } = this.options;
      return r && r.getProps().transformTemplate;
    }
    willUpdate(r = !0) {
      if (this.root.hasTreeAnimated = !0, this.root.isUpdateBlocked()) {
        this.options.onExitComplete && this.options.onExitComplete();
        return;
      }
      if (window.MotionCancelOptimisedAnimation && !this.hasCheckedOptimisedAppear && or(this), !this.root.isUpdating && this.root.startUpdate(), this.isLayoutDirty)
        return;
      this.isLayoutDirty = !0;
      for (let c = 0; c < this.path.length; c++) {
        const h = this.path[c];
        h.shouldResetTransform = !0, h.updateScroll("snapshot"), h.options.layoutRoot && h.willUpdate(!1);
      }
      const { layoutId: a, layout: l } = this.options;
      if (a === void 0 && !l)
        return;
      const u = this.getTransformTemplate();
      this.prevTransformTemplateValue = u ? u(this.latestValues, "") : void 0, this.updateSnapshot(), r && this.notifyListeners("willUpdate");
    }
    update() {
      if (this.updateScheduled = !1, this.isUpdateBlocked()) {
        this.unblockUpdate(), this.clearAllSnapshots(), this.nodes.forEach(Ss);
        return;
      }
      this.isUpdating || this.nodes.forEach(cu), this.isUpdating = !1, this.nodes.forEach(hu), this.nodes.forEach(ru), this.nodes.forEach(ou), this.clearAllSnapshots();
      const a = $.now();
      R.delta = H(0, 1e3 / 60, a - R.timestamp), R.timestamp = a, R.isProcessing = !0, se.update.process(R), se.preRender.process(R), se.render.process(R), R.isProcessing = !1;
    }
    didUpdate() {
      this.updateScheduled || (this.updateScheduled = !0, Ne.read(this.scheduleUpdate));
    }
    clearAllSnapshots() {
      this.nodes.forEach(uu), this.sharedNodes.forEach(pu);
    }
    scheduleUpdateProjection() {
      this.projectionUpdateScheduled || (this.projectionUpdateScheduled = !0, w.preRender(this.updateProjection, !1, !0));
    }
    scheduleCheckAfterUnmount() {
      w.postRender(() => {
        this.isLayoutDirty ? this.root.didUpdate() : this.root.checkUpdateFailed();
      });
    }
    /**
     * Update measurements
     */
    updateSnapshot() {
      this.snapshot || !this.instance || (this.snapshot = this.measure(), this.snapshot && !F(this.snapshot.measuredBox.x) && !F(this.snapshot.measuredBox.y) && (this.snapshot = void 0));
    }
    updateLayout() {
      if (!this.instance || (this.updateScroll(), !(this.options.alwaysMeasureLayout && this.isLead()) && !this.isLayoutDirty))
        return;
      if (this.resumeFrom && !this.resumeFrom.instance)
        for (let l = 0; l < this.path.length; l++)
          this.path[l].updateScroll();
      const r = this.layout;
      this.layout = this.measure(!1), this.layoutCorrected = M(), this.isLayoutDirty = !1, this.projectionDelta = void 0, this.notifyListeners("measure", this.layout.layoutBox);
      const { visualElement: a } = this.options;
      a && a.notify("LayoutMeasure", this.layout.layoutBox, r ? r.layoutBox : void 0);
    }
    updateScroll(r = "measure") {
      let a = !!(this.options.layoutScroll && this.instance);
      if (this.scroll && this.scroll.animationId === this.root.animationId && this.scroll.phase === r && (a = !1), a) {
        const l = s(this.instance);
        this.scroll = {
          animationId: this.root.animationId,
          phase: r,
          isRoot: l,
          offset: n(this.instance),
          wasRoot: this.scroll ? this.scroll.isRoot : l
        };
      }
    }
    resetTransform() {
      if (!i)
        return;
      const r = this.isLayoutDirty || this.shouldResetTransform || this.options.alwaysMeasureLayout, a = this.projectionDelta && !ir(this.projectionDelta), l = this.getTransformTemplate(), u = l ? l(this.latestValues, "") : void 0, c = u !== this.prevTransformTemplateValue;
      r && (a || tt(this.latestValues) || c) && (i(this.instance, u), this.shouldResetTransform = !1, this.scheduleRender());
    }
    measure(r = !0) {
      const a = this.measurePageBox();
      let l = this.removeElementScroll(a);
      return r && (l = this.removeTransform(l)), Tu(l), {
        animationId: this.root.animationId,
        measuredBox: a,
        layoutBox: l,
        latestValues: {},
        source: this.id
      };
    }
    measurePageBox() {
      var r;
      const { visualElement: a } = this.options;
      if (!a)
        return M();
      const l = a.measureViewportBox();
      if (!(((r = this.scroll) === null || r === void 0 ? void 0 : r.wasRoot) || this.path.some(xu))) {
        const { scroll: c } = this.root;
        c && (ut(l.x, c.offset.x), ut(l.y, c.offset.y));
      }
      return l;
    }
    removeElementScroll(r) {
      var a;
      const l = M();
      if (N(l, r), !((a = this.scroll) === null || a === void 0) && a.wasRoot)
        return l;
      for (let u = 0; u < this.path.length; u++) {
        const c = this.path[u], { scroll: h, options: f } = c;
        c !== this.root && h && f.layoutScroll && (h.wasRoot && N(l, r), ut(l.x, h.offset.x), ut(l.y, h.offset.y));
      }
      return l;
    }
    applyTransform(r, a = !1) {
      const l = M();
      N(l, r);
      for (let u = 0; u < this.path.length; u++) {
        const c = this.path[u];
        !a && c.options.layoutScroll && c.scroll && c !== c.root && ct(l, {
          x: -c.scroll.offset.x,
          y: -c.scroll.offset.y
        }), tt(c.latestValues) && ct(l, c.latestValues);
      }
      return tt(this.latestValues) && ct(l, this.latestValues), l;
    }
    removeTransform(r) {
      const a = M();
      N(a, r);
      for (let l = 0; l < this.path.length; l++) {
        const u = this.path[l];
        if (!u.instance || !tt(u.latestValues))
          continue;
        Ce(u.latestValues) && u.updateSnapshot();
        const c = M(), h = u.measurePageBox();
        N(c, h), ps(a, u.latestValues, u.snapshot ? u.snapshot.layoutBox : void 0, c);
      }
      return tt(this.latestValues) && ps(a, this.latestValues), a;
    }
    setTargetDelta(r) {
      this.targetDelta = r, this.root.scheduleUpdateProjection(), this.isProjectionDirty = !0;
    }
    setOptions(r) {
      this.options = {
        ...this.options,
        ...r,
        crossfade: r.crossfade !== void 0 ? r.crossfade : !0
      };
    }
    clearMeasurements() {
      this.scroll = void 0, this.layout = void 0, this.snapshot = void 0, this.prevTransformTemplateValue = void 0, this.targetDelta = void 0, this.target = void 0, this.isLayoutDirty = !1;
    }
    forceRelativeParentToResolveTarget() {
      this.relativeParent && this.relativeParent.resolvedRelativeTargetAt !== R.timestamp && this.relativeParent.resolveTargetDelta(!0);
    }
    resolveTargetDelta(r = !1) {
      var a;
      const l = this.getLead();
      this.isProjectionDirty || (this.isProjectionDirty = l.isProjectionDirty), this.isTransformDirty || (this.isTransformDirty = l.isTransformDirty), this.isSharedProjectionDirty || (this.isSharedProjectionDirty = l.isSharedProjectionDirty);
      const u = !!this.resumingFrom || this !== l;
      if (!(r || u && this.isSharedProjectionDirty || this.isProjectionDirty || !((a = this.parent) === null || a === void 0) && a.isProjectionDirty || this.attemptToResolveRelativeTarget || this.root.updateBlockedByResize))
        return;
      const { layout: h, layoutId: f } = this.options;
      if (!(!this.layout || !(h || f))) {
        if (this.resolvedRelativeTargetAt = R.timestamp, !this.targetDelta && !this.relativeTarget) {
          const d = this.getClosestProjectingParent();
          d && d.layout && this.animationProgress !== 1 ? (this.relativeParent = d, this.forceRelativeParentToResolveTarget(), this.relativeTarget = M(), this.relativeTargetOrigin = M(), wt(this.relativeTargetOrigin, this.layout.layoutBox, d.layout.layoutBox), N(this.relativeTarget, this.relativeTargetOrigin)) : this.relativeParent = this.relativeTarget = void 0;
        }
        if (!(!this.relativeTarget && !this.targetDelta) && (this.target || (this.target = M(), this.targetWithTransforms = M()), this.relativeTarget && this.relativeTargetOrigin && this.relativeParent && this.relativeParent.target ? (this.forceRelativeParentToResolveTarget(), Sl(this.target, this.relativeTarget, this.relativeParent.target)) : this.targetDelta ? (this.resumingFrom ? this.target = this.applyTransform(this.layout.layoutBox) : N(this.target, this.layout.layoutBox), qi(this.target, this.targetDelta)) : N(this.target, this.layout.layoutBox), this.attemptToResolveRelativeTarget)) {
          this.attemptToResolveRelativeTarget = !1;
          const d = this.getClosestProjectingParent();
          d && !!d.resumingFrom == !!this.resumingFrom && !d.options.layoutScroll && d.target && this.animationProgress !== 1 ? (this.relativeParent = d, this.forceRelativeParentToResolveTarget(), this.relativeTarget = M(), this.relativeTargetOrigin = M(), wt(this.relativeTargetOrigin, this.target, d.target), N(this.relativeTarget, this.relativeTargetOrigin)) : this.relativeParent = this.relativeTarget = void 0;
        }
      }
    }
    getClosestProjectingParent() {
      if (!(!this.parent || Ce(this.parent.latestValues) || Xi(this.parent.latestValues)))
        return this.parent.isProjecting() ? this.parent : this.parent.getClosestProjectingParent();
    }
    isProjecting() {
      return !!((this.relativeTarget || this.targetDelta || this.options.layoutRoot) && this.layout);
    }
    calcProjection() {
      var r;
      const a = this.getLead(), l = !!this.resumingFrom || this !== a;
      let u = !0;
      if ((this.isProjectionDirty || !((r = this.parent) === null || r === void 0) && r.isProjectionDirty) && (u = !1), l && (this.isSharedProjectionDirty || this.isTransformDirty) && (u = !1), this.resolvedRelativeTargetAt === R.timestamp && (u = !1), u)
        return;
      const { layout: c, layoutId: h } = this.options;
      if (this.isTreeAnimating = !!(this.parent && this.parent.isTreeAnimating || this.currentAnimation || this.pendingAnimation), this.isTreeAnimating || (this.targetDelta = this.relativeTarget = void 0), !this.layout || !(c || h))
        return;
      N(this.layoutCorrected, this.layout.layoutBox);
      const f = this.treeScale.x, d = this.treeScale.y;
      bl(this.layoutCorrected, this.treeScale, this.path, l), a.layout && !a.target && (this.treeScale.x !== 1 || this.treeScale.y !== 1) && (a.target = a.layout.layoutBox, a.targetWithTransforms = M());
      const { target: m } = a;
      if (!m) {
        this.prevProjectionDelta && (this.createProjectionDeltas(), this.scheduleRender());
        return;
      }
      !this.projectionDelta || !this.prevProjectionDelta ? this.createProjectionDeltas() : (fs(this.prevProjectionDelta.x, this.projectionDelta.x), fs(this.prevProjectionDelta.y, this.projectionDelta.y)), At(this.projectionDelta, this.layoutCorrected, m, this.latestValues), (this.treeScale.x !== f || this.treeScale.y !== d || !xs(this.projectionDelta.x, this.prevProjectionDelta.x) || !xs(this.projectionDelta.y, this.prevProjectionDelta.y)) && (this.hasProjected = !0, this.scheduleRender(), this.notifyListeners("projectionUpdate", m));
    }
    hide() {
      this.isVisible = !1;
    }
    show() {
      this.isVisible = !0;
    }
    scheduleRender(r = !0) {
      var a;
      if ((a = this.options.visualElement) === null || a === void 0 || a.scheduleRender(), r) {
        const l = this.getStack();
        l && l.scheduleRender();
      }
      this.resumingFrom && !this.resumingFrom.instance && (this.resumingFrom = void 0);
    }
    createProjectionDeltas() {
      this.prevProjectionDelta = lt(), this.projectionDelta = lt(), this.projectionDeltaWithTransform = lt();
    }
    setAnimationOrigin(r, a = !1) {
      const l = this.snapshot, u = l ? l.latestValues : {}, c = { ...this.latestValues }, h = lt();
      (!this.relativeParent || !this.relativeParent.options.layoutRoot) && (this.relativeTarget = this.relativeTargetOrigin = void 0), this.attemptToResolveRelativeTarget = !a;
      const f = M(), d = l ? l.source : void 0, m = this.layout ? this.layout.source : void 0, p = d !== m, y = this.getStack(), g = !y || y.members.length <= 1, v = !!(p && !g && this.options.crossfade === !0 && !this.path.some(yu));
      this.animationProgress = 0;
      let T;
      this.mixTargetDelta = (b) => {
        const P = b / 1e3;
        As(h.x, r.x, P), As(h.y, r.y, P), this.setTargetDelta(h), this.relativeTarget && this.relativeTargetOrigin && this.layout && this.relativeParent && this.relativeParent.layout && (wt(f, this.layout.layoutBox, this.relativeParent.layout.layoutBox), gu(this.relativeTarget, this.relativeTargetOrigin, f, P), T && tu(this.relativeTarget, T) && (this.isProjectionDirty = !1), T || (T = M()), N(T, this.relativeTarget)), p && (this.animationValues = c, Yl(c, u, this.latestValues, P, v, g)), this.root.scheduleUpdateProjection(), this.scheduleRender(), this.animationProgress = P;
      }, this.mixTargetDelta(this.options.layoutRoot ? 1e3 : 0);
    }
    startAnimation(r) {
      this.notifyListeners("animationStart"), this.currentAnimation && this.currentAnimation.stop(), this.resumingFrom && this.resumingFrom.currentAnimation && this.resumingFrom.currentAnimation.stop(), this.pendingAnimation && (q(this.pendingAnimation), this.pendingAnimation = void 0), this.pendingAnimation = w.update(() => {
        $t.hasAnimatedSinceResize = !0, this.currentAnimation = Kl(0, Ps, {
          ...r,
          onUpdate: (a) => {
            this.mixTargetDelta(a), r.onUpdate && r.onUpdate(a);
          },
          onStop: () => {
          },
          onComplete: () => {
            r.onComplete && r.onComplete(), this.completeAnimation();
          }
        }), this.resumingFrom && (this.resumingFrom.currentAnimation = this.currentAnimation), this.pendingAnimation = void 0;
      });
    }
    completeAnimation() {
      this.resumingFrom && (this.resumingFrom.currentAnimation = void 0, this.resumingFrom.preserveOpacity = void 0);
      const r = this.getStack();
      r && r.exitAnimationComplete(), this.resumingFrom = this.currentAnimation = this.animationValues = void 0, this.notifyListeners("animationComplete");
    }
    finishAnimation() {
      this.currentAnimation && (this.mixTargetDelta && this.mixTargetDelta(Ps), this.currentAnimation.stop()), this.completeAnimation();
    }
    applyTransformsToTarget() {
      const r = this.getLead();
      let { targetWithTransforms: a, target: l, layout: u, latestValues: c } = r;
      if (!(!a || !l || !u)) {
        if (this !== r && this.layout && u && lr(this.options.animationType, this.layout.layoutBox, u.layoutBox)) {
          l = this.target || M();
          const h = F(this.layout.layoutBox.x);
          l.x.min = r.target.x.min, l.x.max = l.x.min + h;
          const f = F(this.layout.layoutBox.y);
          l.y.min = r.target.y.min, l.y.max = l.y.min + f;
        }
        N(a, l), ct(a, c), At(this.projectionDeltaWithTransform, this.layoutCorrected, a, c);
      }
    }
    registerSharedNode(r, a) {
      this.sharedNodes.has(r) || this.sharedNodes.set(r, new eu()), this.sharedNodes.get(r).add(a);
      const u = a.options.initialPromotionConfig;
      a.promote({
        transition: u ? u.transition : void 0,
        preserveFollowOpacity: u && u.shouldPreserveFollowOpacity ? u.shouldPreserveFollowOpacity(a) : void 0
      });
    }
    isLead() {
      const r = this.getStack();
      return r ? r.lead === this : !0;
    }
    getLead() {
      var r;
      const { layoutId: a } = this.options;
      return a ? ((r = this.getStack()) === null || r === void 0 ? void 0 : r.lead) || this : this;
    }
    getPrevLead() {
      var r;
      const { layoutId: a } = this.options;
      return a ? (r = this.getStack()) === null || r === void 0 ? void 0 : r.prevLead : void 0;
    }
    getStack() {
      const { layoutId: r } = this.options;
      if (r)
        return this.root.sharedNodes.get(r);
    }
    promote({ needsReset: r, transition: a, preserveFollowOpacity: l } = {}) {
      const u = this.getStack();
      u && u.promote(this, l), r && (this.projectionDelta = void 0, this.needsReset = !0), a && this.setOptions({ transition: a });
    }
    relegate() {
      const r = this.getStack();
      return r ? r.relegate(this) : !1;
    }
    resetSkewAndRotation() {
      const { visualElement: r } = this.options;
      if (!r)
        return;
      let a = !1;
      const { latestValues: l } = r;
      if ((l.z || l.rotate || l.rotateX || l.rotateY || l.rotateZ || l.skewX || l.skewY) && (a = !0), !a)
        return;
      const u = {};
      l.z && he("z", r, u, this.animationValues);
      for (let c = 0; c < ce.length; c++)
        he(`rotate${ce[c]}`, r, u, this.animationValues), he(`skew${ce[c]}`, r, u, this.animationValues);
      r.render();
      for (const c in u)
        r.setStaticValue(c, u[c]), this.animationValues && (this.animationValues[c] = u[c]);
      r.scheduleRender();
    }
    getProjectionStyles(r) {
      var a, l;
      if (!this.instance || this.isSVG)
        return;
      if (!this.isVisible)
        return su;
      const u = {
        visibility: ""
      }, c = this.getTransformTemplate();
      if (this.needsReset)
        return this.needsReset = !1, u.opacity = "", u.pointerEvents = _t(r == null ? void 0 : r.pointerEvents) || "", u.transform = c ? c(this.latestValues, "") : "none", u;
      const h = this.getLead();
      if (!this.projectionDelta || !this.layout || !h.target) {
        const p = {};
        return this.options.layoutId && (p.opacity = this.latestValues.opacity !== void 0 ? this.latestValues.opacity : 1, p.pointerEvents = _t(r == null ? void 0 : r.pointerEvents) || ""), this.hasProjected && !tt(this.latestValues) && (p.transform = c ? c({}, "") : "none", this.hasProjected = !1), p;
      }
      const f = h.animationValues || h.latestValues;
      this.applyTransformsToTarget(), u.transform = nu(this.projectionDeltaWithTransform, this.treeScale, f), c && (u.transform = c(f, u.transform));
      const { x: d, y: m } = this.projectionDelta;
      u.transformOrigin = `${d.origin * 100}% ${m.origin * 100}% 0`, h.animationValues ? u.opacity = h === this ? (l = (a = f.opacity) !== null && a !== void 0 ? a : this.latestValues.opacity) !== null && l !== void 0 ? l : 1 : this.preserveOpacity ? this.latestValues.opacity : f.opacityExit : u.opacity = h === this ? f.opacity !== void 0 ? f.opacity : "" : f.opacityExit !== void 0 ? f.opacityExit : 0;
      for (const p in Dt) {
        if (f[p] === void 0)
          continue;
        const { correct: y, applyTo: g, isCSSVariable: v } = Dt[p], T = u.transform === "none" ? f[p] : y(f[p], h);
        if (g) {
          const b = g.length;
          for (let P = 0; P < b; P++)
            u[g[P]] = T;
        } else
          v ? this.options.visualElement.renderState.vars[p] = T : u[p] = T;
      }
      return this.options.layoutId && (u.pointerEvents = h === this ? _t(r == null ? void 0 : r.pointerEvents) || "" : "none"), u;
    }
    clearSnapshot() {
      this.resumeFrom = this.snapshot = void 0;
    }
    // Only run on root
    resetTree() {
      this.root.nodes.forEach((r) => {
        var a;
        return (a = r.currentAnimation) === null || a === void 0 ? void 0 : a.stop();
      }), this.root.nodes.forEach(Ss), this.root.sharedNodes.clear();
    }
  };
}
function ru(t) {
  t.updateLayout();
}
function ou(t) {
  var e;
  const n = ((e = t.resumeFrom) === null || e === void 0 ? void 0 : e.snapshot) || t.snapshot;
  if (t.isLead() && t.layout && n && t.hasListeners("didUpdate")) {
    const { layoutBox: s, measuredBox: i } = t.layout, { animationType: o } = t.options, r = n.source !== t.layout.source;
    o === "size" ? U((h) => {
      const f = r ? n.measuredBox[h] : n.layoutBox[h], d = F(f);
      f.min = s[h].min, f.max = f.min + d;
    }) : lr(o, n.layoutBox, s) && U((h) => {
      const f = r ? n.measuredBox[h] : n.layoutBox[h], d = F(s[h]);
      f.max = f.min + d, t.relativeTarget && !t.currentAnimation && (t.isProjectionDirty = !0, t.relativeTarget[h].max = t.relativeTarget[h].min + d);
    });
    const a = lt();
    At(a, s, n.layoutBox);
    const l = lt();
    r ? At(l, t.applyTransform(i, !0), n.measuredBox) : At(l, s, n.layoutBox);
    const u = !ir(a);
    let c = !1;
    if (!t.resumeFrom) {
      const h = t.getClosestProjectingParent();
      if (h && !h.resumeFrom) {
        const { snapshot: f, layout: d } = h;
        if (f && d) {
          const m = M();
          wt(m, n.layoutBox, f.layoutBox);
          const p = M();
          wt(p, s, d.layoutBox), rr(m, p) || (c = !0), h.options.layoutRoot && (t.relativeTarget = p, t.relativeTargetOrigin = m, t.relativeParent = h);
        }
      }
    }
    t.notifyListeners("didUpdate", {
      layout: s,
      snapshot: n,
      delta: l,
      layoutDelta: a,
      hasLayoutChanged: u,
      hasRelativeLayoutChanged: c
    });
  } else if (t.isLead()) {
    const { onExitComplete: s } = t.options;
    s && s();
  }
  t.options.transition = void 0;
}
function au(t) {
  t.parent && (t.isProjecting() || (t.isProjectionDirty = t.parent.isProjectionDirty), t.isSharedProjectionDirty || (t.isSharedProjectionDirty = !!(t.isProjectionDirty || t.parent.isProjectionDirty || t.parent.isSharedProjectionDirty)), t.isTransformDirty || (t.isTransformDirty = t.parent.isTransformDirty));
}
function lu(t) {
  t.isProjectionDirty = t.isSharedProjectionDirty = t.isTransformDirty = !1;
}
function uu(t) {
  t.clearSnapshot();
}
function Ss(t) {
  t.clearMeasurements();
}
function cu(t) {
  t.isLayoutDirty = !1;
}
function hu(t) {
  const { visualElement: e } = t.options;
  e && e.getProps().onBeforeLayoutMeasure && e.notify("BeforeLayoutMeasure"), t.resetTransform();
}
function bs(t) {
  t.finishAnimation(), t.targetDelta = t.relativeTarget = t.target = void 0, t.isProjectionDirty = !0;
}
function fu(t) {
  t.resolveTargetDelta();
}
function du(t) {
  t.calcProjection();
}
function mu(t) {
  t.resetSkewAndRotation();
}
function pu(t) {
  t.removeLeadSnapshot();
}
function As(t, e, n) {
  t.translate = D(e.translate, 0, n), t.scale = D(e.scale, 1, n), t.origin = e.origin, t.originPoint = e.originPoint;
}
function ws(t, e, n, s) {
  t.min = D(e.min, n.min, s), t.max = D(e.max, n.max, s);
}
function gu(t, e, n, s) {
  ws(t.x, e.x, n.x, s), ws(t.y, e.y, n.y, s);
}
function yu(t) {
  return t.animationValues && t.animationValues.opacityExit !== void 0;
}
const vu = {
  duration: 0.45,
  ease: [0.4, 0, 0.1, 1]
}, Vs = (t) => typeof navigator < "u" && navigator.userAgent && navigator.userAgent.toLowerCase().includes(t), Ds = Vs("applewebkit/") && !Vs("chrome/") ? Math.round : I;
function Ms(t) {
  t.min = Ds(t.min), t.max = Ds(t.max);
}
function Tu(t) {
  Ms(t.x), Ms(t.y);
}
function lr(t, e, n) {
  return t === "position" || t === "preserve-aspect" && !Pl(Ts(e), Ts(n), 0.2);
}
function xu(t) {
  var e;
  return t !== t.root && ((e = t.scroll) === null || e === void 0 ? void 0 : e.wasRoot);
}
const Pu = ar({
  attachResizeListener: (t, e) => Lt(t, "resize", e),
  measureScroll: () => ({
    x: document.documentElement.scrollLeft || document.body.scrollLeft,
    y: document.documentElement.scrollTop || document.body.scrollTop
  }),
  checkIsScrollRoot: () => !0
}), fe = {
  current: void 0
}, ur = ar({
  measureScroll: (t) => ({
    x: t.scrollLeft,
    y: t.scrollTop
  }),
  defaultParent: () => {
    if (!fe.current) {
      const t = new Pu({});
      t.mount(window), t.setOptions({ layoutScroll: !0 }), fe.current = t;
    }
    return fe.current;
  },
  resetTransform: (t, e) => {
    t.style.transform = e !== void 0 ? e : "none";
  },
  checkIsScrollRoot: (t) => window.getComputedStyle(t).position === "fixed"
}), Su = {
  pan: {
    Feature: jl
  },
  drag: {
    Feature: Il,
    ProjectionNode: ur,
    MeasureLayout: er
  }
};
function bu(t, e, n) {
  var s;
  if (t instanceof EventTarget)
    return [t];
  if (typeof t == "string") {
    let i = document;
    const o = (s = void 0) !== null && s !== void 0 ? s : i.querySelectorAll(t);
    return o ? Array.from(o) : [];
  }
  return Array.from(t);
}
function cr(t, e) {
  const n = bu(t), s = new AbortController(), i = {
    passive: !0,
    ...e,
    signal: s.signal
  };
  return [n, i, () => s.abort()];
}
function Cs(t) {
  return !(t.pointerType === "touch" || Gi());
}
function Au(t, e, n = {}) {
  const [s, i, o] = cr(t, n), r = (a) => {
    if (!Cs(a))
      return;
    const { target: l } = a, u = e(l, a);
    if (typeof u != "function" || !l)
      return;
    const c = (h) => {
      Cs(h) && (u(h), l.removeEventListener("pointerleave", c));
    };
    l.addEventListener("pointerleave", c, i);
  };
  return s.forEach((a) => {
    a.addEventListener("pointerenter", r, i);
  }), o;
}
function Rs(t, e, n) {
  const { props: s } = t;
  t.animationState && s.whileHover && t.animationState.setActive("whileHover", n === "Start");
  const i = "onHover" + n, o = s[i];
  o && w.postRender(() => o(e, Ot(e)));
}
class wu extends J {
  mount() {
    const { current: e } = this.node;
    e && (this.unmount = Au(e, (n, s) => (Rs(this.node, s, "Start"), (i) => Rs(this.node, i, "End"))));
  }
  unmount() {
  }
}
class Vu extends J {
  constructor() {
    super(...arguments), this.isActive = !1;
  }
  onFocus() {
    let e = !1;
    try {
      e = this.node.current.matches(":focus-visible");
    } catch {
      e = !0;
    }
    !e || !this.node.animationState || (this.node.animationState.setActive("whileFocus", !0), this.isActive = !0);
  }
  onBlur() {
    !this.isActive || !this.node.animationState || (this.node.animationState.setActive("whileFocus", !1), this.isActive = !1);
  }
  mount() {
    this.unmount = kt(Lt(this.node.current, "focus", () => this.onFocus()), Lt(this.node.current, "blur", () => this.onBlur()));
  }
  unmount() {
  }
}
function Es(t, e) {
  const n = `${e}PointerCapture`;
  if (t.target instanceof Element && n in t.target && t.pointerId !== void 0)
    try {
      t.target[n](t.pointerId);
    } catch {
    }
}
const hr = (t, e) => e ? t === e ? !0 : hr(t, e.parentElement) : !1, Du = /* @__PURE__ */ new Set([
  "BUTTON",
  "INPUT",
  "SELECT",
  "TEXTAREA",
  "A"
]);
function Mu(t) {
  return Du.has(t.tagName) || t.tabIndex !== -1;
}
const xt = /* @__PURE__ */ new WeakSet();
function Ls(t) {
  return (e) => {
    e.key === "Enter" && t(e);
  };
}
function de(t, e) {
  t.dispatchEvent(new PointerEvent("pointer" + e, { isPrimary: !0, bubbles: !0 }));
}
const Cu = (t, e) => {
  const n = t.currentTarget;
  if (!n)
    return;
  const s = Ls(() => {
    if (xt.has(n))
      return;
    de(n, "down");
    const i = Ls(() => {
      de(n, "up");
    }), o = () => de(n, "cancel");
    n.addEventListener("keyup", i, e), n.addEventListener("blur", o, e);
  });
  n.addEventListener("keydown", s, e), n.addEventListener("blur", () => n.removeEventListener("keydown", s), e);
};
function Bs(t) {
  return dn(t) && !Gi();
}
function Ru(t, e, n = {}) {
  const [s, i, o] = cr(t, n), r = (a) => {
    const l = a.currentTarget;
    if (!l || !Bs(a) || xt.has(l))
      return;
    xt.add(l), Es(a, "set");
    const u = e(l, a), c = (d, m) => {
      l.removeEventListener("pointerup", h), l.removeEventListener("pointercancel", f), Es(d, "release"), !(!Bs(d) || !xt.has(l)) && (xt.delete(l), typeof u == "function" && u(d, { success: m }));
    }, h = (d) => {
      (d.isTrusted ? Eu(d, l instanceof Element ? l.getBoundingClientRect() : {
        left: 0,
        top: 0,
        right: window.innerWidth,
        bottom: window.innerHeight
      }) : !1) ? c(d, !1) : c(d, !(l instanceof Element) || hr(l, d.target));
    }, f = (d) => {
      c(d, !1);
    };
    l.addEventListener("pointerup", h, i), l.addEventListener("pointercancel", f, i), l.addEventListener("lostpointercapture", f, i);
  };
  return s.forEach((a) => {
    a = n.useGlobalTarget ? window : a;
    let l = !1;
    a instanceof HTMLElement && (l = !0, !Mu(a) && a.getAttribute("tabindex") === null && (a.tabIndex = 0)), a.addEventListener("pointerdown", r, i), l && a.addEventListener("focus", (u) => Cu(u, i), i);
  }), o;
}
function Eu(t, e) {
  return t.clientX < e.left || t.clientX > e.right || t.clientY < e.top || t.clientY > e.bottom;
}
function Fs(t, e, n) {
  const { props: s } = t;
  if (t.current instanceof HTMLButtonElement && t.current.disabled)
    return;
  t.animationState && s.whileTap && t.animationState.setActive("whileTap", n === "Start");
  const i = "onTap" + (n === "End" ? "" : n), o = s[i];
  o && w.postRender(() => o(e, Ot(e)));
}
class Lu extends J {
  mount() {
    const { current: e } = this.node;
    e && (this.unmount = Ru(e, (n, s) => (Fs(this.node, s, "Start"), (i, { success: o }) => Fs(this.node, i, o ? "End" : "Cancel")), { useGlobalTarget: this.node.props.globalTapTarget }));
  }
  unmount() {
  }
}
const Le = /* @__PURE__ */ new WeakMap(), me = /* @__PURE__ */ new WeakMap(), Bu = (t) => {
  const e = Le.get(t.target);
  e && e(t);
}, Fu = (t) => {
  t.forEach(Bu);
};
function ku({ root: t, ...e }) {
  const n = t || document;
  me.has(n) || me.set(n, {});
  const s = me.get(n), i = JSON.stringify(e);
  return s[i] || (s[i] = new IntersectionObserver(Fu, { root: t, ...e })), s[i];
}
function Ou(t, e, n) {
  const s = ku(e);
  return Le.set(t, n), s.observe(t), () => {
    Le.delete(t), s.unobserve(t);
  };
}
const Iu = {
  some: 0,
  all: 1
};
class ju extends J {
  constructor() {
    super(...arguments), this.hasEnteredView = !1, this.isInView = !1;
  }
  startObserver() {
    this.unmount();
    const { viewport: e = {} } = this.node.getProps(), { root: n, margin: s, amount: i = "some", once: o } = e, r = {
      root: n ? n.current : void 0,
      rootMargin: s,
      threshold: typeof i == "number" ? i : Iu[i]
    }, a = (l) => {
      const { isIntersecting: u } = l;
      if (this.isInView === u || (this.isInView = u, o && !u && this.hasEnteredView))
        return;
      u && (this.hasEnteredView = !0), this.node.animationState && this.node.animationState.setActive("whileInView", u);
      const { onViewportEnter: c, onViewportLeave: h } = this.node.getProps(), f = u ? c : h;
      f && f(l);
    };
    return Ou(this.node.current, r, a);
  }
  mount() {
    this.startObserver();
  }
  update() {
    if (typeof IntersectionObserver > "u")
      return;
    const { props: e, prevProps: n } = this.node;
    ["amount", "margin", "root"].some(Nu(e, n)) && this.startObserver();
  }
  unmount() {
  }
}
function Nu({ viewport: t = {} }, { viewport: e = {} } = {}) {
  return (n) => t[n] !== e[n];
}
const Uu = {
  inView: {
    Feature: ju
  },
  tap: {
    Feature: Lu
  },
  focus: {
    Feature: Vu
  },
  hover: {
    Feature: wu
  }
}, _u = {
  layout: {
    ProjectionNode: ur,
    MeasureLayout: er
  }
}, Be = { current: null }, fr = { current: !1 };
function Ku() {
  if (fr.current = !0, !!ke)
    if (window.matchMedia) {
      const t = window.matchMedia("(prefers-reduced-motion)"), e = () => Be.current = t.matches;
      t.addListener(e), e();
    } else
      Be.current = !1;
}
const $u = [...Li, E, Z], Wu = (t) => $u.find(Ei(t)), Gu = /* @__PURE__ */ new WeakMap();
function zu(t, e, n) {
  for (const s in e) {
    const i = e[s], o = n[s];
    if (L(i))
      t.addValue(s, i), process.env.NODE_ENV === "development" && qt(i.version === "12.5.0", `Attempting to mix Motion versions ${i.version} with 12.5.0 may not work as expected.`);
    else if (L(o))
      t.addValue(s, Rt(i, { owner: t }));
    else if (o !== i)
      if (t.hasValue(s)) {
        const r = t.getValue(s);
        r.liveStyle === !0 ? r.jump(i) : r.hasAnimated || r.set(i);
      } else {
        const r = t.getStaticValue(s);
        t.addValue(s, Rt(r !== void 0 ? r : i, { owner: t }));
      }
  }
  for (const s in n)
    e[s] === void 0 && t.removeValue(s);
  return e;
}
const ks = [
  "AnimationStart",
  "AnimationComplete",
  "Update",
  "BeforeLayoutMeasure",
  "LayoutMeasure",
  "LayoutAnimationStart",
  "LayoutAnimationComplete"
];
class Hu {
  /**
   * This method takes React props and returns found MotionValues. For example, HTML
   * MotionValues will be found within the style prop, whereas for Three.js within attribute arrays.
   *
   * This isn't an abstract method as it needs calling in the constructor, but it is
   * intended to be one.
   */
  scrapeMotionValuesFromProps(e, n, s) {
    return {};
  }
  constructor({ parent: e, props: n, presenceContext: s, reducedMotionConfig: i, blockInitialAnimation: o, visualState: r }, a = {}) {
    this.current = null, this.children = /* @__PURE__ */ new Set(), this.isVariantNode = !1, this.isControllingVariants = !1, this.shouldReduceMotion = null, this.values = /* @__PURE__ */ new Map(), this.KeyframeResolver = un, this.features = {}, this.valueSubscriptions = /* @__PURE__ */ new Map(), this.prevMotionValues = {}, this.events = {}, this.propEventSubscriptions = {}, this.notifyUpdate = () => this.notify("Update", this.latestValues), this.render = () => {
      this.current && (this.triggerBuild(), this.renderInstance(this.current, this.renderState, this.props.style, this.projection));
    }, this.renderScheduledAt = 0, this.scheduleRender = () => {
      const d = $.now();
      this.renderScheduledAt < d && (this.renderScheduledAt = d, w.render(this.render, !1, !0));
    };
    const { latestValues: l, renderState: u, onUpdate: c } = r;
    this.onUpdate = c, this.latestValues = l, this.baseTarget = { ...l }, this.initialValues = n.initial ? { ...l } : {}, this.renderState = u, this.parent = e, this.props = n, this.presenceContext = s, this.depth = e ? e.depth + 1 : 0, this.reducedMotionConfig = i, this.options = a, this.blockInitialAnimation = !!o, this.isControllingVariants = Qt(n), this.isVariantNode = Ws(n), this.isVariantNode && (this.variantChildren = /* @__PURE__ */ new Set()), this.manuallyAnimateOnMount = !!(e && e.current);
    const { willChange: h, ...f } = this.scrapeMotionValuesFromProps(n, {}, this);
    for (const d in f) {
      const m = f[d];
      l[d] !== void 0 && L(m) && m.set(l[d], !1);
    }
  }
  mount(e) {
    this.current = e, Gu.set(e, this), this.projection && !this.projection.instance && this.projection.mount(e), this.parent && this.isVariantNode && !this.isControllingVariants && (this.removeFromVariantTree = this.parent.addVariantChild(this)), this.values.forEach((n, s) => this.bindToMotionValue(s, n)), fr.current || Ku(), this.shouldReduceMotion = this.reducedMotionConfig === "never" ? !1 : this.reducedMotionConfig === "always" ? !0 : Be.current, process.env.NODE_ENV !== "production" && qt(this.shouldReduceMotion !== !0, "You have Reduced Motion enabled on your device. Animations may not appear as expected."), this.parent && this.parent.children.add(this), this.update(this.props, this.presenceContext);
  }
  unmount() {
    this.projection && this.projection.unmount(), q(this.notifyUpdate), q(this.render), this.valueSubscriptions.forEach((e) => e()), this.valueSubscriptions.clear(), this.removeFromVariantTree && this.removeFromVariantTree(), this.parent && this.parent.children.delete(this);
    for (const e in this.events)
      this.events[e].clear();
    for (const e in this.features) {
      const n = this.features[e];
      n && (n.unmount(), n.isMounted = !1);
    }
    this.current = null;
  }
  bindToMotionValue(e, n) {
    this.valueSubscriptions.has(e) && this.valueSubscriptions.get(e)();
    const s = it.has(e);
    s && this.onBindTransform && this.onBindTransform();
    const i = n.on("change", (a) => {
      this.latestValues[e] = a, this.props.onUpdate && w.preRender(this.notifyUpdate), s && this.projection && (this.projection.isTransformDirty = !0);
    }), o = n.on("renderRequest", this.scheduleRender);
    let r;
    window.MotionCheckAppearSync && (r = window.MotionCheckAppearSync(this, e, n)), this.valueSubscriptions.set(e, () => {
      i(), o(), r && r(), n.owner && n.stop();
    });
  }
  sortNodePosition(e) {
    return !this.current || !this.sortInstanceNodePosition || this.type !== e.type ? 0 : this.sortInstanceNodePosition(this.current, e.current);
  }
  updateFeatures() {
    let e = "animation";
    for (e in ht) {
      const n = ht[e];
      if (!n)
        continue;
      const { isEnabled: s, Feature: i } = n;
      if (!this.features[e] && i && s(this.props) && (this.features[e] = new i(this)), this.features[e]) {
        const o = this.features[e];
        o.isMounted ? o.update() : (o.mount(), o.isMounted = !0);
      }
    }
  }
  triggerBuild() {
    this.build(this.renderState, this.latestValues, this.props);
  }
  /**
   * Measure the current viewport box with or without transforms.
   * Only measures axis-aligned boxes, rotate and skew must be manually
   * removed with a re-render to work.
   */
  measureViewportBox() {
    return this.current ? this.measureInstanceViewportBox(this.current, this.props) : M();
  }
  getStaticValue(e) {
    return this.latestValues[e];
  }
  setStaticValue(e, n) {
    this.latestValues[e] = n;
  }
  /**
   * Update the provided props. Ensure any newly-added motion values are
   * added to our map, old ones removed, and listeners updated.
   */
  update(e, n) {
    (e.transformTemplate || this.props.transformTemplate) && this.scheduleRender(), this.prevProps = this.props, this.props = e, this.prevPresenceContext = this.presenceContext, this.presenceContext = n;
    for (let s = 0; s < ks.length; s++) {
      const i = ks[s];
      this.propEventSubscriptions[i] && (this.propEventSubscriptions[i](), delete this.propEventSubscriptions[i]);
      const o = "on" + i, r = e[o];
      r && (this.propEventSubscriptions[i] = this.on(i, r));
    }
    this.prevMotionValues = zu(this, this.scrapeMotionValuesFromProps(e, this.prevProps, this), this.prevMotionValues), this.handleChildMotionValue && this.handleChildMotionValue(), this.onUpdate && this.onUpdate(this);
  }
  getProps() {
    return this.props;
  }
  /**
   * Returns the variant definition with a given name.
   */
  getVariant(e) {
    return this.props.variants ? this.props.variants[e] : void 0;
  }
  /**
   * Returns the defined default transition on this component.
   */
  getDefaultTransition() {
    return this.props.transition;
  }
  getTransformPagePoint() {
    return this.props.transformPagePoint;
  }
  getClosestVariantNode() {
    return this.isVariantNode ? this : this.parent ? this.parent.getClosestVariantNode() : void 0;
  }
  /**
   * Add a child visual element to our set of children.
   */
  addVariantChild(e) {
    const n = this.getClosestVariantNode();
    if (n)
      return n.variantChildren && n.variantChildren.add(e), () => n.variantChildren.delete(e);
  }
  /**
   * Add a motion value and bind it to this visual element.
   */
  addValue(e, n) {
    const s = this.values.get(e);
    n !== s && (s && this.removeValue(e), this.bindToMotionValue(e, n), this.values.set(e, n), this.latestValues[e] = n.get());
  }
  /**
   * Remove a motion value and unbind any active subscriptions.
   */
  removeValue(e) {
    this.values.delete(e);
    const n = this.valueSubscriptions.get(e);
    n && (n(), this.valueSubscriptions.delete(e)), delete this.latestValues[e], this.removeValueFromRenderState(e, this.renderState);
  }
  /**
   * Check whether we have a motion value for this key
   */
  hasValue(e) {
    return this.values.has(e);
  }
  getValue(e, n) {
    if (this.props.values && this.props.values[e])
      return this.props.values[e];
    let s = this.values.get(e);
    return s === void 0 && n !== void 0 && (s = Rt(n === null ? void 0 : n, { owner: this }), this.addValue(e, s)), s;
  }
  /**
   * If we're trying to animate to a previously unencountered value,
   * we need to check for it in our state and as a last resort read it
   * directly from the instance (which might have performance implications).
   */
  readValue(e, n) {
    var s;
    let i = this.latestValues[e] !== void 0 || !this.current ? this.latestValues[e] : (s = this.getBaseTargetFromProps(this.props, e)) !== null && s !== void 0 ? s : this.readValueFromInstance(this.current, e, this.options);
    return i != null && (typeof i == "string" && (Ci(i) || Ti(i)) ? i = parseFloat(i) : !Wu(i) && Z.test(n) && (i = Vi(e, n)), this.setBaseTarget(e, L(i) ? i.get() : i)), L(i) ? i.get() : i;
  }
  /**
   * Set the base target to later animate back to. This is currently
   * only hydrated on creation and when we first read a value.
   */
  setBaseTarget(e, n) {
    this.baseTarget[e] = n;
  }
  /**
   * Find the base target for a value thats been removed from all animation
   * props.
   */
  getBaseTarget(e) {
    var n;
    const { initial: s } = this.props;
    let i;
    if (typeof s == "string" || typeof s == "object") {
      const r = Xe(this.props, s, (n = this.presenceContext) === null || n === void 0 ? void 0 : n.custom);
      r && (i = r[e]);
    }
    if (s && i !== void 0)
      return i;
    const o = this.getBaseTargetFromProps(this.props, e);
    return o !== void 0 && !L(o) ? o : this.initialValues[e] !== void 0 && i === void 0 ? void 0 : this.baseTarget[e];
  }
  on(e, n) {
    return this.events[e] || (this.events[e] = new tn()), this.events[e].add(n);
  }
  notify(e, ...n) {
    this.events[e] && this.events[e].notify(...n);
  }
}
class dr extends Hu {
  constructor() {
    super(...arguments), this.KeyframeResolver = Bi;
  }
  sortInstanceNodePosition(e, n) {
    return e.compareDocumentPosition(n) & 2 ? 1 : -1;
  }
  getBaseTargetFromProps(e, n) {
    return e.style ? e.style[n] : void 0;
  }
  removeValueFromRenderState(e, { vars: n, style: s }) {
    delete n[e], delete s[e];
  }
  handleChildMotionValue() {
    this.childSubscription && (this.childSubscription(), delete this.childSubscription);
    const { children: e } = this.props;
    L(e) && (this.childSubscription = e.on("change", (n) => {
      this.current && (this.current.textContent = `${n}`);
    }));
  }
}
function Yu(t) {
  return window.getComputedStyle(t);
}
class Xu extends dr {
  constructor() {
    super(...arguments), this.type = "html", this.renderInstance = ei;
  }
  readValueFromInstance(e, n) {
    if (it.has(n))
      return Jo(e, n);
    {
      const s = Yu(e), i = (_e(n) ? s.getPropertyValue(n) : s[n]) || 0;
      return typeof i == "string" ? i.trim() : i;
    }
  }
  measureInstanceViewportBox(e, { transformPagePoint: n }) {
    return Zi(e, n);
  }
  build(e, n, s) {
    We(e, n, s.transformTemplate);
  }
  scrapeMotionValuesFromProps(e, n, s) {
    return qe(e, n, s);
  }
}
class qu extends dr {
  constructor() {
    super(...arguments), this.type = "svg", this.isSVGTag = !1, this.measureInstanceViewportBox = M, this.updateDimensions = () => {
      this.current && !this.renderState.dimensions && ti(this.current, this.renderState);
    };
  }
  getBaseTargetFromProps(e, n) {
    return e[n];
  }
  readValueFromInstance(e, n) {
    if (it.has(n)) {
      const s = wi(n);
      return s && s.default || 0;
    }
    return n = ni.has(n) ? n : Ue(n), e.getAttribute(n);
  }
  scrapeMotionValuesFromProps(e, n, s) {
    return ii(e, n, s);
  }
  onBindTransform() {
    this.current && !this.renderState.dimensions && w.postRender(this.updateDimensions);
  }
  build(e, n, s) {
    He(e, n, this.isSVGTag, s.transformTemplate);
  }
  renderInstance(e, n, s, i) {
    si(e, n, s, i);
  }
  mount(e) {
    this.isSVGTag = Ye(e.tagName), super.mount(e);
  }
}
const Zu = (t, e) => ze(t) ? new qu(e) : new Xu(e, {
  allowProjection: t !== js
}), Ju = /* @__PURE__ */ yo({
  ...fl,
  ...Uu,
  ...Su,
  ..._u
}, Zu), ic = /* @__PURE__ */ Lr(Ju);
export {
  Ns as L,
  Us as M,
  Oe as P,
  Ar as a,
  br as b,
  ic as m,
  Sr as u
};
