// tailgrid/dist/components/ui/avatar.js
import { jsx as l } from "react/jsx-runtime";
import * as i from "react";
import { c as C, P as c, u as x, a as g } from "../../index-CDSAOBSC.js";
import { c as f } from "../../utils-B9KdtH6G.js";
var v = "Avatar", [N, P] = C(v), [R, A] = N(v), p = i.forwardRef(
  (a, e) => {
    const { __scopeAvatar: s, ...t } = a, [o, r] = i.useState("idle");
    return /* @__PURE__ */ l(
      R,
      {
        scope: s,
        imageLoadingStatus: o,
        onImageLoadingStatusChange: r,
        children: /* @__PURE__ */ l(c.span, { ...t, ref: e })
      }
    );
  }
);
p.displayName = v;
var S = "AvatarImage", L = i.forwardRef(
  (a, e) => {
    const { __scopeAvatar: s, src: t, onLoadingStatusChange: o = () => {
    }, ...r } = a, u = A(S, s), n = _(t, r.referrerPolicy), d = x((m) => {
      o(m), u.onImageLoadingStatusChange(m);
    });
    return g(() => {
      n !== "idle" && d(n);
    }, [n, d]), n === "loaded" ? /* @__PURE__ */ l(c.img, { ...r, ref: e, src: t }) : null;
  }
);
L.displayName = S;
var w = "AvatarFallback", I = i.forwardRef(
  (a, e) => {
    const { __scopeAvatar: s, delayMs: t, ...o } = a, r = A(w, s), [u, n] = i.useState(t === void 0);
    return i.useEffect(() => {
      if (t !== void 0) {
        const d = window.setTimeout(() => n(!0), t);
        return () => window.clearTimeout(d);
      }
    }, [t]), u && r.imageLoadingStatus !== "loaded" ? /* @__PURE__ */ l(c.span, { ...o, ref: e }) : null;
  }
);
I.displayName = w;
function _(a, e) {
  const [s, t] = i.useState("idle");
  return g(() => {
    if (!a) {
      t("error");
      return;
    }
    let o = !0;
    const r = new window.Image(), u = (n) => () => {
      o && t(n);
    };
    return t("loading"), r.onload = u("loaded"), r.onerror = u("error"), r.src = a, e && (r.referrerPolicy = e), () => {
      o = !1;
    };
  }, [a, e]), s;
}
var b = p, h = L, k = I;
function F({
  className: a,
  ...e
}) {
  return /* @__PURE__ */ l(
    b,
    {
      "data-slot": "avatar",
      className: f(
        "relative flex size-8 shrink-0 overflow-hidden rounded-full",
        a
      ),
      ...e
    }
  );
}
function z({
  className: a,
  ...e
}) {
  return /* @__PURE__ */ l(
    h,
    {
      "data-slot": "avatar-image",
      className: f("aspect-square size-full", a),
      ...e
    }
  );
}
function T({
  className: a,
  ...e
}) {
  return /* @__PURE__ */ l(
    k,
    {
      "data-slot": "avatar-fallback",
      className: f(
        "bg-muted flex size-full items-center justify-center rounded-full",
        a
      ),
      ...e
    }
  );
}
export {
  F as Avatar,
  T as AvatarFallback,
  z as AvatarImage
};
