// tailgrid/dist/components/ui/dropdown-menu.js
import { jsx as r, jsxs as oe } from "react/jsx-runtime";
import * as c from "react";
import { c as g, u as j, a as Ce } from "../../index-WxTJUbsU.js";
import { u as F, S as xn, c as Re } from "../../index-nyDQUcyF.js";
import { c as be, u as re, P as T, d as Dn } from "../../index-CDSAOBSC.js";
import { c as Sn, u as In } from "../../index-BOS8297p.js";
import { c as xe, R as De, A as Pn, P as yn, D as En, C as Nn, a as Tn } from "../../index-Uwi29ejC.js";
import { h as An, u as On, R as kn, F as Ln, C as Gn } from "../../check-Cr0B_rVR.js";
import { P as Z } from "../../index-C2uZYgGF.js";
import { c as Se, I as Fn, R as Kn } from "../../index-DgiLjZTI.js";
import { c as x } from "../../utils-B9KdtH6G.js";
import { c as $n } from "../../createLucideIcon-DUBSZsqM.js";
import { C as Un } from "../../chevron-right-CO-eg3PL.js";
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const zn = [["circle", { cx: "12", cy: "12", r: "10", key: "1mglay" }]], Bn = $n("Circle", zn);
var te = ["Enter", " "], Vn = ["ArrowDown", "PageUp", "Home"], Ie = ["ArrowUp", "PageDown", "End"], Xn = [...Vn, ...Ie], Yn = {
  ltr: [...te, "ArrowRight"],
  rtl: [...te, "ArrowLeft"]
}, jn = {
  ltr: ["ArrowLeft"],
  rtl: ["ArrowRight"]
}, K = "Menu", [L, Hn, Wn] = Sn(K), [P, Pe] = be(K, [
  Wn,
  xe,
  Se
]), $ = xe(), ye = Se(), [Ee, S] = P(K), [Zn, U] = P(K), Ne = (e) => {
  const { __scopeMenu: t, open: n = !1, children: o, dir: a, onOpenChange: s, modal: i = !0 } = e, d = $(t), [f, v] = c.useState(null), p = c.useRef(!1), u = re(s), m = In(a);
  return c.useEffect(() => {
    const h = () => {
      p.current = !0, document.addEventListener("pointerdown", M, { capture: !0, once: !0 }), document.addEventListener("pointermove", M, { capture: !0, once: !0 });
    }, M = () => p.current = !1;
    return document.addEventListener("keydown", h, { capture: !0 }), () => {
      document.removeEventListener("keydown", h, { capture: !0 }), document.removeEventListener("pointerdown", M, { capture: !0 }), document.removeEventListener("pointermove", M, { capture: !0 });
    };
  }, []), /* @__PURE__ */ r(De, { ...d, children: /* @__PURE__ */ r(
    Ee,
    {
      scope: t,
      open: n,
      onOpenChange: u,
      content: f,
      onContentChange: v,
      children: /* @__PURE__ */ r(
        Zn,
        {
          scope: t,
          onClose: c.useCallback(() => u(!1), [u]),
          isUsingKeyboardRef: p,
          dir: m,
          modal: i,
          children: o
        }
      )
    }
  ) });
};
Ne.displayName = K;
var qn = "MenuAnchor", ae = c.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...o } = e, a = $(n);
    return /* @__PURE__ */ r(Pn, { ...a, ...o, ref: t });
  }
);
ae.displayName = qn;
var se = "MenuPortal", [Jn, Te] = P(se, {
  forceMount: void 0
}), Ae = (e) => {
  const { __scopeMenu: t, forceMount: n, children: o, container: a } = e, s = S(se, t);
  return /* @__PURE__ */ r(Jn, { scope: t, forceMount: n, children: /* @__PURE__ */ r(Z, { present: n || s.open, children: /* @__PURE__ */ r(yn, { asChild: !0, container: a, children: o }) }) });
};
Ae.displayName = se;
var C = "MenuContent", [Qn, ce] = P(C), Oe = c.forwardRef(
  (e, t) => {
    const n = Te(C, e.__scopeMenu), { forceMount: o = n.forceMount, ...a } = e, s = S(C, e.__scopeMenu), i = U(C, e.__scopeMenu);
    return /* @__PURE__ */ r(L.Provider, { scope: e.__scopeMenu, children: /* @__PURE__ */ r(Z, { present: o || s.open, children: /* @__PURE__ */ r(L.Slot, { scope: e.__scopeMenu, children: i.modal ? /* @__PURE__ */ r(et, { ...a, ref: t }) : /* @__PURE__ */ r(nt, { ...a, ref: t }) }) }) });
  }
), et = c.forwardRef(
  (e, t) => {
    const n = S(C, e.__scopeMenu), o = c.useRef(null), a = F(t, o);
    return c.useEffect(() => {
      const s = o.current;
      if (s) return An(s);
    }, []), /* @__PURE__ */ r(
      ue,
      {
        ...e,
        ref: a,
        trapFocus: n.open,
        disableOutsidePointerEvents: n.open,
        disableOutsideScroll: !0,
        onFocusOutside: g(
          e.onFocusOutside,
          (s) => s.preventDefault(),
          { checkForDefaultPrevented: !1 }
        ),
        onDismiss: () => n.onOpenChange(!1)
      }
    );
  }
), nt = c.forwardRef((e, t) => {
  const n = S(C, e.__scopeMenu);
  return /* @__PURE__ */ r(
    ue,
    {
      ...e,
      ref: t,
      trapFocus: !1,
      disableOutsidePointerEvents: !1,
      disableOutsideScroll: !1,
      onDismiss: () => n.onOpenChange(!1)
    }
  );
}), ue = c.forwardRef(
  (e, t) => {
    const {
      __scopeMenu: n,
      loop: o = !1,
      trapFocus: a,
      onOpenAutoFocus: s,
      onCloseAutoFocus: i,
      disableOutsidePointerEvents: d,
      onEntryFocus: f,
      onEscapeKeyDown: v,
      onPointerDownOutside: p,
      onFocusOutside: u,
      onInteractOutside: m,
      onDismiss: h,
      disableOutsideScroll: M,
      ...I
    } = e, y = S(C, n), A = U(C, n), z = $(n), B = ye(n), me = Hn(n), [hn, ve] = c.useState(null), V = c.useRef(null), wn = F(t, V, y.onContentChange), X = c.useRef(0), Y = c.useRef(""), _n = c.useRef(0), J = c.useRef(null), ge = c.useRef("right"), Q = c.useRef(0), Cn = M ? kn : c.Fragment, Rn = M ? { as: xn, allowPinchZoom: !0 } : void 0, bn = (l) => {
      var N, he;
      const _ = Y.current + l, R = me().filter((b) => !b.disabled), D = document.activeElement, ee = (N = R.find((b) => b.ref.current === D)) == null ? void 0 : N.textValue, ne = R.map((b) => b.textValue), Me = ft(ne, _, ee), O = (he = R.find((b) => b.textValue === Me)) == null ? void 0 : he.ref.current;
      (function b(we) {
        Y.current = we, window.clearTimeout(X.current), we !== "" && (X.current = window.setTimeout(() => b(""), 1e3));
      })(_), O && setTimeout(() => O.focus());
    };
    c.useEffect(() => () => window.clearTimeout(X.current), []), On();
    const E = c.useCallback((l) => {
      var R, D;
      return ge.current === ((R = J.current) == null ? void 0 : R.side) && vt(l, (D = J.current) == null ? void 0 : D.area);
    }, []);
    return /* @__PURE__ */ r(
      Qn,
      {
        scope: n,
        searchRef: Y,
        onItemEnter: c.useCallback(
          (l) => {
            E(l) && l.preventDefault();
          },
          [E]
        ),
        onItemLeave: c.useCallback(
          (l) => {
            var _;
            E(l) || ((_ = V.current) == null || _.focus(), ve(null));
          },
          [E]
        ),
        onTriggerLeave: c.useCallback(
          (l) => {
            E(l) && l.preventDefault();
          },
          [E]
        ),
        pointerGraceTimerRef: _n,
        onPointerGraceIntentChange: c.useCallback((l) => {
          J.current = l;
        }, []),
        children: /* @__PURE__ */ r(Cn, { ...Rn, children: /* @__PURE__ */ r(
          Ln,
          {
            asChild: !0,
            trapped: a,
            onMountAutoFocus: g(s, (l) => {
              var _;
              l.preventDefault(), (_ = V.current) == null || _.focus({ preventScroll: !0 });
            }),
            onUnmountAutoFocus: i,
            children: /* @__PURE__ */ r(
              En,
              {
                asChild: !0,
                disableOutsidePointerEvents: d,
                onEscapeKeyDown: v,
                onPointerDownOutside: p,
                onFocusOutside: u,
                onInteractOutside: m,
                onDismiss: h,
                children: /* @__PURE__ */ r(
                  Kn,
                  {
                    asChild: !0,
                    ...B,
                    dir: A.dir,
                    orientation: "vertical",
                    loop: o,
                    currentTabStopId: hn,
                    onCurrentTabStopIdChange: ve,
                    onEntryFocus: g(f, (l) => {
                      A.isUsingKeyboardRef.current || l.preventDefault();
                    }),
                    preventScrollOnEntryFocus: !0,
                    children: /* @__PURE__ */ r(
                      Nn,
                      {
                        role: "menu",
                        "aria-orientation": "vertical",
                        "data-state": qe(y.open),
                        "data-radix-menu-content": "",
                        dir: A.dir,
                        ...z,
                        ...I,
                        ref: wn,
                        style: { outline: "none", ...I.style },
                        onKeyDown: g(I.onKeyDown, (l) => {
                          const R = l.target.closest("[data-radix-menu-content]") === l.currentTarget, D = l.ctrlKey || l.altKey || l.metaKey, ee = l.key.length === 1;
                          R && (l.key === "Tab" && l.preventDefault(), !D && ee && bn(l.key));
                          const ne = V.current;
                          if (l.target !== ne || !Xn.includes(l.key)) return;
                          l.preventDefault();
                          const O = me().filter((N) => !N.disabled).map((N) => N.ref.current);
                          Ie.includes(l.key) && O.reverse(), lt(O);
                        }),
                        onBlur: g(e.onBlur, (l) => {
                          l.currentTarget.contains(l.target) || (window.clearTimeout(X.current), Y.current = "");
                        }),
                        onPointerMove: g(
                          e.onPointerMove,
                          G((l) => {
                            const _ = l.target, R = Q.current !== l.clientX;
                            if (l.currentTarget.contains(_) && R) {
                              const D = l.clientX > Q.current ? "right" : "left";
                              ge.current = D, Q.current = l.clientX;
                            }
                          })
                        )
                      }
                    )
                  }
                )
              }
            )
          }
        ) })
      }
    );
  }
);
Oe.displayName = C;
var tt = "MenuGroup", ie = c.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...o } = e;
    return /* @__PURE__ */ r(T.div, { role: "group", ...o, ref: t });
  }
);
ie.displayName = tt;
var ot = "MenuLabel", ke = c.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...o } = e;
    return /* @__PURE__ */ r(T.div, { ...o, ref: t });
  }
);
ke.displayName = ot;
var H = "MenuItem", _e = "menu.itemSelect", q = c.forwardRef(
  (e, t) => {
    const { disabled: n = !1, onSelect: o, ...a } = e, s = c.useRef(null), i = U(H, e.__scopeMenu), d = ce(H, e.__scopeMenu), f = F(t, s), v = c.useRef(!1), p = () => {
      const u = s.current;
      if (!n && u) {
        const m = new CustomEvent(_e, { bubbles: !0, cancelable: !0 });
        u.addEventListener(_e, (h) => o == null ? void 0 : o(h), { once: !0 }), Dn(u, m), m.defaultPrevented ? v.current = !1 : i.onClose();
      }
    };
    return /* @__PURE__ */ r(
      Le,
      {
        ...a,
        ref: f,
        disabled: n,
        onClick: g(e.onClick, p),
        onPointerDown: (u) => {
          var m;
          (m = e.onPointerDown) == null || m.call(e, u), v.current = !0;
        },
        onPointerUp: g(e.onPointerUp, (u) => {
          var m;
          v.current || (m = u.currentTarget) == null || m.click();
        }),
        onKeyDown: g(e.onKeyDown, (u) => {
          const m = d.searchRef.current !== "";
          n || m && u.key === " " || te.includes(u.key) && (u.currentTarget.click(), u.preventDefault());
        })
      }
    );
  }
);
q.displayName = H;
var Le = c.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, disabled: o = !1, textValue: a, ...s } = e, i = ce(H, n), d = ye(n), f = c.useRef(null), v = F(t, f), [p, u] = c.useState(!1), [m, h] = c.useState("");
    return c.useEffect(() => {
      const M = f.current;
      M && h((M.textContent ?? "").trim());
    }, [s.children]), /* @__PURE__ */ r(
      L.ItemSlot,
      {
        scope: n,
        disabled: o,
        textValue: a ?? m,
        children: /* @__PURE__ */ r(Fn, { asChild: !0, ...d, focusable: !o, children: /* @__PURE__ */ r(
          T.div,
          {
            role: "menuitem",
            "data-highlighted": p ? "" : void 0,
            "aria-disabled": o || void 0,
            "data-disabled": o ? "" : void 0,
            ...s,
            ref: v,
            onPointerMove: g(
              e.onPointerMove,
              G((M) => {
                o ? i.onItemLeave(M) : (i.onItemEnter(M), M.defaultPrevented || M.currentTarget.focus({ preventScroll: !0 }));
              })
            ),
            onPointerLeave: g(
              e.onPointerLeave,
              G((M) => i.onItemLeave(M))
            ),
            onFocus: g(e.onFocus, () => u(!0)),
            onBlur: g(e.onBlur, () => u(!1))
          }
        ) })
      }
    );
  }
), rt = "MenuCheckboxItem", Ge = c.forwardRef(
  (e, t) => {
    const { checked: n = !1, onCheckedChange: o, ...a } = e;
    return /* @__PURE__ */ r(ze, { scope: e.__scopeMenu, checked: n, children: /* @__PURE__ */ r(
      q,
      {
        role: "menuitemcheckbox",
        "aria-checked": W(n) ? "mixed" : n,
        ...a,
        ref: t,
        "data-state": pe(n),
        onSelect: g(
          a.onSelect,
          () => o == null ? void 0 : o(W(n) ? !0 : !n),
          { checkForDefaultPrevented: !1 }
        )
      }
    ) });
  }
);
Ge.displayName = rt;
var Fe = "MenuRadioGroup", [at, st] = P(
  Fe,
  { value: void 0, onValueChange: () => {
  } }
), Ke = c.forwardRef(
  (e, t) => {
    const { value: n, onValueChange: o, ...a } = e, s = re(o);
    return /* @__PURE__ */ r(at, { scope: e.__scopeMenu, value: n, onValueChange: s, children: /* @__PURE__ */ r(ie, { ...a, ref: t }) });
  }
);
Ke.displayName = Fe;
var $e = "MenuRadioItem", Ue = c.forwardRef(
  (e, t) => {
    const { value: n, ...o } = e, a = st($e, e.__scopeMenu), s = n === a.value;
    return /* @__PURE__ */ r(ze, { scope: e.__scopeMenu, checked: s, children: /* @__PURE__ */ r(
      q,
      {
        role: "menuitemradio",
        "aria-checked": s,
        ...o,
        ref: t,
        "data-state": pe(s),
        onSelect: g(
          o.onSelect,
          () => {
            var i;
            return (i = a.onValueChange) == null ? void 0 : i.call(a, n);
          },
          { checkForDefaultPrevented: !1 }
        )
      }
    ) });
  }
);
Ue.displayName = $e;
var de = "MenuItemIndicator", [ze, ct] = P(
  de,
  { checked: !1 }
), Be = c.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, forceMount: o, ...a } = e, s = ct(de, n);
    return /* @__PURE__ */ r(
      Z,
      {
        present: o || W(s.checked) || s.checked === !0,
        children: /* @__PURE__ */ r(
          T.span,
          {
            ...a,
            ref: t,
            "data-state": pe(s.checked)
          }
        )
      }
    );
  }
);
Be.displayName = de;
var ut = "MenuSeparator", Ve = c.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...o } = e;
    return /* @__PURE__ */ r(
      T.div,
      {
        role: "separator",
        "aria-orientation": "horizontal",
        ...o,
        ref: t
      }
    );
  }
);
Ve.displayName = ut;
var it = "MenuArrow", Xe = c.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...o } = e, a = $(n);
    return /* @__PURE__ */ r(Tn, { ...a, ...o, ref: t });
  }
);
Xe.displayName = it;
var le = "MenuSub", [dt, Ye] = P(le), je = (e) => {
  const { __scopeMenu: t, children: n, open: o = !1, onOpenChange: a } = e, s = S(le, t), i = $(t), [d, f] = c.useState(null), [v, p] = c.useState(null), u = re(a);
  return c.useEffect(() => (s.open === !1 && u(!1), () => u(!1)), [s.open, u]), /* @__PURE__ */ r(De, { ...i, children: /* @__PURE__ */ r(
    Ee,
    {
      scope: t,
      open: o,
      onOpenChange: u,
      content: v,
      onContentChange: p,
      children: /* @__PURE__ */ r(
        dt,
        {
          scope: t,
          contentId: j(),
          triggerId: j(),
          trigger: d,
          onTriggerChange: f,
          children: n
        }
      )
    }
  ) });
};
je.displayName = le;
var k = "MenuSubTrigger", He = c.forwardRef(
  (e, t) => {
    const n = S(k, e.__scopeMenu), o = U(k, e.__scopeMenu), a = Ye(k, e.__scopeMenu), s = ce(k, e.__scopeMenu), i = c.useRef(null), { pointerGraceTimerRef: d, onPointerGraceIntentChange: f } = s, v = { __scopeMenu: e.__scopeMenu }, p = c.useCallback(() => {
      i.current && window.clearTimeout(i.current), i.current = null;
    }, []);
    return c.useEffect(() => p, [p]), c.useEffect(() => {
      const u = d.current;
      return () => {
        window.clearTimeout(u), f(null);
      };
    }, [d, f]), /* @__PURE__ */ r(ae, { asChild: !0, ...v, children: /* @__PURE__ */ r(
      Le,
      {
        id: a.triggerId,
        "aria-haspopup": "menu",
        "aria-expanded": n.open,
        "aria-controls": a.contentId,
        "data-state": qe(n.open),
        ...e,
        ref: Re(t, a.onTriggerChange),
        onClick: (u) => {
          var m;
          (m = e.onClick) == null || m.call(e, u), !(e.disabled || u.defaultPrevented) && (u.currentTarget.focus(), n.open || n.onOpenChange(!0));
        },
        onPointerMove: g(
          e.onPointerMove,
          G((u) => {
            s.onItemEnter(u), !u.defaultPrevented && !e.disabled && !n.open && !i.current && (s.onPointerGraceIntentChange(null), i.current = window.setTimeout(() => {
              n.onOpenChange(!0), p();
            }, 100));
          })
        ),
        onPointerLeave: g(
          e.onPointerLeave,
          G((u) => {
            var h, M;
            p();
            const m = (h = n.content) == null ? void 0 : h.getBoundingClientRect();
            if (m) {
              const I = (M = n.content) == null ? void 0 : M.dataset.side, y = I === "right", A = y ? -5 : 5, z = m[y ? "left" : "right"], B = m[y ? "right" : "left"];
              s.onPointerGraceIntentChange({
                area: [
                  // Apply a bleed on clientX to ensure that our exit point is
                  // consistently within polygon bounds
                  { x: u.clientX + A, y: u.clientY },
                  { x: z, y: m.top },
                  { x: B, y: m.top },
                  { x: B, y: m.bottom },
                  { x: z, y: m.bottom }
                ],
                side: I
              }), window.clearTimeout(d.current), d.current = window.setTimeout(
                () => s.onPointerGraceIntentChange(null),
                300
              );
            } else {
              if (s.onTriggerLeave(u), u.defaultPrevented) return;
              s.onPointerGraceIntentChange(null);
            }
          })
        ),
        onKeyDown: g(e.onKeyDown, (u) => {
          var h;
          const m = s.searchRef.current !== "";
          e.disabled || m && u.key === " " || Yn[o.dir].includes(u.key) && (n.onOpenChange(!0), (h = n.content) == null || h.focus(), u.preventDefault());
        })
      }
    ) });
  }
);
He.displayName = k;
var We = "MenuSubContent", Ze = c.forwardRef(
  (e, t) => {
    const n = Te(C, e.__scopeMenu), { forceMount: o = n.forceMount, ...a } = e, s = S(C, e.__scopeMenu), i = U(C, e.__scopeMenu), d = Ye(We, e.__scopeMenu), f = c.useRef(null), v = F(t, f);
    return /* @__PURE__ */ r(L.Provider, { scope: e.__scopeMenu, children: /* @__PURE__ */ r(Z, { present: o || s.open, children: /* @__PURE__ */ r(L.Slot, { scope: e.__scopeMenu, children: /* @__PURE__ */ r(
      ue,
      {
        id: d.contentId,
        "aria-labelledby": d.triggerId,
        ...a,
        ref: v,
        align: "start",
        side: i.dir === "rtl" ? "left" : "right",
        disableOutsidePointerEvents: !1,
        disableOutsideScroll: !1,
        trapFocus: !1,
        onOpenAutoFocus: (p) => {
          var u;
          i.isUsingKeyboardRef.current && ((u = f.current) == null || u.focus()), p.preventDefault();
        },
        onCloseAutoFocus: (p) => p.preventDefault(),
        onFocusOutside: g(e.onFocusOutside, (p) => {
          p.target !== d.trigger && s.onOpenChange(!1);
        }),
        onEscapeKeyDown: g(e.onEscapeKeyDown, (p) => {
          i.onClose(), p.preventDefault();
        }),
        onKeyDown: g(e.onKeyDown, (p) => {
          var h;
          const u = p.currentTarget.contains(p.target), m = jn[i.dir].includes(p.key);
          u && m && (s.onOpenChange(!1), (h = d.trigger) == null || h.focus(), p.preventDefault());
        })
      }
    ) }) }) });
  }
);
Ze.displayName = We;
function qe(e) {
  return e ? "open" : "closed";
}
function W(e) {
  return e === "indeterminate";
}
function pe(e) {
  return W(e) ? "indeterminate" : e ? "checked" : "unchecked";
}
function lt(e) {
  const t = document.activeElement;
  for (const n of e)
    if (n === t || (n.focus(), document.activeElement !== t)) return;
}
function pt(e, t) {
  return e.map((n, o) => e[(t + o) % e.length]);
}
function ft(e, t, n) {
  const a = t.length > 1 && Array.from(t).every((v) => v === t[0]) ? t[0] : t, s = n ? e.indexOf(n) : -1;
  let i = pt(e, Math.max(s, 0));
  a.length === 1 && (i = i.filter((v) => v !== n));
  const f = i.find(
    (v) => v.toLowerCase().startsWith(a.toLowerCase())
  );
  return f !== n ? f : void 0;
}
function mt(e, t) {
  const { x: n, y: o } = e;
  let a = !1;
  for (let s = 0, i = t.length - 1; s < t.length; i = s++) {
    const d = t[s].x, f = t[s].y, v = t[i].x, p = t[i].y;
    f > o != p > o && n < (v - d) * (o - f) / (p - f) + d && (a = !a);
  }
  return a;
}
function vt(e, t) {
  if (!t) return !1;
  const n = { x: e.clientX, y: e.clientY };
  return mt(n, t);
}
function G(e) {
  return (t) => t.pointerType === "mouse" ? e(t) : void 0;
}
var gt = Ne, Mt = ae, ht = Ae, wt = Oe, _t = ie, Ct = ke, Rt = q, bt = Ge, xt = Ke, Dt = Ue, St = Be, It = Ve, Pt = Xe, yt = je, Et = He, Nt = Ze, fe = "DropdownMenu", [Tt, _o] = be(
  fe,
  [Pe]
), w = Pe(), [At, Je] = Tt(fe), Qe = (e) => {
  const {
    __scopeDropdownMenu: t,
    children: n,
    dir: o,
    open: a,
    defaultOpen: s,
    onOpenChange: i,
    modal: d = !0
  } = e, f = w(t), v = c.useRef(null), [p = !1, u] = Ce({
    prop: a,
    defaultProp: s,
    onChange: i
  });
  return /* @__PURE__ */ r(
    At,
    {
      scope: t,
      triggerId: j(),
      triggerRef: v,
      contentId: j(),
      open: p,
      onOpenChange: u,
      onOpenToggle: c.useCallback(() => u((m) => !m), [u]),
      modal: d,
      children: /* @__PURE__ */ r(gt, { ...f, open: p, onOpenChange: u, dir: o, modal: d, children: n })
    }
  );
};
Qe.displayName = fe;
var en = "DropdownMenuTrigger", nn = c.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, disabled: o = !1, ...a } = e, s = Je(en, n), i = w(n);
    return /* @__PURE__ */ r(Mt, { asChild: !0, ...i, children: /* @__PURE__ */ r(
      T.button,
      {
        type: "button",
        id: s.triggerId,
        "aria-haspopup": "menu",
        "aria-expanded": s.open,
        "aria-controls": s.open ? s.contentId : void 0,
        "data-state": s.open ? "open" : "closed",
        "data-disabled": o ? "" : void 0,
        disabled: o,
        ...a,
        ref: Re(t, s.triggerRef),
        onPointerDown: g(e.onPointerDown, (d) => {
          !o && d.button === 0 && d.ctrlKey === !1 && (s.onOpenToggle(), s.open || d.preventDefault());
        }),
        onKeyDown: g(e.onKeyDown, (d) => {
          o || (["Enter", " "].includes(d.key) && s.onOpenToggle(), d.key === "ArrowDown" && s.onOpenChange(!0), ["Enter", " ", "ArrowDown"].includes(d.key) && d.preventDefault());
        })
      }
    ) });
  }
);
nn.displayName = en;
var Ot = "DropdownMenuPortal", tn = (e) => {
  const { __scopeDropdownMenu: t, ...n } = e, o = w(t);
  return /* @__PURE__ */ r(ht, { ...o, ...n });
};
tn.displayName = Ot;
var on = "DropdownMenuContent", rn = c.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...o } = e, a = Je(on, n), s = w(n), i = c.useRef(!1);
    return /* @__PURE__ */ r(
      wt,
      {
        id: a.contentId,
        "aria-labelledby": a.triggerId,
        ...s,
        ...o,
        ref: t,
        onCloseAutoFocus: g(e.onCloseAutoFocus, (d) => {
          var f;
          i.current || (f = a.triggerRef.current) == null || f.focus(), i.current = !1, d.preventDefault();
        }),
        onInteractOutside: g(e.onInteractOutside, (d) => {
          const f = d.detail.originalEvent, v = f.button === 0 && f.ctrlKey === !0, p = f.button === 2 || v;
          (!a.modal || p) && (i.current = !0);
        }),
        style: {
          ...e.style,
          "--radix-dropdown-menu-content-transform-origin": "var(--radix-popper-transform-origin)",
          "--radix-dropdown-menu-content-available-width": "var(--radix-popper-available-width)",
          "--radix-dropdown-menu-content-available-height": "var(--radix-popper-available-height)",
          "--radix-dropdown-menu-trigger-width": "var(--radix-popper-anchor-width)",
          "--radix-dropdown-menu-trigger-height": "var(--radix-popper-anchor-height)"
        }
      }
    );
  }
);
rn.displayName = on;
var kt = "DropdownMenuGroup", an = c.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
    return /* @__PURE__ */ r(_t, { ...a, ...o, ref: t });
  }
);
an.displayName = kt;
var Lt = "DropdownMenuLabel", sn = c.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
    return /* @__PURE__ */ r(Ct, { ...a, ...o, ref: t });
  }
);
sn.displayName = Lt;
var Gt = "DropdownMenuItem", cn = c.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
    return /* @__PURE__ */ r(Rt, { ...a, ...o, ref: t });
  }
);
cn.displayName = Gt;
var Ft = "DropdownMenuCheckboxItem", un = c.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
  return /* @__PURE__ */ r(bt, { ...a, ...o, ref: t });
});
un.displayName = Ft;
var Kt = "DropdownMenuRadioGroup", dn = c.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
  return /* @__PURE__ */ r(xt, { ...a, ...o, ref: t });
});
dn.displayName = Kt;
var $t = "DropdownMenuRadioItem", ln = c.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
  return /* @__PURE__ */ r(Dt, { ...a, ...o, ref: t });
});
ln.displayName = $t;
var Ut = "DropdownMenuItemIndicator", pn = c.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
  return /* @__PURE__ */ r(St, { ...a, ...o, ref: t });
});
pn.displayName = Ut;
var zt = "DropdownMenuSeparator", fn = c.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
  return /* @__PURE__ */ r(It, { ...a, ...o, ref: t });
});
fn.displayName = zt;
var Bt = "DropdownMenuArrow", Vt = c.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
    return /* @__PURE__ */ r(Pt, { ...a, ...o, ref: t });
  }
);
Vt.displayName = Bt;
var Xt = (e) => {
  const { __scopeDropdownMenu: t, children: n, open: o, onOpenChange: a, defaultOpen: s } = e, i = w(t), [d = !1, f] = Ce({
    prop: o,
    defaultProp: s,
    onChange: a
  });
  return /* @__PURE__ */ r(yt, { ...i, open: d, onOpenChange: f, children: n });
}, Yt = "DropdownMenuSubTrigger", mn = c.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
  return /* @__PURE__ */ r(Et, { ...a, ...o, ref: t });
});
mn.displayName = Yt;
var jt = "DropdownMenuSubContent", vn = c.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...o } = e, a = w(n);
  return /* @__PURE__ */ r(
    Nt,
    {
      ...a,
      ...o,
      ref: t,
      style: {
        ...e.style,
        "--radix-dropdown-menu-content-transform-origin": "var(--radix-popper-transform-origin)",
        "--radix-dropdown-menu-content-available-width": "var(--radix-popper-available-width)",
        "--radix-dropdown-menu-content-available-height": "var(--radix-popper-available-height)",
        "--radix-dropdown-menu-trigger-width": "var(--radix-popper-anchor-width)",
        "--radix-dropdown-menu-trigger-height": "var(--radix-popper-anchor-height)"
      }
    }
  );
});
vn.displayName = jt;
var Ht = Qe, Wt = nn, gn = tn, Zt = rn, qt = an, Jt = sn, Qt = cn, eo = un, no = dn, to = ln, Mn = pn, oo = fn, ro = Xt, ao = mn, so = vn;
function Co({
  ...e
}) {
  return /* @__PURE__ */ r(Ht, { "data-slot": "dropdown-menu", ...e });
}
function Ro({
  ...e
}) {
  return /* @__PURE__ */ r(gn, { "data-slot": "dropdown-menu-portal", ...e });
}
function bo({
  ...e
}) {
  return /* @__PURE__ */ r(
    Wt,
    {
      "data-slot": "dropdown-menu-trigger",
      ...e
    }
  );
}
function xo({
  className: e,
  sideOffset: t = 4,
  ...n
}) {
  return /* @__PURE__ */ r(gn, { children: /* @__PURE__ */ r(
    Zt,
    {
      "data-slot": "dropdown-menu-content",
      sideOffset: t,
      className: x(
        "bg-popover text-popover-foreground data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 max-h-(--radix-dropdown-menu-content-available-height) min-w-[8rem] origin-(--radix-dropdown-menu-content-transform-origin) overflow-x-hidden overflow-y-auto rounded-md border p-1 shadow-md",
        e
      ),
      ...n
    }
  ) });
}
function Do({
  ...e
}) {
  return /* @__PURE__ */ r(qt, { "data-slot": "dropdown-menu-group", ...e });
}
function So({
  className: e,
  inset: t,
  variant: n = "default",
  ...o
}) {
  return /* @__PURE__ */ r(
    Qt,
    {
      "data-slot": "dropdown-menu-item",
      "data-inset": t,
      "data-variant": n,
      className: x(
        "focus:bg-accent focus:text-accent-foreground data-[variant=destructive]:text-destructive data-[variant=destructive]:focus:bg-destructive/10 dark:data-[variant=destructive]:focus:bg-destructive/20 data-[variant=destructive]:focus:text-destructive data-[variant=destructive]:*:[svg]:!text-destructive [&_svg:not([class*='text-'])]:text-muted-foreground relative flex cursor-default items-center gap-2 rounded-sm px-2 py-1.5 text-sm outline-hidden select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 data-[inset]:pl-8 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4",
        e
      ),
      ...o
    }
  );
}
function Io({
  className: e,
  children: t,
  checked: n,
  ...o
}) {
  return /* @__PURE__ */ oe(
    eo,
    {
      "data-slot": "dropdown-menu-checkbox-item",
      className: x(
        "focus:bg-accent focus:text-accent-foreground relative flex cursor-default items-center gap-2 rounded-sm py-1.5 pr-2 pl-8 text-sm outline-hidden select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4",
        e
      ),
      checked: n,
      ...o,
      children: [
        /* @__PURE__ */ r("span", { className: "pointer-events-none absolute left-2 flex size-3.5 items-center justify-center", children: /* @__PURE__ */ r(Mn, { children: /* @__PURE__ */ r(Gn, { className: "size-4" }) }) }),
        t
      ]
    }
  );
}
function Po({
  ...e
}) {
  return /* @__PURE__ */ r(
    no,
    {
      "data-slot": "dropdown-menu-radio-group",
      ...e
    }
  );
}
function yo({
  className: e,
  children: t,
  ...n
}) {
  return /* @__PURE__ */ oe(
    to,
    {
      "data-slot": "dropdown-menu-radio-item",
      className: x(
        "focus:bg-accent focus:text-accent-foreground relative flex cursor-default items-center gap-2 rounded-sm py-1.5 pr-2 pl-8 text-sm outline-hidden select-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4",
        e
      ),
      ...n,
      children: [
        /* @__PURE__ */ r("span", { className: "pointer-events-none absolute left-2 flex size-3.5 items-center justify-center", children: /* @__PURE__ */ r(Mn, { children: /* @__PURE__ */ r(Bn, { className: "size-2 fill-current" }) }) }),
        t
      ]
    }
  );
}
function Eo({
  className: e,
  inset: t,
  ...n
}) {
  return /* @__PURE__ */ r(
    Jt,
    {
      "data-slot": "dropdown-menu-label",
      "data-inset": t,
      className: x(
        "px-2 py-1.5 text-sm font-medium data-[inset]:pl-8",
        e
      ),
      ...n
    }
  );
}
function No({
  className: e,
  ...t
}) {
  return /* @__PURE__ */ r(
    oo,
    {
      "data-slot": "dropdown-menu-separator",
      className: x("bg-border -mx-1 my-1 h-px", e),
      ...t
    }
  );
}
function To({
  className: e,
  ...t
}) {
  return /* @__PURE__ */ r(
    "span",
    {
      "data-slot": "dropdown-menu-shortcut",
      className: x(
        "text-muted-foreground ml-auto text-xs tracking-widest",
        e
      ),
      ...t
    }
  );
}
function Ao({
  ...e
}) {
  return /* @__PURE__ */ r(ro, { "data-slot": "dropdown-menu-sub", ...e });
}
function Oo({
  className: e,
  inset: t,
  children: n,
  ...o
}) {
  return /* @__PURE__ */ oe(
    ao,
    {
      "data-slot": "dropdown-menu-sub-trigger",
      "data-inset": t,
      className: x(
        "focus:bg-accent focus:text-accent-foreground data-[state=open]:bg-accent data-[state=open]:text-accent-foreground flex cursor-default items-center rounded-sm px-2 py-1.5 text-sm outline-hidden select-none data-[inset]:pl-8",
        e
      ),
      ...o,
      children: [
        n,
        /* @__PURE__ */ r(Un, { className: "ml-auto size-4" })
      ]
    }
  );
}
function ko({
  className: e,
  ...t
}) {
  return /* @__PURE__ */ r(
    so,
    {
      "data-slot": "dropdown-menu-sub-content",
      className: x(
        "bg-popover text-popover-foreground data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 min-w-[8rem] origin-(--radix-dropdown-menu-content-transform-origin) overflow-hidden rounded-md border p-1 shadow-lg",
        e
      ),
      ...t
    }
  );
}
export {
  Co as DropdownMenu,
  Io as DropdownMenuCheckboxItem,
  xo as DropdownMenuContent,
  Do as DropdownMenuGroup,
  So as DropdownMenuItem,
  Eo as DropdownMenuLabel,
  Ro as DropdownMenuPortal,
  Po as DropdownMenuRadioGroup,
  yo as DropdownMenuRadioItem,
  No as DropdownMenuSeparator,
  To as DropdownMenuShortcut,
  Ao as DropdownMenuSub,
  ko as DropdownMenuSubContent,
  Oo as DropdownMenuSubTrigger,
  bo as DropdownMenuTrigger
};
