// tailgrid/dist/components/ui/pagination.js
import { jsx as e, jsxs as o } from "react/jsx-runtime";
import * as r from "react";
import { c as n } from "../../utils-B9KdtH6G.js";
import { Button as c, buttonVariants as m } from "./button.js";
import { c as p } from "../../createLucideIcon-DUBSZsqM.js";
import { C as g } from "../../chevron-right-CO-eg3PL.js";
import { E as f } from "../../ellipsis-CU_hPMy5.js";
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const d = [["path", { d: "m15 18-6-6 6-6", key: "1wnfg3" }]], N = p("ChevronLeft", d), E = ({ className: i, ...a }) => /* @__PURE__ */ e(
  "nav",
  {
    role: "navigation",
    "aria-label": "pagination",
    className: n("mx-auto flex w-full justify-center", i),
    ...a
  }
), h = r.forwardRef(({ className: i, ...a }, t) => /* @__PURE__ */ e(
  "ul",
  {
    ref: t,
    className: n("flex flex-row items-center gap-1", i),
    ...a
  }
));
h.displayName = "PaginationContent";
const u = r.forwardRef(({ className: i, ...a }, t) => /* @__PURE__ */ e("li", { ref: t, className: n("", i), ...a }));
u.displayName = "PaginationItem";
const s = ({
  className: i,
  isActive: a,
  size: t = "icon",
  ...l
}) => /* @__PURE__ */ e(
  c,
  {
    "aria-current": a ? "page" : void 0,
    className: n(
      m({
        variant: a ? "outline" : "ghost",
        size: t
      }),
      i
    ),
    ...l
  }
);
s.displayName = "PaginationLink";
const P = ({
  className: i,
  ...a
}) => /* @__PURE__ */ o(
  s,
  {
    "aria-label": "Go to previous page",
    size: "default",
    className: n("gap-1 pl-2.5", i),
    ...a,
    children: [
      /* @__PURE__ */ e(N, { className: "h-4 w-4" }),
      /* @__PURE__ */ e("span", { children: "Previous" })
    ]
  }
);
P.displayName = "PaginationPrevious";
const x = ({
  className: i,
  ...a
}) => /* @__PURE__ */ o(
  s,
  {
    "aria-label": "Go to next page",
    size: "default",
    className: n("gap-1 pr-2.5", i),
    ...a,
    children: [
      /* @__PURE__ */ e("span", { children: "Next" }),
      /* @__PURE__ */ e(g, { className: "h-4 w-4" })
    ]
  }
);
x.displayName = "PaginationNext";
const v = ({
  className: i,
  ...a
}) => /* @__PURE__ */ o(
  "span",
  {
    "aria-hidden": !0,
    className: n("flex h-9 w-9 items-center justify-center", i),
    ...a,
    children: [
      /* @__PURE__ */ e(f, { className: "h-4 w-4" }),
      /* @__PURE__ */ e("span", { className: "sr-only", children: "More pages" })
    ]
  }
);
v.displayName = "PaginationEllipsis";
export {
  E as Pagination,
  h as PaginationContent,
  v as PaginationEllipsis,
  u as PaginationItem,
  s as PaginationLink,
  x as PaginationNext,
  P as PaginationPrevious
};
