// tailgrid/dist/components/ui/tabs.js
import { jsx as r } from "react/jsx-runtime";
import * as f from "react";
import { a as M, u as k, c as m } from "../../index-WxTJUbsU.js";
import { c as E, P as g } from "../../index-CDSAOBSC.js";
import { c as h, R as F, I as $ } from "../../index-DgiLjZTI.js";
import { P as D } from "../../index-C2uZYgGF.js";
import { u as V } from "../../index-BOS8297p.js";
import { c as p } from "../../utils-B9KdtH6G.js";
var T = "Tabs", [G, X] = E(T, [
  h
]), C = h(), [L, x] = G(T), I = f.forwardRef(
  (e, t) => {
    const {
      __scopeTabs: c,
      value: a,
      onValueChange: s,
      defaultValue: d,
      orientation: o = "horizontal",
      dir: u,
      activationMode: b = "automatic",
      ...v
    } = e, i = V(u), [n, l] = M({
      prop: a,
      onChange: s,
      defaultProp: d
    });
    return /* @__PURE__ */ r(
      L,
      {
        scope: c,
        baseId: k(),
        value: n,
        onValueChange: l,
        orientation: o,
        dir: i,
        activationMode: b,
        children: /* @__PURE__ */ r(
          g.div,
          {
            dir: i,
            "data-orientation": o,
            ...v,
            ref: t
          }
        )
      }
    );
  }
);
I.displayName = T;
var y = "TabsList", _ = f.forwardRef(
  (e, t) => {
    const { __scopeTabs: c, loop: a = !0, ...s } = e, d = x(y, c), o = C(c);
    return /* @__PURE__ */ r(
      F,
      {
        asChild: !0,
        ...o,
        orientation: d.orientation,
        dir: d.dir,
        loop: a,
        children: /* @__PURE__ */ r(
          g.div,
          {
            role: "tablist",
            "aria-orientation": d.orientation,
            ...s,
            ref: t
          }
        )
      }
    );
  }
);
_.displayName = y;
var N = "TabsTrigger", R = f.forwardRef(
  (e, t) => {
    const { __scopeTabs: c, value: a, disabled: s = !1, ...d } = e, o = x(N, c), u = C(c), b = P(o.baseId, a), v = S(o.baseId, a), i = a === o.value;
    return /* @__PURE__ */ r(
      $,
      {
        asChild: !0,
        ...u,
        focusable: !s,
        active: i,
        children: /* @__PURE__ */ r(
          g.button,
          {
            type: "button",
            role: "tab",
            "aria-selected": i,
            "aria-controls": v,
            "data-state": i ? "active" : "inactive",
            "data-disabled": s ? "" : void 0,
            disabled: s,
            id: b,
            ...d,
            ref: t,
            onMouseDown: m(e.onMouseDown, (n) => {
              !s && n.button === 0 && n.ctrlKey === !1 ? o.onValueChange(a) : n.preventDefault();
            }),
            onKeyDown: m(e.onKeyDown, (n) => {
              [" ", "Enter"].includes(n.key) && o.onValueChange(a);
            }),
            onFocus: m(e.onFocus, () => {
              const n = o.activationMode !== "manual";
              !i && !s && n && o.onValueChange(a);
            })
          }
        )
      }
    );
  }
);
R.displayName = N;
var w = "TabsContent", A = f.forwardRef(
  (e, t) => {
    const { __scopeTabs: c, value: a, forceMount: s, children: d, ...o } = e, u = x(w, c), b = P(u.baseId, a), v = S(u.baseId, a), i = a === u.value, n = f.useRef(i);
    return f.useEffect(() => {
      const l = requestAnimationFrame(() => n.current = !1);
      return () => cancelAnimationFrame(l);
    }, []), /* @__PURE__ */ r(D, { present: s || i, children: ({ present: l }) => /* @__PURE__ */ r(
      g.div,
      {
        "data-state": i ? "active" : "inactive",
        "data-orientation": u.orientation,
        role: "tabpanel",
        "aria-labelledby": b,
        hidden: !l,
        id: v,
        tabIndex: 0,
        ...o,
        ref: t,
        style: {
          ...e.style,
          animationDuration: n.current ? "0s" : void 0
        },
        children: l && d
      }
    ) });
  }
);
A.displayName = w;
function P(e, t) {
  return `${e}-trigger-${t}`;
}
function S(e, t) {
  return `${e}-content-${t}`;
}
var j = I, z = _, K = R, B = A;
function Y({
  className: e,
  ...t
}) {
  return /* @__PURE__ */ r(
    j,
    {
      "data-slot": "tabs",
      className: p("flex flex-col gap-2", e),
      ...t
    }
  );
}
function Z({
  className: e,
  ...t
}) {
  return /* @__PURE__ */ r(
    z,
    {
      "data-slot": "tabs-list",
      className: p(
        "bg-muted text-muted-foreground inline-flex h-9 w-fit items-center justify-center rounded-lg p-[3px]",
        e
      ),
      ...t
    }
  );
}
function ee({
  className: e,
  ...t
}) {
  return /* @__PURE__ */ r(
    K,
    {
      "data-slot": "tabs-trigger",
      className: p(
        "data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4",
        e
      ),
      ...t
    }
  );
}
function te({
  className: e,
  ...t
}) {
  return /* @__PURE__ */ r(
    B,
    {
      "data-slot": "tabs-content",
      className: p("flex-1 outline-none", e),
      ...t
    }
  );
}
export {
  Y as Tabs,
  te as TabsContent,
  Z as TabsList,
  ee as TabsTrigger
};
