// tailgrid/dist/components/ui/tooltip.js
import { jsx as u, jsxs as H } from "react/jsx-runtime";
import * as s from "react";
import { u as K, a as W, c as C } from "../../index-WxTJUbsU.js";
import { u as S, a as J } from "../../index-nyDQUcyF.js";
import { c as Q, P as Z } from "../../index-CDSAOBSC.js";
import { c as G, R as ee, A as te, P as oe, a as re, D as ne, C as se } from "../../index-Uwi29ejC.js";
import { P as F } from "../../index-C2uZYgGF.js";
import { R as ae } from "../../index-C_tBg_Tl.js";
import { c as ie } from "../../utils-B9KdtH6G.js";
var [_, Se] = Q("Tooltip", [
  G
]), D = G(), j = "TooltipProvider", le = 700, A = "tooltip.open", [ce, k] = _(j), z = (e) => {
  const {
    __scopeTooltip: o,
    delayDuration: t = le,
    skipDelayDuration: r = 300,
    disableHoverableContent: n = !1,
    children: i
  } = e, [c, v] = s.useState(!0), a = s.useRef(!1), p = s.useRef(0);
  return s.useEffect(() => {
    const l = p.current;
    return () => window.clearTimeout(l);
  }, []), /* @__PURE__ */ u(
    ce,
    {
      scope: o,
      isOpenDelayed: c,
      delayDuration: t,
      onOpen: s.useCallback(() => {
        window.clearTimeout(p.current), v(!1);
      }, []),
      onClose: s.useCallback(() => {
        window.clearTimeout(p.current), p.current = window.setTimeout(
          () => v(!0),
          r
        );
      }, [r]),
      isPointerInTransitRef: a,
      onPointerInTransitChange: s.useCallback((l) => {
        a.current = l;
      }, []),
      disableHoverableContent: n,
      children: i
    }
  );
};
z.displayName = j;
var O = "Tooltip", [ue, R] = _(O), $ = (e) => {
  const {
    __scopeTooltip: o,
    children: t,
    open: r,
    defaultOpen: n = !1,
    onOpenChange: i,
    disableHoverableContent: c,
    delayDuration: v
  } = e, a = k(O, e.__scopeTooltip), p = D(o), [l, f] = s.useState(null), h = K(), d = s.useRef(0), m = c ?? a.disableHoverableContent, y = v ?? a.delayDuration, T = s.useRef(!1), [g = !1, x] = W({
    prop: r,
    defaultProp: n,
    onChange: (N) => {
      N ? (a.onOpen(), document.dispatchEvent(new CustomEvent(A))) : a.onClose(), i == null || i(N);
    }
  }), w = s.useMemo(() => g ? T.current ? "delayed-open" : "instant-open" : "closed", [g]), P = s.useCallback(() => {
    window.clearTimeout(d.current), d.current = 0, T.current = !1, x(!0);
  }, [x]), E = s.useCallback(() => {
    window.clearTimeout(d.current), d.current = 0, x(!1);
  }, [x]), M = s.useCallback(() => {
    window.clearTimeout(d.current), d.current = window.setTimeout(() => {
      T.current = !0, x(!0), d.current = 0;
    }, y);
  }, [y, x]);
  return s.useEffect(() => () => {
    d.current && (window.clearTimeout(d.current), d.current = 0);
  }, []), /* @__PURE__ */ u(ee, { ...p, children: /* @__PURE__ */ u(
    ue,
    {
      scope: o,
      contentId: h,
      open: g,
      stateAttribute: w,
      trigger: l,
      onTriggerChange: f,
      onTriggerEnter: s.useCallback(() => {
        a.isOpenDelayed ? M() : P();
      }, [a.isOpenDelayed, M, P]),
      onTriggerLeave: s.useCallback(() => {
        m ? E() : (window.clearTimeout(d.current), d.current = 0);
      }, [E, m]),
      onOpen: P,
      onClose: E,
      disableHoverableContent: m,
      children: t
    }
  ) });
};
$.displayName = O;
var L = "TooltipTrigger", B = s.forwardRef(
  (e, o) => {
    const { __scopeTooltip: t, ...r } = e, n = R(L, t), i = k(L, t), c = D(t), v = s.useRef(null), a = S(o, v, n.onTriggerChange), p = s.useRef(!1), l = s.useRef(!1), f = s.useCallback(() => p.current = !1, []);
    return s.useEffect(() => () => document.removeEventListener("pointerup", f), [f]), /* @__PURE__ */ u(te, { asChild: !0, ...c, children: /* @__PURE__ */ u(
      Z.button,
      {
        "aria-describedby": n.open ? n.contentId : void 0,
        "data-state": n.stateAttribute,
        ...r,
        ref: a,
        onPointerMove: C(e.onPointerMove, (h) => {
          h.pointerType !== "touch" && !l.current && !i.isPointerInTransitRef.current && (n.onTriggerEnter(), l.current = !0);
        }),
        onPointerLeave: C(e.onPointerLeave, () => {
          n.onTriggerLeave(), l.current = !1;
        }),
        onPointerDown: C(e.onPointerDown, () => {
          p.current = !0, document.addEventListener("pointerup", f, { once: !0 });
        }),
        onFocus: C(e.onFocus, () => {
          p.current || n.onOpen();
        }),
        onBlur: C(e.onBlur, n.onClose),
        onClick: C(e.onClick, n.onClose)
      }
    ) });
  }
);
B.displayName = L;
var I = "TooltipPortal", [de, pe] = _(I, {
  forceMount: void 0
}), U = (e) => {
  const { __scopeTooltip: o, forceMount: t, children: r, container: n } = e, i = R(I, o);
  return /* @__PURE__ */ u(de, { scope: o, forceMount: t, children: /* @__PURE__ */ u(F, { present: t || i.open, children: /* @__PURE__ */ u(oe, { asChild: !0, container: n, children: r }) }) });
};
U.displayName = I;
var b = "TooltipContent", V = s.forwardRef(
  (e, o) => {
    const t = pe(b, e.__scopeTooltip), { forceMount: r = t.forceMount, side: n = "top", ...i } = e, c = R(b, e.__scopeTooltip);
    return /* @__PURE__ */ u(F, { present: r || c.open, children: c.disableHoverableContent ? /* @__PURE__ */ u(Y, { side: n, ...i, ref: o }) : /* @__PURE__ */ u(fe, { side: n, ...i, ref: o }) });
  }
), fe = s.forwardRef((e, o) => {
  const t = R(b, e.__scopeTooltip), r = k(b, e.__scopeTooltip), n = s.useRef(null), i = S(o, n), [c, v] = s.useState(null), { trigger: a, onClose: p } = t, l = n.current, { onPointerInTransitChange: f } = r, h = s.useCallback(() => {
    v(null), f(!1);
  }, [f]), d = s.useCallback(
    (m, y) => {
      const T = m.currentTarget, g = { x: m.clientX, y: m.clientY }, x = he(g, T.getBoundingClientRect()), w = Te(g, x), P = ye(y.getBoundingClientRect()), E = ge([...w, ...P]);
      v(E), f(!0);
    },
    [f]
  );
  return s.useEffect(() => () => h(), [h]), s.useEffect(() => {
    if (a && l) {
      const m = (T) => d(T, l), y = (T) => d(T, a);
      return a.addEventListener("pointerleave", m), l.addEventListener("pointerleave", y), () => {
        a.removeEventListener("pointerleave", m), l.removeEventListener("pointerleave", y);
      };
    }
  }, [a, l, d, h]), s.useEffect(() => {
    if (c) {
      const m = (y) => {
        const T = y.target, g = { x: y.clientX, y: y.clientY }, x = (a == null ? void 0 : a.contains(T)) || (l == null ? void 0 : l.contains(T)), w = !xe(g, c);
        x ? h() : w && (h(), p());
      };
      return document.addEventListener("pointermove", m), () => document.removeEventListener("pointermove", m);
    }
  }, [a, l, c, p, h]), /* @__PURE__ */ u(Y, { ...e, ref: i });
}), [ve, me] = _(O, { isInside: !1 }), Y = s.forwardRef(
  (e, o) => {
    const {
      __scopeTooltip: t,
      children: r,
      "aria-label": n,
      onEscapeKeyDown: i,
      onPointerDownOutside: c,
      ...v
    } = e, a = R(b, t), p = D(t), { onClose: l } = a;
    return s.useEffect(() => (document.addEventListener(A, l), () => document.removeEventListener(A, l)), [l]), s.useEffect(() => {
      if (a.trigger) {
        const f = (h) => {
          const d = h.target;
          d != null && d.contains(a.trigger) && l();
        };
        return window.addEventListener("scroll", f, { capture: !0 }), () => window.removeEventListener("scroll", f, { capture: !0 });
      }
    }, [a.trigger, l]), /* @__PURE__ */ u(
      ne,
      {
        asChild: !0,
        disableOutsidePointerEvents: !1,
        onEscapeKeyDown: i,
        onPointerDownOutside: c,
        onFocusOutside: (f) => f.preventDefault(),
        onDismiss: l,
        children: /* @__PURE__ */ H(
          se,
          {
            "data-state": a.stateAttribute,
            ...p,
            ...v,
            ref: o,
            style: {
              ...v.style,
              "--radix-tooltip-content-transform-origin": "var(--radix-popper-transform-origin)",
              "--radix-tooltip-content-available-width": "var(--radix-popper-available-width)",
              "--radix-tooltip-content-available-height": "var(--radix-popper-available-height)",
              "--radix-tooltip-trigger-width": "var(--radix-popper-anchor-width)",
              "--radix-tooltip-trigger-height": "var(--radix-popper-anchor-height)"
            },
            children: [
              /* @__PURE__ */ u(J, { children: r }),
              /* @__PURE__ */ u(ve, { scope: t, isInside: !0, children: /* @__PURE__ */ u(ae, { id: a.contentId, role: "tooltip", children: n || r }) })
            ]
          }
        )
      }
    );
  }
);
V.displayName = b;
var q = "TooltipArrow", X = s.forwardRef(
  (e, o) => {
    const { __scopeTooltip: t, ...r } = e, n = D(t);
    return me(
      q,
      t
    ).isInside ? null : /* @__PURE__ */ u(re, { ...n, ...r, ref: o });
  }
);
X.displayName = q;
function he(e, o) {
  const t = Math.abs(o.top - e.y), r = Math.abs(o.bottom - e.y), n = Math.abs(o.right - e.x), i = Math.abs(o.left - e.x);
  switch (Math.min(t, r, n, i)) {
    case i:
      return "left";
    case n:
      return "right";
    case t:
      return "top";
    case r:
      return "bottom";
    default:
      throw new Error("unreachable");
  }
}
function Te(e, o, t = 5) {
  const r = [];
  switch (o) {
    case "top":
      r.push(
        { x: e.x - t, y: e.y + t },
        { x: e.x + t, y: e.y + t }
      );
      break;
    case "bottom":
      r.push(
        { x: e.x - t, y: e.y - t },
        { x: e.x + t, y: e.y - t }
      );
      break;
    case "left":
      r.push(
        { x: e.x + t, y: e.y - t },
        { x: e.x + t, y: e.y + t }
      );
      break;
    case "right":
      r.push(
        { x: e.x - t, y: e.y - t },
        { x: e.x - t, y: e.y + t }
      );
      break;
  }
  return r;
}
function ye(e) {
  const { top: o, right: t, bottom: r, left: n } = e;
  return [
    { x: n, y: o },
    { x: t, y: o },
    { x: t, y: r },
    { x: n, y: r }
  ];
}
function xe(e, o) {
  const { x: t, y: r } = e;
  let n = !1;
  for (let i = 0, c = o.length - 1; i < o.length; c = i++) {
    const v = o[i].x, a = o[i].y, p = o[c].x, l = o[c].y;
    a > r != l > r && t < (p - v) * (r - a) / (l - a) + v && (n = !n);
  }
  return n;
}
function ge(e) {
  const o = e.slice();
  return o.sort((t, r) => t.x < r.x ? -1 : t.x > r.x ? 1 : t.y < r.y ? -1 : t.y > r.y ? 1 : 0), Ce(o);
}
function Ce(e) {
  if (e.length <= 1) return e.slice();
  const o = [];
  for (let r = 0; r < e.length; r++) {
    const n = e[r];
    for (; o.length >= 2; ) {
      const i = o[o.length - 1], c = o[o.length - 2];
      if ((i.x - c.x) * (n.y - c.y) >= (i.y - c.y) * (n.x - c.x)) o.pop();
      else break;
    }
    o.push(n);
  }
  o.pop();
  const t = [];
  for (let r = e.length - 1; r >= 0; r--) {
    const n = e[r];
    for (; t.length >= 2; ) {
      const i = t[t.length - 1], c = t[t.length - 2];
      if ((i.x - c.x) * (n.y - c.y) >= (i.y - c.y) * (n.x - c.x)) t.pop();
      else break;
    }
    t.push(n);
  }
  return t.pop(), o.length === 1 && t.length === 1 && o[0].x === t[0].x && o[0].y === t[0].y ? o : o.concat(t);
}
var be = z, we = $, Pe = B, Ee = U, Re = V, _e = X;
function De({
  delayDuration: e = 0,
  ...o
}) {
  return /* @__PURE__ */ u(
    be,
    {
      "data-slot": "tooltip-provider",
      delayDuration: e,
      ...o
    }
  );
}
function Ge({
  ...e
}) {
  return /* @__PURE__ */ u(De, { children: /* @__PURE__ */ u(we, { "data-slot": "tooltip", ...e }) });
}
function Fe({
  ...e
}) {
  return /* @__PURE__ */ u(Pe, { "data-slot": "tooltip-trigger", ...e });
}
function je({
  className: e,
  sideOffset: o = 0,
  children: t,
  ...r
}) {
  return /* @__PURE__ */ u(Ee, { children: /* @__PURE__ */ H(
    Re,
    {
      "data-slot": "tooltip-content",
      sideOffset: o,
      className: ie(
        "bg-primary text-primary-foreground animate-in fade-in-0 zoom-in-95 data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=closed]:zoom-out-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 w-fit origin-(--radix-tooltip-content-transform-origin) rounded-md px-3 py-1.5 text-xs text-balance",
        e
      ),
      ...r,
      children: [
        t,
        /* @__PURE__ */ u(_e, { className: "bg-primary fill-primary z-50 size-2.5 translate-y-[calc(-50%_-_2px)] rotate-45 rounded-[2px]" })
      ]
    }
  ) });
}
export {
  Ge as Tooltip,
  je as TooltipContent,
  De as TooltipProvider,
  Fe as TooltipTrigger
};
