// tailgrid/dist/components/TailGrid/ErrorBoundary.js
var c = Object.defineProperty;
var l = (t, e, r) => e in t ? c(t, e, { enumerable: !0, configurable: !0, writable: !0, value: r }) : t[e] = r;
var a = (t, e, r) => l(t, typeof e != "symbol" ? e + "" : e, r);
import { jsxs as o, jsx as s } from "react/jsx-runtime";
import { Component as h } from "react";
import { Button as m } from "../ui/button.js";
import { Alert as d, AlertTitle as p, AlertDescription as u } from "../ui/alert.js";
import { c as i } from "../../createLucideIcon-DUBSZsqM.js";
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const y = [
  ["circle", { cx: "12", cy: "12", r: "10", key: "1mglay" }],
  ["line", { x1: "12", x2: "12", y1: "8", y2: "12", key: "1pkeuh" }],
  ["line", { x1: "12", x2: "12.01", y1: "16", y2: "16", key: "4dfq90" }]
], f = i("CircleAlert", y);
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const k = [
  ["path", { d: "M3 12a9 9 0 0 1 9-9 9.75 9.75 0 0 1 6.74 2.74L21 8", key: "v9h5vc" }],
  ["path", { d: "M21 3v5h-5", key: "1q7to0" }],
  ["path", { d: "M21 12a9 9 0 0 1-9 9 9.75 9.75 0 0 1-6.74-2.74L3 16", key: "3uifl3" }],
  ["path", { d: "M8 16H3v5", key: "1cv678" }]
], v = i("RefreshCw", k);
class E extends h {
  constructor(r) {
    super(r);
    a(this, "handleReset", () => {
      this.setState({
        hasError: !1,
        error: null
      });
    });
    this.state = {
      hasError: !1,
      error: null
    };
  }
  static getDerivedStateFromError(r) {
    return {
      hasError: !0,
      error: r
    };
  }
  componentDidCatch(r, n) {
    console.error("TailGrid error:", r, n);
  }
  render() {
    var r;
    return this.state.hasError ? this.props.fallback ? this.props.fallback : /* @__PURE__ */ o(d, { variant: "destructive", className: "my-4", children: [
      /* @__PURE__ */ s(f, { className: "h-4 w-4" }),
      /* @__PURE__ */ s(p, { children: "Something went wrong" }),
      /* @__PURE__ */ o(u, { className: "mt-2", children: [
        /* @__PURE__ */ s("p", { className: "mb-2", children: ((r = this.state.error) == null ? void 0 : r.message) || "An error occurred while rendering the table." }),
        /* @__PURE__ */ o(m, { size: "sm", variant: "outline", onClick: this.handleReset, className: "mt-2", children: [
          /* @__PURE__ */ s(v, { className: "mr-2 h-4 w-4" }),
          "Try again"
        ] })
      ] })
    ] }) : this.props.children;
  }
}
export {
  E as ErrorBoundary
};
