// tailgrid/dist/components/TailGrid/TailGrid.js
import { jsx as e, jsxs as m } from "react/jsx-runtime";
import me, { useMemo as q, useCallback as ue } from "react";
import { useTailGrid as ge } from "../../hooks/useTailGrid.js";
import ve from "./Pagination/PaginationControls.js";
import fe from "./Table/TableBody.js";
import we from "./Filters/FiltersPanel.js";
import ye from "./Table/TableHeader.js";
import { ErrorBoundary as Pe } from "./ErrorBoundary.js";
import { TailGridProvider as Re } from "../../context/TailGridContext.js";
import { A as Te } from "../../index-DiJ3a9ld.js";
import { m as D } from "../../proxy-g3OKwrzk.js";
import '../../assets/TailGrid.css';const Ne = ({
  initialData: H = [],
  columns: t,
  title: d,
  subtitle: s,
  filterFields: $ = [],
  actions: a = [],
  sortable: x = !0,
  pagination: I = !0,
  pageSizeOptions: j = [5, 10, 25, 50, 100],
  defaultPageSize: J = 10,
  onDataFetch: n,
  onServerFilter: Q,
  onServerSort: U,
  className: W,
  expandableConfig: r,
  hasFilterCapability: z = !0,
  customFilterComponent: B,
  isServerSide: X = !1,
  onServerSearch: Y,
  rowKey: h = "id",
  showHeader: _ = !0,
  customLoadingComponent: L,
  customNoDataComponent: M,
  virtualized: Z = !1,
  rowHeight: F = 56,
  emptyStateMessage: K = "No data available",
  loadingMessage: O = "Loading data...",
  keyboardNavigation: u = !0,
  onRowClick: b,
  highlightOnHover: C = !0,
  stickyHeader: ee = !0,
  maxHeight: V,
  containerClassName: re = "",
  headerClassName: ae = "",
  bodyClassName: te = "",
  footerClassName: le = "",
  rowClassName: de = "",
  cellClassName: oe = "",
  headerCellClassName: ie = ""
}) => {
  const {
    data: l,
    isLoading: c,
    currentPage: o,
    pageSize: i,
    totalCount: g,
    totalPages: v,
    handleFilterChange: f,
    handleSortChange: w,
    handleSearchChange: y,
    setCurrentPage: P,
    setPageSize: R,
    expandedRows: T,
    toggleExpand: N,
    filters: S,
    searchTerm: p,
    sortConfig: A,
    resetFilters: E,
    activeRowIndex: G,
    setActiveRowIndex: k,
    handleKeyDown: se
  } = ge({
    initialData: H,
    filterFields: $,
    onDataFetch: n,
    onServerFilter: Q,
    columns: t,
    sortable: x,
    actions: a,
    className: W,
    customLoadingComponent: L,
    customNoDataComponent: M,
    isServerSide: X,
    pageSizeOptions: j,
    pagination: I,
    rowKey: h,
    showHeader: _,
    subtitle: s,
    title: d,
    customFilterComponent: B,
    expandableConfig: r,
    hasFilterCapability: z,
    onServerSort: U,
    onServerSearch: Y,
    defaultPageSize: J,
    keyboardNavigation: u
  }), ne = q(
    () => ({
      data: l,
      isLoading: c,
      currentPage: o,
      pageSize: i,
      totalCount: g,
      totalPages: v,
      filters: S,
      searchTerm: p,
      sortConfig: A,
      expandedRows: T,
      activeRowIndex: G,
      rowKey: h,
      handleFilterChange: f,
      handleSortChange: w,
      handleSearchChange: y,
      setCurrentPage: P,
      setPageSize: R,
      toggleExpand: N,
      resetFilters: E,
      setActiveRowIndex: k
    }),
    [
      l,
      c,
      o,
      i,
      g,
      v,
      S,
      p,
      A,
      T,
      G,
      h,
      f,
      w,
      y,
      P,
      R,
      N,
      E,
      k
    ]
  ), he = ue(async () => {
    n && await n(o, i);
  }, [o, i, n]), ce = q(
    () => t.length + (r != null && r.enabled ? 1 : 0) + (a.length > 0 ? 1 : 0),
    [t.length, r == null ? void 0 : r.enabled, a.length]
  );
  return /* @__PURE__ */ e(Pe, { children: /* @__PURE__ */ e(Re, { value: ne, children: /* @__PURE__ */ m(
    "div",
    {
      className: `w-full space-y-6 ${re}`,
      onKeyDown: u ? se : void 0,
      tabIndex: u ? 0 : void 0,
      role: "grid",
      "aria-rowcount": l.length,
      "aria-colcount": t.length,
      "aria-busy": c,
      "aria-label": d || "Data table",
      children: [
        (d || s) && /* @__PURE__ */ m("div", { className: "space-y-1", children: [
          d && /* @__PURE__ */ e("h1", { className: "text-2xl font-semibold tracking-tight", children: d }),
          s && /* @__PURE__ */ e("p", { className: "text-sm text-muted-foreground", children: s })
        ] }),
        z && /* @__PURE__ */ e(Te, { children: /* @__PURE__ */ e(
          D.div,
          {
            initial: { opacity: 0, y: -10 },
            animate: { opacity: 1, y: 0 },
            transition: { duration: 0.2 },
            children: /* @__PURE__ */ e(
              we,
              {
                filters: S,
                filterFields: $,
                onFilterChange: f,
                onSearchChange: y,
                searchTerm: p,
                customFilterComponent: B,
                data: l,
                onResetFilters: E
              }
            )
          }
        ) }),
        /* @__PURE__ */ m(
          D.div,
          {
            initial: { opacity: 0 },
            animate: { opacity: 1 },
            transition: { duration: 0.3 },
            className: "rounded-lg border bg-card text-card-foreground shadow-sm",
            children: [
              /* @__PURE__ */ e("div", { className: "overflow-hidden", children: /* @__PURE__ */ e(
                "div",
                {
                  className: "relative w-full overflow-auto",
                  style: { maxHeight: V ? `${V}px` : void 0 },
                  children: /* @__PURE__ */ m("table", { className: "w-full caption-bottom text-sm", children: [
                    _ && /* @__PURE__ */ e(
                      ye,
                      {
                        sortable: x,
                        hasExpandableRows: (r == null ? void 0 : r.enabled) || !1,
                        columns: t,
                        sortConfig: A,
                        hasActions: (a == null ? void 0 : a.length) > 0,
                        onSort: w,
                        stickyHeader: ee,
                        headerCellClassName: ie,
                        headerClassName: ae
                      }
                    ),
                    /* @__PURE__ */ e(
                      fe,
                      {
                        data: l,
                        columns: t,
                        expandableConfig: r,
                        actions: a,
                        rowKey: h,
                        expandedRows: T,
                        onToggleExpand: N,
                        onReload: he,
                        isLoading: c,
                        colSpan: ce,
                        customLoadingComponent: L,
                        customNoDataComponent: M,
                        emptyStateMessage: K,
                        loadingMessage: O,
                        virtualized: Z,
                        rowHeight: F,
                        activeRowIndex: G,
                        onRowClick: b,
                        highlightOnHover: C,
                        bodyClassName: te,
                        rowClassName: de,
                        cellClassName: oe
                      }
                    )
                  ] })
                }
              ) }),
              I && l.length > 0 && /* @__PURE__ */ e("div", { className: `border-t px-4 py-4 ${le}`, children: /* @__PURE__ */ e(
                ve,
                {
                  currentPage: o,
                  totalCount: g,
                  pageSize: i,
                  totalPages: v,
                  onPageSizeChange: R,
                  onPageChange: P,
                  pageSizeOptions: j
                }
              ) })
            ]
          }
        )
      ]
    }
  ) }) });
}, _e = me.memo(Ne);
export {
  _e as default
};
