// tailgrid/dist/components/TailGrid/Filters/FiltersPanel.js
import { jsxs as a, jsx as t } from "react/jsx-runtime";
import h from "./SearchInput.js";
import u from "./FilterField.js";
import { Input as g } from "../../ui/input.js";
import { c as v } from "../../../utils-B9KdtH6G.js";
import { getUniqueOptions as y } from "../../../utils/getUniqueOptions.js";
import { Button as x } from "../../ui/button.js";
import N from "react";
import { m as k } from "../../../proxy-g3OKwrzk.js";
import { c as f } from "../../../createLucideIcon-DUBSZsqM.js";
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const b = [
  [
    "path",
    {
      d: "M12.531 3H3a1 1 0 0 0-.742 1.67l7.225 7.989A2 2 0 0 1 10 14v6a1 1 0 0 0 .553.895l2 1A1 1 0 0 0 14 21v-7a2 2 0 0 1 .517-1.341l.427-.473",
      key: "ol2ft2"
    }
  ],
  ["path", { d: "m16.5 3.5 5 5", key: "15e6fa" }],
  ["path", { d: "m21.5 3.5-5 5", key: "m0lwru" }]
], F = f("FunnelX", b), w = ({
  filters: l,
  filterFields: n,
  onFilterChange: o,
  onSearchChange: c,
  searchTerm: r,
  customFilterComponent: i,
  data: m,
  onResetFilters: p
}) => {
  const d = Object.keys(l).length > 0 || r.length > 0;
  return /* @__PURE__ */ a("div", { className: "rounded-lg border bg-card text-card-foreground shadow-sm mb-6 transition-all", children: [
    /* @__PURE__ */ t("div", { className: "flex flex-col space-y-1.5 p-4 sm:p-6", children: /* @__PURE__ */ a("div", { className: "flex items-center justify-between", children: [
      /* @__PURE__ */ t("h3", { className: "text-lg font-semibold leading-none tracking-tight", children: "Filters" }),
      /* @__PURE__ */ a("div", { className: "flex items-center gap-2", children: [
        d && /* @__PURE__ */ t(
          k.div,
          {
            initial: { opacity: 0, scale: 0.9 },
            animate: { opacity: 1, scale: 1 },
            exit: { opacity: 0, scale: 0.9 },
            children: /* @__PURE__ */ a(x, { variant: "outline", size: "sm", onClick: p, className: "flex items-center gap-1", children: [
              /* @__PURE__ */ t(F, { className: "h-4 w-4" }),
              /* @__PURE__ */ t("span", { children: "Reset" })
            ] })
          }
        ),
        i && /* @__PURE__ */ t("div", { children: i })
      ] })
    ] }) }),
    /* @__PURE__ */ t("div", { className: "p-4 sm:p-6 pt-0", children: /* @__PURE__ */ a("div", { className: "grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4", children: [
      /* @__PURE__ */ t(h, { onSearchChange: c, searchTerm: r }),
      n.map((e) => {
        if (e.type === "select") {
          const s = e.options || y(e.key, m);
          return /* @__PURE__ */ t(
            u,
            {
              options: s,
              field: e,
              onChange: o,
              value: l[e.key] || "all"
            },
            e.key
          );
        }
        return /* @__PURE__ */ a("div", { className: "relative", children: [
          e.icon && /* @__PURE__ */ t("i", { className: `absolute left-2.5 top-2.5 h-4 w-4 text-muted-foreground ${e.icon}` }),
          /* @__PURE__ */ t(
            g,
            {
              type: "text",
              placeholder: e.placeholder || `Filter by ${e.label}...`,
              value: l[e.key] || "",
              onChange: (s) => o(e.key, s.target.value),
              className: v(e.icon && "pl-8", "transition-all focus-visible:ring-2")
            }
          )
        ] }, e.key);
      })
    ] }) })
  ] });
}, z = N.memo(w);
export {
  z as default
};
