// tailgrid/dist/components/TailGrid/Filters/SearchInput.js
import { jsxs as s, jsx as e } from "react/jsx-runtime";
import { Input as n } from "../../ui/input.js";
import { c as l } from "../../../utils-B9KdtH6G.js";
import { Button as m } from "../../ui/button.js";
import p from "react";
import { c as o } from "../../../createLucideIcon-DUBSZsqM.js";
import { A as h } from "../../../index-DiJ3a9ld.js";
import { m as d } from "../../../proxy-g3OKwrzk.js";
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const u = [
  ["circle", { cx: "11", cy: "11", r: "8", key: "4ej97u" }],
  ["path", { d: "m21 21-4.3-4.3", key: "1qie3q" }]
], f = o("Search", u);
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const b = [
  ["path", { d: "M18 6 6 18", key: "1bl5f8" }],
  ["path", { d: "m6 6 12 12", key: "d8bk6v" }]
], v = o("X", b), y = ({ searchTerm: a, onSearchChange: t }) => {
  const r = (i) => {
    t(i.target.value);
  }, c = () => {
    t("");
  };
  return /* @__PURE__ */ s("div", { className: "relative", children: [
    /* @__PURE__ */ e(f, { className: "absolute left-2.5 top-2.5 h-4 w-4 text-muted-foreground" }),
    /* @__PURE__ */ e(
      n,
      {
        type: "search",
        placeholder: "Search...",
        value: a,
        onChange: r,
        className: l(
          "pl-8 pr-8 transition-all",
          "focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-0"
        ),
        "aria-label": "Search table data"
      }
    ),
    /* @__PURE__ */ e(h, { children: a && /* @__PURE__ */ e(
      d.div,
      {
        initial: { opacity: 0, scale: 0.8 },
        animate: { opacity: 1, scale: 1 },
        exit: { opacity: 0, scale: 0.8 },
        transition: { duration: 0.15 },
        className: "absolute right-2 top-2",
        children: /* @__PURE__ */ e(
          m,
          {
            type: "button",
            variant: "ghost",
            size: "icon",
            className: "h-5 w-5",
            onClick: c,
            "aria-label": "Clear search",
            children: /* @__PURE__ */ e(v, { className: "h-3 w-3" })
          }
        )
      }
    ) })
  ] });
}, j = p.memo(y);
export {
  j as default
};
