// tailgrid/dist/components/TailGrid/Table/TableHeader.js
import { jsx as a, jsxs as i } from "react/jsx-runtime";
import { TableHeader as N, TableRow as k, TableHead as c } from "../../ui/table.js";
import { c as s } from "../../../utils-B9KdtH6G.js";
import { memo as v } from "react";
import { m as b } from "../../../proxy-g3OKwrzk.js";
import { c as n } from "../../../createLucideIcon-DUBSZsqM.js";
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const T = [
  ["path", { d: "M12 5v14", key: "s699le" }],
  ["path", { d: "m19 12-7 7-7-7", key: "1idqje" }]
], A = n("ArrowDown", T);
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const H = [
  ["path", { d: "m5 12 7-7 7 7", key: "hav0vg" }],
  ["path", { d: "M12 19V5", key: "x0mq9r" }]
], d = n("ArrowUp", H), j = ({
  columns: l,
  sortConfig: t,
  hasExpandableRows: m,
  hasActions: p,
  onSort: o,
  sortable: r,
  stickyHeader: h = !0,
  headerCellClassName: y = "",
  headerClassName: w = ""
}) => {
  const x = (e) => {
    r && o(e);
  };
  return /* @__PURE__ */ a(N, { className: s(h && "sticky top-0 z-10 bg-card", w), children: /* @__PURE__ */ i(k, { children: [
    m && /* @__PURE__ */ a(c, { className: "w-10" }),
    l.map((e) => /* @__PURE__ */ a(
      c,
      {
        className: s(
          e.align === "end" ? "text-end" : e.align === "center" ? "text-center" : "text-start",
          r && e.sortable ? "cursor-pointer select-none" : "",
          y
        ),
        onClick: () => r && e.sortable && x(e.key),
        "aria-sort": (t == null ? void 0 : t.field) === e.key ? t.direction === "asc" ? "ascending" : "descending" : void 0,
        children: /* @__PURE__ */ i("div", { className: "flex items-center gap-2 group", children: [
          e.header,
          r && e.sortable && /* @__PURE__ */ a("div", { className: "ml-1 flex h-4 w-4 items-center justify-center", children: (t == null ? void 0 : t.field) === e.key ? /* @__PURE__ */ a(
            b.div,
            {
              initial: { opacity: 0, scale: 0.5 },
              animate: { opacity: 1, scale: 1 },
              transition: { duration: 0.2 },
              children: t.direction === "asc" ? /* @__PURE__ */ a(d, { className: "h-3.5 w-3.5 text-primary" }) : /* @__PURE__ */ a(A, { className: "h-3.5 w-3.5 text-primary" })
            }
          ) : /* @__PURE__ */ a("div", { className: "h-3.5 w-3.5 opacity-0 group-hover:opacity-50 transition-opacity", children: /* @__PURE__ */ a(d, { className: "h-3.5 w-3.5" }) }) })
        ] })
      },
      e.key
    )),
    p && /* @__PURE__ */ a(c, { className: "text-end w-[100px]", children: "Actions" })
  ] }) });
}, U = v(j);
export {
  U as default
};
