// tailgrid/dist/components/TailGrid/Table/TableRow.js
import { jsxs as y, Fragment as $, jsx as r } from "react/jsx-runtime";
import j from "./TableActions.js";
import { TableRow as D, TableCell as i } from "../../ui/table.js";
import { c as v } from "../../../utils-B9KdtH6G.js";
import { memo as E } from "react";
import { m as o } from "../../../proxy-g3OKwrzk.js";
import { C as A } from "../../../chevron-down-C5iFRv8a.js";
import { C as F } from "../../../chevron-right-CO-eg3PL.js";
import { A as I } from "../../../index-DiJ3a9ld.js";
const P = ({
  row: s,
  columns: d,
  expandableConfig: t,
  expanded: m,
  onToggleExpand: n,
  actions: c,
  rowKey: a,
  onReload: N,
  isActive: l = !1,
  onClick: h,
  highlightOnHover: R = !0,
  rowClassName: T = "",
  cellClassName: k = ""
}) => {
  const w = (t == null ? void 0 : t.enabled) && m && t.renderExpandedContent(s), u = () => {
    t != null && t.enabled && n(s[a]), h && h(s);
  };
  return /* @__PURE__ */ y($, { children: [
    /* @__PURE__ */ y(
      D,
      {
        className: v(
          t != null && t.enabled || h ? "cursor-pointer" : "",
          "group transition-colors",
          R && "hover:bg-muted/50",
          l && "bg-muted/70",
          T
        ),
        onClick: u,
        tabIndex: 0,
        onKeyDown: (e) => {
          (e.key === "Enter" || e.key === " ") && (e.preventDefault(), u());
        },
        "aria-expanded": t != null && t.enabled ? m : void 0,
        "data-state": l ? "active" : void 0,
        children: [
          (t == null ? void 0 : t.enabled) && /* @__PURE__ */ r(i, { className: "w-10 p-2", children: /* @__PURE__ */ r("div", { className: "flex h-7 w-7 items-center justify-center rounded-md border border-transparent group-hover:border-border", children: /* @__PURE__ */ r(o.div, { animate: { rotate: m ? 90 : 0 }, transition: { duration: 0.2 }, children: m ? /* @__PURE__ */ r(A, { className: "h-4 w-4 text-muted-foreground" }) : /* @__PURE__ */ r(F, { className: "h-4 w-4 text-muted-foreground" }) }) }) }),
          d.map((e) => /* @__PURE__ */ r(
            i,
            {
              className: v(
                e.align === "end" ? "text-end" : e.align === "center" ? "text-center" : "text-start",
                k
              ),
              children: e.render ? e.render(s[e.key], s) : s[e.key]
            },
            `${a}-${e.key}`
          )),
          c.length > 0 && /* @__PURE__ */ r(i, { className: "p-2", children: /* @__PURE__ */ r(j, { actions: c, onReload: N, row: s }) })
        ]
      }
    ),
    /* @__PURE__ */ r(I, { children: (t == null ? void 0 : t.enabled) && m && /* @__PURE__ */ r(
      o.tr,
      {
        initial: { opacity: 0, height: 0 },
        animate: { opacity: 1, height: "auto" },
        exit: { opacity: 0, height: 0 },
        transition: { duration: 0.2 },
        className: "overflow-hidden",
        children: /* @__PURE__ */ r(i, { colSpan: d.length + 1 + (c.length > 0 ? 1 : 0), className: "bg-muted/50 p-0", children: /* @__PURE__ */ r(
          o.div,
          {
            initial: { opacity: 0, y: -10 },
            animate: { opacity: 1, y: 0 },
            transition: { duration: 0.2, delay: 0.1 },
            className: "p-4",
            children: /* @__PURE__ */ r("div", { className: "rounded-md border bg-card p-4", children: w })
          }
        ) })
      }
    ) })
  ] });
}, M = E(P);
export {
  M as default
};
