// tailgrid/dist/components/TailGrid/Table/TableActions.js
import { jsxs as n, jsx as o } from "react/jsx-runtime";
import d from "react";
import { DropdownMenu as h, DropdownMenuTrigger as u, DropdownMenuContent as g, DropdownMenuItem as b } from "../../ui/dropdown-menu.js";
import { Button as c } from "../../ui/button.js";
import { TooltipProvider as f, Tooltip as C, TooltipTrigger as w, TooltipContent as N } from "../../ui/tooltip.js";
import { E as T } from "../../../ellipsis-CU_hPMy5.js";
const v = ({
  actions: r,
  row: s,
  onReload: m
}) => {
  const p = r.slice(0, 2), t = r.length > 2 ? r.slice(2) : [], a = (e, l) => {
    e.stopPropagation(), l.onClick(s);
  };
  return /* @__PURE__ */ n("div", { className: "flex justify-end items-center gap-1", children: [
    p.map((e, l) => /* @__PURE__ */ o(d.Fragment, { children: e.component ? e.component(s, m) : /* @__PURE__ */ o(f, { children: /* @__PURE__ */ n(C, { children: [
      /* @__PURE__ */ o(w, { asChild: !0, children: /* @__PURE__ */ n(
        c,
        {
          variant: "ghost",
          size: "icon",
          onClick: (i) => a(i, e),
          className: "h-8 w-8 rounded-md transition-colors hover:bg-muted",
          "aria-label": e.label || "Action",
          children: [
            e.icon && /* @__PURE__ */ o("span", { className: e.icon }),
            !e.icon && e.label && /* @__PURE__ */ o("span", { className: "sr-only", children: e.label })
          ]
        }
      ) }),
      e.label && /* @__PURE__ */ o(N, { children: e.label })
    ] }) }) }, l)),
    t.length > 0 && /* @__PURE__ */ n(h, { children: [
      /* @__PURE__ */ o(u, { asChild: !0, onClick: (e) => e.stopPropagation(), children: /* @__PURE__ */ o(
        c,
        {
          variant: "ghost",
          size: "icon",
          className: "h-8 w-8 rounded-md transition-colors hover:bg-muted",
          "aria-label": "More actions",
          children: /* @__PURE__ */ o(T, { className: "h-4 w-4" })
        }
      ) }),
      /* @__PURE__ */ o(g, { align: "end", children: t.map((e, l) => /* @__PURE__ */ n(b, { onClick: (i) => a(i, e), className: "cursor-pointer", children: [
        e.icon && /* @__PURE__ */ o("span", { className: `${e.icon} mr-2 h-4 w-4` }),
        e.label
      ] }, l)) })
    ] })
  ] });
}, P = d.memo(v);
export {
  P as default
};
