// tailgrid/dist/components/TailGrid/Table/TableBody.js
import { jsx as e, jsxs as j } from "react/jsx-runtime";
import V from "react";
import $ from "./TableRow.js";
import { TableBody as a, TableRow as B, TableCell as _ } from "../../ui/table.js";
import { useVirtualization as Y } from "../../../hooks/useVirtualization.js";
import { m as q } from "../../../proxy-g3OKwrzk.js";
import { c as C } from "../../../createLucideIcon-DUBSZsqM.js";
/**
 * @license lucide-react v0.483.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const F = [["path", { d: "M21 12a9 9 0 1 1-6.219-8.56", key: "13zald" }]], G = C("LoaderCircle", F), J = ({
  data: t,
  columns: n,
  rowKey: i,
  expandableConfig: m,
  expandedRows: o,
  onToggleExpand: d,
  actions: h,
  isLoading: f,
  colSpan: u,
  customLoadingComponent: p,
  customNoDataComponent: x,
  emptyStateMessage: k = "No data available",
  loadingMessage: A = "Loading data...",
  virtualized: l = !1,
  rowHeight: H = 56,
  activeRowIndex: v,
  onRowClick: y,
  highlightOnHover: N = !0,
  bodyClassName: I = "",
  rowClassName: b = "",
  cellClassName: g = "",
  onReload: T
}) => {
  const L = Y({
    itemCount: l ? t.length : 0,
    itemHeight: H,
    overscan: 5
  }), { virtualItems: z, totalHeight: M, containerRef: R } = l ? L : { virtualItems: [], totalHeight: 0, containerRef: { current: null } };
  return f && p ? /* @__PURE__ */ e(a, { children: p }) : t.length === 0 && x ? /* @__PURE__ */ e(a, { children: x }) : /* @__PURE__ */ e(a, { className: I, children: f ? /* @__PURE__ */ e(B, { children: /* @__PURE__ */ e(_, { colSpan: u, className: "h-24 text-center", children: /* @__PURE__ */ j(
    q.div,
    {
      initial: { opacity: 0 },
      animate: { opacity: 1 },
      className: "flex flex-col items-center justify-center gap-2",
      children: [
        /* @__PURE__ */ e(G, { className: "h-6 w-6 animate-spin text-primary" }),
        /* @__PURE__ */ e("p", { className: "text-sm text-muted-foreground", children: A })
      ]
    }
  ) }) }) : (t == null ? void 0 : t.length) === 0 ? /* @__PURE__ */ e(B, { children: /* @__PURE__ */ e(_, { colSpan: u, className: "h-24 text-center", children: /* @__PURE__ */ j("div", { className: "flex flex-col items-center justify-center gap-1", children: [
    /* @__PURE__ */ e("p", { className: "text-sm font-medium", children: k }),
    /* @__PURE__ */ e("p", { className: "text-sm text-muted-foreground", children: "Try adjusting your filters or search terms" })
  ] }) }) }) : l ? /* @__PURE__ */ e(
    "div",
    {
      ref: R,
      style: { height: "100%", overflow: "auto" },
      className: "relative",
      children: /* @__PURE__ */ e("div", { style: { height: `${M}px`, position: "relative" }, children: z.map(({ index: r, start: c }) => {
        const s = t[r];
        return /* @__PURE__ */ e(
          "div",
          {
            style: {
              position: "absolute",
              top: 0,
              transform: `translateY(${c}px)`,
              width: "100%"
            },
            children: /* @__PURE__ */ e(
              $,
              {
                row: s,
                columns: n,
                expandableConfig: m,
                expanded: o.includes(s[i]),
                onToggleExpand: d,
                actions: h,
                rowKey: i,
                onReload: T,
                isActive: v === r,
                onClick: y,
                highlightOnHover: N,
                rowClassName: b,
                cellClassName: g
              }
            )
          },
          s[i]
        );
      }) })
    }
  ) : t.map((r, c) => /* @__PURE__ */ e(
    $,
    {
      row: r,
      columns: n,
      expandableConfig: m,
      expanded: o.includes(r[i]),
      onToggleExpand: d,
      actions: h,
      rowKey: i,
      onReload: T,
      isActive: v === c,
      onClick: y,
      highlightOnHover: N,
      rowClassName: b,
      cellClassName: g
    },
    r[i]
  )) });
}, E = V.memo(J);
export {
  E as default
};
